---
title: "GOenrich"
author: "Eneritz"
date: "7/28/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
supfigure4 Go terms

```{r}
#####input the liss of genes per group and plot it as a barplot
K14_K9_genes <- read.table("/home/eneritz/KIshared/Molekylär\ Neurobiologi/Castelo-Branco/Eneritz/ASCHRM/ASCHROM_120119/K14_K9_E_gene_list" , sep="\t" ) 
K20_K79_genes <- read.table("/home/eneritz/KIshared/Molekylär\ Neurobiologi/Castelo-Branco/Eneritz/ASCHRM/ASCHROM_120119/K20_K79_E_gene_list" , sep="\t" ) 

K4_K27_genes <- read.table("/home/eneritz/KIshared/Molekylär\ Neurobiologi/Castelo-Branco/Eneritz/ASCHRM/ASCHROM_120119/K4_K27_E_gene_list" , sep="\t" ) 
K9_5mC_E_genes <- read.table("/home/eneritz/KIshared/Molekylär\ Neurobiologi/Castelo-Branco/Eneritz/ASCHRM/ASCHROM_120119/K9_5mC_E_gene_list" , sep="\t" ) 

K20_K91_genes <- read.table("/home/eneritz/KIshared/Molekylär\ Neurobiologi/Castelo-Branco/Eneritz/ASCHRM/ASCHROM_120119/K20_K91_I_gene_list" , sep="\t" )

K4me1_K4me2_genes <- read.table("/home/eneritz/KIshared/Molekylär\ Neurobiologi/Castelo-Branco/Eneritz/ASCHRM/ASCHROM_120119/K4me1_K4me2_I_gene_list" , sep="\t" ) 

K9_5mC_I_genes <- read.table("/home/eneritz/KIshared/Molekylär\ Neurobiologi/Castelo-Branco/Eneritz/ASCHRM/ASCHROM_120119/K9_5mC_I_gene_list" , sep="\t" ) 

Excluded_genes <- read.table("/home/eneritz/KIshared/Molekylär\ Neurobiologi/Castelo-Branco/Eneritz/ASCHRM/ASCHROM_120119/Unique_Genes.EXCLUDED_exons.notSelected" , sep="\t" ) 
Included_genes <- read.table("/home/eneritz/KIshared/Molekylär\ Neurobiologi/Castelo-Branco/Eneritz/ASCHRM/ASCHROM_120119/Unique_Genes.INCLUDED_exons.notSelected" , sep="\t" ) 

library(enrichR)


library(enrichR)
dbs <- listEnrichrDbs()
dbs <- c("GO_Molecular_Function_2018", "GO_Cellular_Component_2018", "GO_Biological_Process_2018" , "Reactome_2016" ,"KEGG_2019_Human")
enriched <- enrichr(as.vector(t(K14_K9_genes)), dbs)
#printEnrich(enriched, "output.txt" , sep = "\t", columns = c(1:9))

bp1 <- enriched[["GO_Biological_Process_2018"]]

bp2 <- enriched[["Reactome_2016"]]

bp3 <- enriched[["KEGG_2019_Human"]]

write.table( bp1 , file =  "/home/eneritz/ASCHROM0818/GO_terms_supfig4/K14_K9_E_GOBP.txt", sep="\t",row.names = T, col.names = TRUE, quote = FALSE)
write.table( bp2 , file =  "/home/eneritz/ASCHROM0818/GO_terms_supfig4/K14_K9_E_reactome16.txt", sep="\t",row.names = T, col.names = TRUE, quote = FALSE)
write.table( bp3 , file =  "/home/eneritz/ASCHROM0818/GO_terms_supfig4/K14_K9_E_KEGG19.txt", sep="\t",row.names = T, col.names = TRUE, quote = FALSE)

enriched <- enrichr(as.vector(t(K20_K79_genes)), dbs)
#printEnrich(enriched, "output.txt" , sep = "\t", columns = c(1:9))

bp1 <- enriched[["GO_Biological_Process_2018"]]

bp2 <- enriched[["Reactome_2016"]]

bp3 <- enriched[["KEGG_2019_Human"]]

write.table( bp1 , file =  "/home/eneritz/ASCHROM0818/GO_terms_supfig4/K20_K79_E_GOBP.txt", sep="\t",row.names = T, col.names = TRUE, quote = FALSE)
write.table( bp2 , file =  "/home/eneritz/ASCHROM0818/GO_terms_supfig4/K20_K79_E_reactome16.txt", sep="\t",row.names = T, col.names = TRUE, quote = FALSE)
write.table( bp3 , file =  "/home/eneritz/ASCHROM0818/GO_terms_supfig4/K20_K79_E_KEGG19.txt", sep="\t",row.names = T, col.names = TRUE, quote = FALSE)



enriched <- enrichr(as.vector(t(K4_K27_genes)), dbs)
#printEnrich(enriched, "output.txt" , sep = "\t", columns = c(1:9))

bp1 <- enriched[["GO_Biological_Process_2018"]]

bp2 <- enriched[["Reactome_2016"]]

bp3 <- enriched[["KEGG_2019_Human"]]

write.table( bp1 , file =  "/home/eneritz/ASCHROM0818/GO_terms_supfig4/K4_K27_E_GOBP.txt", sep="\t",row.names = T, col.names = TRUE, quote = FALSE)
write.table( bp2 , file =  "/home/eneritz/ASCHROM0818/GO_terms_supfig4/K4_K27_E_reactome16.txt", sep="\t",row.names = T, col.names = TRUE, quote = FALSE)
write.table( bp3 , file =  "/home/eneritz/ASCHROM0818/GO_terms_supfig4/K4_K27_E_KEGG19.txt", sep="\t",row.names = T, col.names = TRUE, quote = FALSE)



enriched <- enrichr(as.vector(t(K9_5mC_E_genes)), dbs)
#printEnrich(enriched, "output.txt" , sep = "\t", columns = c(1:9))

bp1 <- enriched[["GO_Biological_Process_2018"]]

bp2 <- enriched[["Reactome_2016"]]

bp3 <- enriched[["KEGG_2019_Human"]]

write.table( bp1 , file =  "/home/eneritz/ASCHROM0818/GO_terms_supfig4/K9_5mC_E_GOBP.txt", sep="\t",row.names = T, col.names = TRUE, quote = FALSE)
write.table( bp2 , file =  "/home/eneritz/ASCHROM0818/GO_terms_supfig4/K9_5mC_E_reactome16.txt", sep="\t",row.names = T, col.names = TRUE, quote = FALSE)
write.table( bp3 , file =  "/home/eneritz/ASCHROM0818/GO_terms_supfig4/K9_5mC_E_KEGG19.txt", sep="\t",row.names = T, col.names = TRUE, quote = FALSE)



enriched <- enrichr(as.vector(t(K20_K91_genes)), dbs)
#printEnrich(enriched, "output.txt" , sep = "\t", columns = c(1:9))

bp1 <- enriched[["GO_Biological_Process_2018"]]

bp2 <- enriched[["Reactome_2016"]]

bp3 <- enriched[["KEGG_2019_Human"]]

write.table( bp1 , file =  "/home/eneritz/ASCHROM0818/GO_terms_supfig4/K20_K91_GOBP.txt", sep="\t",row.names = T, col.names = TRUE, quote = FALSE)
write.table( bp2 , file =  "/home/eneritz/ASCHROM0818/GO_terms_supfig4/K20_K91_reactome16.txt", sep="\t",row.names = T, col.names = TRUE, quote = FALSE)
write.table( bp3 , file =  "/home/eneritz/ASCHROM0818/GO_terms_supfig4/K20_K91_KEGG19.txt", sep="\t",row.names = T, col.names = TRUE, quote = FALSE)



enriched <- enrichr(as.vector(t(K4me1_K4me2_genes)), dbs)
#printEnrich(enriched, "output.txt" , sep = "\t", columns = c(1:9))

bp1 <- enriched[["GO_Biological_Process_2018"]]

bp2 <- enriched[["Reactome_2016"]]

bp3 <- enriched[["KEGG_2019_Human"]]

write.table( bp1 , file =  "/home/eneritz/ASCHROM0818/GO_terms_supfig4/K4me1_K4me2_GOBP.txt", sep="\t",row.names = T, col.names = TRUE, quote = FALSE)
write.table( bp2 , file =  "/home/eneritz/ASCHROM0818/GO_terms_supfig4/K4me1_K4me2_reactome16.txt", sep="\t",row.names = T, col.names = TRUE, quote = FALSE)
write.table( bp3 , file =  "/home/eneritz/ASCHROM0818/GO_terms_supfig4/K4me1_K4me2_KEGG19.txt", sep="\t",row.names = T, col.names = TRUE, quote = FALSE)




enriched <- enrichr(as.vector(t(K9_5mC_I_genes)), dbs)
#printEnrich(enriched, "output.txt" , sep = "\t", columns = c(1:9))

bp1 <- enriched[["GO_Biological_Process_2018"]]

bp2 <- enriched[["Reactome_2016"]]

bp3 <- enriched[["KEGG_2019_Human"]]

write.table( bp1 , file =  "/home/eneritz/ASCHROM0818/GO_terms_supfig4/K9_5mC_I_GOBP.txt", sep="\t",row.names = T, col.names = TRUE, quote = FALSE)
write.table( bp2 , file =  "/home/eneritz/ASCHROM0818/GO_terms_supfig4/K9_5mC_I_reactome16.txt", sep="\t",row.names = T, col.names = TRUE, quote = FALSE)
write.table( bp3 , file =  "/home/eneritz/ASCHROM0818/GO_terms_supfig4/K9_5mC_I_KEGG19.txt", sep="\t",row.names = T, col.names = TRUE, quote = FALSE)



enriched <- enrichr(as.vector(t(Included_genes)), dbs)
#printEnrich(enriched, "output.txt" , sep = "\t", columns = c(1:9))

bp1 <- enriched[["GO_Biological_Process_2018"]]

bp2 <- enriched[["Reactome_2016"]]

bp3 <- enriched[["KEGG_2019_Human"]]

write.table( bp1 , file =  "/home/eneritz/ASCHROM0818/GO_terms_supfig4/Included_GOBP.txt", sep="\t",row.names = T, col.names = TRUE, quote = FALSE)
write.table( bp2 , file =  "/home/eneritz/ASCHROM0818/GO_terms_supfig4/Included_reactome16.txt", sep="\t",row.names = T, col.names = TRUE, quote = FALSE)
write.table( bp3 , file =  "/home/eneritz/ASCHROM0818/GO_terms_supfig4/Included_KEGG19.txt", sep="\t",row.names = T, col.names = TRUE, quote = FALSE)




enriched <- enrichr(as.vector(t(Excluded_genes)), dbs)
#printEnrich(enriched, "output.txt" , sep = "\t", columns = c(1:9))

bp1 <- enriched[["GO_Biological_Process_2018"]]

bp2 <- enriched[["Reactome_2016"]]

bp3 <- enriched[["KEGG_2019_Human"]]

write.table( bp1 , file =  "/home/eneritz/ASCHROM0818/GO_terms_supfig4/Excluded_GOBP.txt", sep="\t",row.names = T, col.names = TRUE, quote = FALSE)
write.table( bp2 , file =  "/home/eneritz/ASCHROM0818/GO_terms_supfig4/Excluded_reactome16.txt", sep="\t",row.names = T, col.names = TRUE, quote = FALSE)
write.table( bp3 , file =  "/home/eneritz/ASCHROM0818/GO_terms_supfig4/Excluded_KEGG19.txt", sep="\t",row.names = T, col.names = TRUE, quote = FALSE)




 df <- bp1[order(bp1$Adjusted.P.value,decreasing = TRUE),]

df <- df[df$Adjusted.P.value < 0.2 , ] 
 
# library(ggplot2)
  ggplot(data=df, aes(x=Term, y=Z.score))+ #, fill=condition)) +
  geom_bar(stat="identity")+
    
coord_cartesian(ylim = c(0, 1) ) +
     coord_flip()
   # geom_text(aes(y=label_ypos, label=pcount), vjust=1.6, 
  #          color="white", size=3.5)+
  #scale_fill_brewer(palette="Paired")+
theme(axis.text.x = element_text(angle = 70, hjust = 1))
      theme_minimal()
      


```

```{r}

#cluster2 K14_K9_E_gene_list
#cluster3 K4_K27_E_gene_list
#cluster4 K20_K79_E_gene_list
#cluster5 K9_5mC_E_gene_list
#
#cluster2 K9_5mC_I_gene_list
#cluster3 K20_K91_I_gene_list
#cluster4 K4me1_K4me2_I_gene_list
#
#awk 'BEGIN {OFS="\t"}; {print $_ }' Inclusion_GO_noFIlter.tbl | egrep None > lalaIN
#awk 'BEGIN {OFS="\t"}; {print $_ }' Inclusion_GO_noFIlter.tbl | egrep Cluster_2 > lalaI2
#awk 'BEGIN {OFS="\t"}; {print $_ }' Inclusion_GO_noFIlter.tbl | egrep Cluster_3 > lalaI3
#awk 'BEGIN {OFS="\t"}; {print $_ }' Inclusion_GO_noFIlter.tbl | egrep Cluster_4 > lalaI4
#
#awk 'BEGIN {OFS="\t"}; {print $_ }' Exclusion_GO_noFIlter.tbl | egrep None > lalaN
#awk 'BEGIN {OFS="\t"}; {print $_ }' Exclusion_GO_noFIlter.tbl | egrep Cluster_2 > lala2
#awk 'BEGIN {OFS="\t"}; {print $_ }' Exclusion_GO_noFIlter.tbl | egrep Cluster_3 > lala3
#awk 'BEGIN {OFS="\t"}; {print $_ }' Exclusion_GO_noFIlter.tbl | egrep Cluster_4 > lala4
#awk 'BEGIN {OFS="\t"}; {print $_ }' Exclusion_GO_noFIlter.tbl | egrep Cluster_5 > lala5


K14_K9_genesC <- read.table("/home/eneritz/ASCHROM0818/lala2" , sep="\t" ) 
K20_K79_genesC <- read.table("/home/eneritz/ASCHROM0818/lala4" , sep="\t" ) 

#there are not
#K4_K27_genes <- read.table("/home/eneritz/ASCHROM0818/lala3" , sep="\t" ) 
K9_5mC_E_genesC <- read.table("/home/eneritz/ASCHROM0818/lala5" , sep="\t" ) 

K20_K91_genesC <- read.table("/home/eneritz/ASCHROM0818/lalaI3" , sep="\t" )

K4me1_K4me2_genesC <- read.table("/home/eneritz/ASCHROM0818/lalaI4" , sep="\t" ) 

K9_5mC_I_genesC <- read.table("/home/eneritz/ASCHROM0818/lalaI2" , sep="\t" ) 





 df <- bp1[order(bp1$Adjusted.P.value,decreasing = TRUE),]

df <- df[df$Adjusted.P.value < 0.2 , ] 
 
# library(ggplot2)
  ggplot(data=df, aes(x=Term, y=Z.score))+ #, fill=condition)) +
  geom_bar(stat="identity")+
    
coord_cartesian(ylim = c(0, 1) ) +
     coord_flip()
   # geom_text(aes(y=label_ypos, label=pcount), vjust=1.6, 
  #          color="white", size=3.5)+
  #scale_fill_brewer(palette="Paired")+
theme(axis.text.x = element_text(angle = 70, hjust = 1))
      theme_minimal()
      


```
K9 K14 excluded revigo plot
```{r}
# --------------------------------------------------------------------------
# If you don't have the ggplot2 package installed, uncomment the following line:
# install.packages( "ggplot2" );
library( ggplot2 );
# --------------------------------------------------------------------------
# If you don't have the scales package installed, uncomment the following line:
# install.packages( "scales" );
library( scales );


# --------------------------------------------------------------------------
# Here is your data from REVIGO. Scroll down for plot configuration options.

revigo.names <- c("term_ID","description","frequency_%","plot_X","plot_Y","plot_size","log10_p_value","uniqueness","dispensability");
revigo.data <- rbind(c("GO:0000051","Fructose and mannose metabolism", 0.559,-5.332,-0.406, 1.991,-0.9208,1.000,0.000),
c("GO:0010256","Endomembrane system organization", 3.353, 5.885,-0.660, 2.765,-3.0000,0.324,0.000),
c("GO:0097581","Lamellipodium organization", 0.427, 2.902, 4.125, 1.875,-0.5850,0.327,0.150),
c("GO:0007009","Plasma membrane organization", 1.662, 3.066,-4.252, 2.461,-0.4949,0.321,0.177));

one.data <- data.frame(revigo.data);
names(one.data) <- revigo.names;
one.data <- one.data [(one.data$plot_X != "null" & one.data$plot_Y != "null"), ];
one.data$plot_X <- as.numeric( as.character(one.data$plot_X) );
one.data$plot_Y <- as.numeric( as.character(one.data$plot_Y) );
one.data$plot_size <- as.numeric( as.character(one.data$plot_size) );
one.data$log10_p_value <- as.numeric( as.character(one.data$log10_p_value) );
one.data$frequency <- as.numeric( as.character(one.data$frequency) );
one.data$uniqueness <- as.numeric( as.character(one.data$uniqueness) );
one.data$dispensability <- as.numeric( as.character(one.data$dispensability) );
#head(one.data);


# --------------------------------------------------------------------------
# Names of the axes, sizes of the numbers and letters, names of the columns,
# etc. can be changed below

p1 <- ggplot( data = one.data );
p1 <- p1 + geom_point( aes( plot_X, plot_Y, colour = log10_p_value, size = plot_size), alpha = I(0.6) ) + scale_size_area();
p1 <- p1 + scale_colour_gradientn( colours = c("blue", "green", "yellow", "red"), limits = c( min(one.data$log10_p_value), 0) );
p1 <- p1 + geom_point( aes(plot_X, plot_Y, size = plot_size), shape = 21, fill = "transparent", colour = I (alpha ("black", 0.6) )) + scale_size_area();
p1 <- p1 + scale_size( range=c(5, 30)) + theme_bw(); # + scale_fill_gradientn(colours = heat_hcl(7), limits = c(-300, 0) );
ex <- one.data [ one.data$dispensability < 0.6, ]; 
p1 <- p1 + geom_text_repel( data = ex, aes(plot_X, plot_Y, label = description), colour = I(alpha("black", 0.85)), size = 4 );
p1 <- p1 + labs (y = "semantic space x", x = "semantic space y");
p1 <- p1 + theme(legend.key = element_blank()) ;
one.x_range = max(one.data$plot_X) - min(one.data$plot_X);
one.y_range = max(one.data$plot_Y) - min(one.data$plot_Y);
p1 <- p1 + xlim(min(one.data$plot_X)-one.x_range/10,max(one.data$plot_X)+one.x_range/10);
p1 <- p1 + ylim(min(one.data$plot_Y)-one.y_range/10,max(one.data$plot_Y)+one.y_range/10);

cairo_ps(file = "/home/eneritz/ASCHROM1909/FIGURES0919/GO_figure/SACS7_GO.eps", onefile = FALSE, fallback_resolution = 3000 , width=6, height=8 )
p1
dev.off()

# --------------------------------------------------------------------------
# Output the plot to screen

p1;

# Uncomment the line below to also save the plot to a file.
# The file type depends on the extension (default=pdf).

# ggsave("C:/Users/path_to_your_file/revigo-plot.pdf");

##########################################################


# --------------------------------------------------------------------------
# If you don't have the ggplot2 package installed, uncomment the following line:
# install.packages( "ggplot2" );
library( ggplot2 );
# --------------------------------------------------------------------------
# If you don't have the scales package installed, uncomment the following line:
# install.packages( "scales" );
library( scales );


# --------------------------------------------------------------------------
# Here is your data from REVIGO. Scroll down for plot configuration options.

revigo.names <- c("term_ID","description","frequency_%","plot_X","plot_Y","plot_size","log10_p_value","uniqueness","dispensability");
revigo.data <- rbind(c("GO:0002478","antigen processing and presentation of exogenous peptide antigen", 0.004, 0.575, 1.266, 2.667,-1.6185,0.965,0.000),
c("GO:0032886","regulation of microtubule-based process", 0.067,-3.619,-2.500, 3.934,-3.1942,0.741,0.000),
c("GO:0099515","actin filament-based transport", 0.002, 3.619, 5.787, 2.468,-2.6806,0.903,0.000),
c("GO:0006468","protein phosphorylation", 4.137,-0.368,-2.886, 5.725,-2.5236,0.867,0.031),
c("GO:0016032","viral process", 0.131, 2.556, 1.318, 4.226,-1.8833,0.955,0.033),
c("GO:0044772","mitotic cell cycle phase transition", 0.201, 3.170,-0.724, 4.411,-2.5449,0.839,0.138),
c("GO:0002115","store-operated calcium entry", 0.007, 2.613, 5.178, 2.953,-1.3071,0.921,0.141),
c("GO:1903551","regulation of extracellular exosome assembly", 0.001, 2.330,-5.635, 2.212,-3.0915,0.694,0.146),
c("GO:1904705","regulation of vascular smooth muscle cell proliferation", 0.004,-3.667,-0.672, 2.665,-1.9358,0.808,0.154),
c("GO:0008016","regulation of heart contraction", 0.036,-5.574, 0.261, 3.664,-1.6417,0.735,0.174),
c("GO:0030336","negative regulation of cell migration", 0.043,-2.409, 4.692, 3.743,-2.9767,0.6,0.18), ###dispensability was 0.18, for plotting option
c("GO:0060070","canonical Wnt signaling pathway", 0.061,-6.013,-0.935, 3.893,-2.9103,0.701,0.184),
c("GO:0016197","endosomal transport", 0.131, 2.927, 5.683, 4.225,-1.8151,0.923,0.191),
c("GO:0043547","positive regulation of GTPase activity", 0.470,-5.765,-1.046, 4.780,-2.8502,0.775,0.199),
c("GO:0035461","vitamin transmembrane transport", 0.065, 3.208, 6.071, 3.921,-1.2128,0.863,0.205),
c("GO:0010942","positive regulation of cell death", 0.114,-4.685,-1.755, 4.165,-2.2326,0.669,0.208),
c("GO:0050657","nucleic acid transport", 0.100, 2.777, 5.742, 4.108,-1.3071,0.909,0.211),
c("GO:0010467","gene expression",19.671,-0.284,-0.774, 6.402,-1.8069,0.950,0.234),
c("GO:0022603","regulation of anatomical structure morphogenesis", 0.830,-6.006,-0.130, 5.027,-2.7259,0.690,0.248),
c("GO:0008361","regulation of cell size", 0.044, 0.918,-5.247, 3.749,-2.0032,0.693,0.256),
c("GO:0016192","vesicle-mediated transport", 1.085, 2.995, 6.095, 5.144,-1.3383,0.911,0.256),
c("GO:0006357","regulation of transcription from RNA polymerase II promoter", 1.273,-5.891,-2.408, 5.213,-1.6284,0.724,0.289),
c("GO:0002183","cytoplasmic translational initiation", 0.025,-0.544,-2.863, 3.501,-1.1029,0.904,0.290),
c("GO:0048384","retinoic acid receptor signaling pathway", 0.007,-6.608,-0.426, 2.970,-1.3071,0.774,0.299),
c("GO:0034614","cellular response to reactive oxygen species", 0.151,-3.652, 0.498, 4.286,-2.0773,0.846,0.305),
c("GO:0019935","cyclic-nucleotide-mediated signaling", 0.013,-6.668,-0.470, 3.230,-1.7375,0.759,0.310),
c("GO:0035020","regulation of Rac protein signal transduction", 0.002,-7.029,-0.504, 2.494,-1.2128,0.752,0.316),
c("GO:0000381","regulation of alternative mRNA splicing, via spliceosome", 0.017,-4.924,-2.095, 3.344,-1.5804,0.787,0.318),
c("GO:0098970","postsynaptic neurotransmitter receptor diffusion trapping", 0.000, 1.277, 5.288, 1.544,-1.2573,0.801,0.324),
c("GO:0061621","canonical glycolysis", 0.001, 0.194,-1.255, 2.199,-1.9358,0.839,0.325),
c("GO:0006002","fructose 6-phosphate metabolic process", 0.060, 0.145,-0.784, 3.885,-1.3637,0.933,0.333),
c("GO:0007030","Golgi organization", 0.039, 4.994,-4.794, 3.701,-2.1279,0.801,0.335),
c("GO:0035608","protein deglutamylation", 0.002, 0.027,-2.307, 2.375,-1.3071,0.922,0.335),
c("GO:0006655","phosphatidylglycerol biosynthetic process", 0.039, 0.552,-1.666, 3.696,-1.1728,0.831,0.343),
c("GO:0034720","histone H3-K4 demethylation", 0.003, 3.502,-5.542, 2.591,-1.3071,0.768,0.348),
c("GO:0090527","actin filament reorganization", 0.003, 5.207,-4.869, 2.534,-1.2128,0.755,0.349),
c("GO:0048041","focal adhesion assembly", 0.012, 4.865,-5.219, 3.190,-2.2022,0.764,0.355),
c("GO:0055064","chloride ion homeostasis", 0.003,-3.150, 0.566, 2.589,-1.1728,0.837,0.360),
c("GO:0034389","lipid particle organization", 0.005, 5.429,-4.921, 2.823,-1.0435,0.834,0.362),
c("GO:0003374","dynamin family protein polymerization involved in mitochondrial fission", 0.002, 4.961,-4.910, 2.449,-1.2128,0.774,0.365),
c("GO:0016052","carbohydrate catabolic process", 1.078, 2.168, 0.675, 5.141,-1.6661,0.949,0.371),
c("GO:0071499","cellular response to laminar fluid shear stress", 0.001,-3.109, 0.358, 2.143,-1.3071,0.896,0.381),
c("GO:0070936","protein K48-linked ubiquitination", 0.011, 0.032,-2.400, 3.157,-1.4179,0.922,0.385),
c("GO:0043545","molybdopterin cofactor metabolic process", 0.223, 0.138,-2.874, 4.457,-1.3071,0.883,0.387),
c("GO:0016070","RNA metabolic process",15.951,-0.284,-2.404, 6.311,-1.4768,0.896,0.390),
c("GO:0035304","regulation of protein dephosphorylation", 0.027,-3.921,-3.111, 3.537,-1.5605,0.751,0.414),
c("GO:0042060","wound healing", 0.094,-3.902, 0.800, 4.079,-1.3864,0.907,0.437),
c("GO:0060147","regulation of posttranscriptional gene silencing", 0.009,-5.555,-1.324, 3.039,-1.3131,0.716,0.449),
c("GO:0061668","mitochondrial ribosome assembly", 0.001, 5.007,-4.969, 2.170,-1.3637,0.820,0.453),
c("GO:0016482","cytosolic transport", 0.076, 3.099, 5.590, 3.988,-2.1655,0.885,0.458),
c("GO:0018105","peptidyl-serine phosphorylation", 0.087, 0.107,-2.899, 4.049,-1.8622,0.892,0.462),
c("GO:0007185","transmembrane receptor protein tyrosine phosphatase signaling pathway", 0.000,-5.938,-0.380, 1.740,-1.3637,0.785,0.470),
c("GO:0034394","protein localization to cell surface", 0.013, 3.589, 6.445, 3.207,-2.0392,0.888,0.472),
c("GO:0043043","peptide biosynthetic process", 5.770, 0.353,-0.357, 5.869,-1.6043,0.900,0.475),
c("GO:0045945","positive regulation of transcription from RNA polymerase III promoter", 0.004,-4.802,-1.868, 2.754,-1.1728,0.734,0.485),
c("GO:0036500","ATF6-mediated unfolded protein response", 0.001,-6.571,-0.268, 1.826,-1.2128,0.772,0.488),
c("GO:0072311","glomerular epithelial cell differentiation", 0.005,-0.220, 1.690, 2.827,-1.3637,0.777,0.489),
c("GO:0003230","cardiac atrium development", 0.008,-0.605, 1.442, 2.999,-1.3071,0.826,0.502),
c("GO:1901880","negative regulation of protein depolymerization", 0.039,-0.277,-5.474, 3.698,-2.1166,0.605,0.504),
c("GO:0007010","cytoskeleton organization", 0.786, 5.020,-4.861, 5.004,-2.0677,0.778,0.505),
c("GO:0006896","Golgi to vacuole transport", 0.012, 3.640, 6.252, 3.200,-1.0435,0.876,0.506),
c("GO:0070875","positive regulation of glycogen metabolic process", 0.004,-4.610,-1.572, 2.719,-1.0435,0.707,0.507),
c("GO:0010887","negative regulation of cholesterol storage", 0.001,-1.787, 4.542, 2.100,-1.3637,0.705,0.507),
c("GO:0000165","MAPK cascade", 0.245,-4.118,-2.789, 4.498,-1.4966,0.662,0.514),
c("GO:0010038","response to metal ion", 0.132,-3.847, 0.835, 4.229,-1.1107,0.898,0.514),
c("GO:1905515","non-motile cilium assembly", 0.018, 4.894,-5.056, 3.368,-2.0770,0.735,0.516),
c("GO:0070848","response to growth factor", 0.150,-3.846, 0.811, 4.284,-1.9042,0.891,0.519),
c("GO:1990778","protein localization to cell periphery", 0.063, 3.268, 6.288, 3.905,-1.4131,0.878,0.524),
c("GO:0030049","muscle filament sliding", 0.001, 0.512, 0.913, 2.286,-1.6009,0.776,0.524),
c("GO:0046164","alcohol catabolic process", 0.140, 1.195, 0.215, 4.255,-1.1364,0.891,0.529),
c("GO:0015939","pantothenate metabolic process", 0.144, 1.642,-1.126, 4.265,-1.2573,0.831,0.532),
c("GO:0044267","cellular protein metabolic process",14.293,-0.462,-3.417, 6.263,-1.0631,0.887,0.536),
c("GO:0034969","histone arginine methylation", 0.003, 3.726,-5.833, 2.649,-1.1029,0.802,0.539),
c("GO:0051342","regulation of cyclic-nucleotide phosphodiesterase activity", 0.001,-2.979,-0.797, 1.833,-1.3071,0.846,0.539),
c("GO:2000480","negative regulation of cAMP-dependent protein kinase activity", 0.002,-3.978,-2.717, 2.394,-1.2573,0.711,0.539),
c("GO:0072337","modified amino acid transport", 0.042, 3.006, 5.992, 3.736,-1.0435,0.915,0.540),
c("GO:0051128","regulation of cellular component organization", 1.586,-0.781,-5.428, 5.308,-1.9130,0.656,0.542),
c("GO:0071243","cellular response to arsenic-containing substance", 0.004,-3.885, 0.924, 2.717,-1.1029,0.888,0.548),
c("GO:1901185","negative regulation of ERBB signaling pathway", 0.007,-7.065,-0.461, 2.965,-2.5722,0.675,0.548),
c("GO:0018107","peptidyl-threonine phosphorylation", 0.032, 0.202,-2.846, 3.618,-1.1530,0.898,0.549),
c("GO:0042147","retrograde transport, endosome to Golgi", 0.055, 3.479, 6.180, 3.847,-1.8080,0.858,0.557),
c("GO:0080135","regulation of cellular response to stress", 0.182,-6.293,-0.491, 4.368,-2.3567,0.710,0.560),
c("GO:0034454","microtubule anchoring at centrosome", 0.003, 4.807,-4.885, 2.624,-1.2128,0.761,0.566),
c("GO:0055001","muscle cell development", 0.041,-0.203, 0.841, 3.717,-2.4672,0.793,0.568),
c("GO:0000184","nuclear-transcribed mRNA catabolic process, nonsense-mediated decay", 0.030, 0.306,-0.537, 3.588,-1.4603,0.907,0.571),
c("GO:0090091","positive regulation of extracellular matrix disassembly", 0.001, 1.537,-5.858, 2.262,-1.3071,0.660,0.572),
c("GO:0035137","hindlimb morphogenesis", 0.008,-0.663, 1.201, 2.997,-1.3637,0.836,0.576),
c("GO:0048048","embryonic eye morphogenesis", 0.009,-0.337, 2.056, 3.053,-2.6032,0.815,0.581),
c("GO:0032331","negative regulation of chondrocyte differentiation", 0.004,-5.324, 0.284, 2.750,-1.1029,0.649,0.589),
c("GO:0018209","peptidyl-serine modification", 0.093, 0.649,-3.515, 4.078,-1.6366,0.903,0.591),
c("GO:0000389","mRNA 3'-splice site recognition", 0.002, 4.242,-5.321, 2.449,-1.3637,0.799,0.592),
c("GO:0006909","phagocytosis", 0.051, 2.946, 5.463, 3.813,-1.2615,0.845,0.593),
c("GO:0045921","positive regulation of exocytosis", 0.017,-1.255, 5.043, 3.344,-2.8604,0.614,0.596),
c("GO:2000479","regulation of cAMP-dependent protein kinase activity", 0.004,-3.396,-2.999, 2.677,-1.0721,0.757,0.599),
c("GO:0043393","regulation of protein binding", 0.038,-4.496,-1.027, 3.686,-1.4701,0.807,0.603),
c("GO:0034505","tooth mineralization", 0.004,-0.413, 1.689, 2.665,-1.3071,0.829,0.605),
c("GO:0090286","cytoskeletal anchoring at nuclear membrane", 0.004, 4.400, 0.581, 2.668,-1.2573,0.639,0.610),
c("GO:0048172","regulation of short-term neuronal synaptic plasticity", 0.003,-4.881,-0.131, 2.540,-1.1364,0.752,0.612),
c("GO:0016310","phosphorylation", 7.764, 0.506,-1.325, 5.998,-1.4443,0.913,0.615),
c("GO:0060336","negative regulation of interferon-gamma-mediated signaling pathway", 0.001,-6.258,-0.076, 1.845,-1.3637,0.679,0.615),
c("GO:0043031","negative regulation of macrophage activation", 0.002,-5.097,-0.523, 2.324,-1.3071,0.740,0.615),
c("GO:0051294","establishment of spindle orientation", 0.014, 6.122, 0.604, 3.245,-2.0770,0.652,0.616),
c("GO:0090342","regulation of cell aging", 0.009,-4.061,-0.183, 3.081,-1.0721,0.729,0.621),
c("GO:0007005","mitochondrion organization", 0.418, 4.996,-4.832, 4.729,-1.0458,0.787,0.621),
c("GO:0099590","neurotransmitter receptor internalization", 0.001, 3.060, 5.075, 1.973,-1.1728,0.817,0.622),
c("GO:0006355","regulation of transcription, DNA-templated", 9.917,-5.672,-2.491, 6.105,-1.6390,0.679,0.622),
c("GO:0034115","negative regulation of heterotypic cell-cell adhesion", 0.002,-4.401,-0.288, 2.318,-1.2128,0.676,0.625),
c("GO:0007420","brain development", 0.142,-0.659, 1.334, 4.260,-2.1528,0.799,0.631),
c("GO:0018212","peptidyl-tyrosine modification", 0.215, 0.510,-3.575, 4.441,-1.3711,0.898,0.631),
c("GO:0010745","negative regulation of macrophage derived foam cell differentiation", 0.002,-4.080, 0.004, 2.348,-1.0721,0.694,0.634),
c("GO:0070296","sarcoplasmic reticulum calcium ion transport", 0.005, 2.689, 5.305, 2.766,-1.2128,0.923,0.635),
c("GO:0051006","positive regulation of lipoprotein lipase activity", 0.002,-3.421,-0.865, 2.400,-1.2573,0.832,0.637),
c("GO:0031116","positive regulation of microtubule polymerization", 0.009, 1.083,-5.688, 3.045,-1.9688,0.572,0.637),
c("GO:0006768","biotin metabolic process", 0.081, 1.396,-0.993, 4.018,-1.1364,0.837,0.638),
c("GO:0031016","pancreas development", 0.021,-0.642, 1.433, 3.438,-1.0435,0.820,0.642),
c("GO:0034453","microtubule anchoring", 0.010, 4.748,-4.934, 3.128,-1.1029,0.749,0.644),
c("GO:0005513","detection of calcium ion", 0.001,-2.976, 0.530, 2.262,-1.0721,0.919,0.645),
c("GO:0071318","cellular response to ATP", 0.002,-3.607, 0.744, 2.338,-1.1728,0.880,0.646),
c("GO:1902115","regulation of organelle assembly", 0.051, 0.943,-5.965, 3.815,-1.4343,0.645,0.648),
c("GO:0010640","regulation of platelet-derived growth factor receptor signaling pathway", 0.004,-7.076,-0.535, 2.711,-1.1728,0.735,0.649),
c("GO:0006511","ubiquitin-dependent protein catabolic process", 0.584,-0.136,-2.366, 4.874,-1.1843,0.885,0.651),
c("GO:0010959","regulation of metal ion transport", 0.069,-1.707, 4.942, 3.948,-2.7561,0.711,0.654),
c("GO:0010875","positive regulation of cholesterol efflux", 0.003,-0.850, 4.960, 2.579,-1.0721,0.671,0.657),
c("GO:0032268","regulation of cellular protein metabolic process", 1.473,-5.646,-3.115, 5.277,-1.3864,0.705,0.659),
c("GO:0006897","endocytosis", 0.235, 3.200, 6.301, 4.480,-1.0663,0.894,0.660),
c("GO:0034145","positive regulation of toll-like receptor 4 signaling pathway", 0.002,-6.302,-0.534, 2.384,-1.2573,0.668,0.660),
c("GO:0061620","glycolytic process through glucose-6-phosphate", 0.002, 0.150,-1.274, 2.290,-1.9358,0.841,0.660),
c("GO:0032368","regulation of lipid transport", 0.020,-0.927, 5.516, 3.412,-1.2573,0.701,0.661),
c("GO:0000045","autophagosome assembly", 0.035, 5.065,-4.950, 3.647,-1.2726,0.792,0.661),
c("GO:0051017","actin filament bundle assembly", 0.045, 4.696,-4.900, 3.762,-1.6220,0.704,0.662),
c("GO:0032816","positive regulation of natural killer cell activation", 0.004,-5.057,-0.985, 2.676,-1.1728,0.718,0.664),
c("GO:0007512","adult heart development", 0.002,-0.578, 1.392, 2.505,-1.2128,0.836,0.664),
c("GO:0018108","peptidyl-tyrosine phosphorylation", 0.188, 0.055,-2.985, 4.381,-1.0427,0.886,0.667),
c("GO:0010557","positive regulation of macromolecule biosynthetic process", 0.633,-5.989,-1.505, 4.909,-1.8924,0.668,0.672),
c("GO:0051647","nucleus localization", 0.012, 3.140, 5.346, 3.198,-1.3637,0.889,0.681),
c("GO:0072657","protein localization to membrane", 0.387, 5.496, 1.028, 4.696,-1.7216,0.652,0.684),
c("GO:0060307","regulation of ventricular cardiac muscle cell membrane repolarization", 0.004,-1.076, 4.917, 2.666,-1.0435,0.738,0.686),
c("GO:1900034","regulation of cellular response to heat", 0.004,-6.369,-0.344, 2.670,-1.0520,0.764,0.686),
c("GO:0036258","multivesicular body assembly", 0.003, 5.016,-4.850, 2.609,-1.7891,0.805,0.687),
c("GO:0006461","protein complex assembly", 0.960, 4.930,-4.923, 5.091,-1.4911,0.764,0.688),
c("GO:0048548","regulation of pinocytosis", 0.002, 4.485,-0.075, 2.375,-1.3071,0.645,0.689),
c("GO:0007097","nuclear migration", 0.008, 3.319, 5.657, 3.035,-1.2128,0.889,0.690),
c("GO:1904036","negative regulation of epithelial cell apoptotic process", 0.007,-4.725,-0.705, 2.955,-1.8163,0.728,0.692),
c("GO:0070972","protein localization to endoplasmic reticulum", 0.187, 3.043, 6.058, 4.381,-1.0721,0.869,0.694),
c("GO:0007167","enzyme linked receptor protein signaling pathway", 0.279,-6.633,-0.510, 4.554,-1.3860,0.704,0.699));

one.data <- data.frame(revigo.data);
names(one.data) <- revigo.names;
one.data <- one.data [(one.data$plot_X != "null" & one.data$plot_Y != "null"), ];
one.data$plot_X <- as.numeric( as.character(one.data$plot_X) );
one.data$plot_Y <- as.numeric( as.character(one.data$plot_Y) );
one.data$plot_size <- as.numeric( as.character(one.data$plot_size) );
one.data$log10_p_value <- as.numeric( as.character(one.data$log10_p_value) );
one.data$frequency <- as.numeric( as.character(one.data$frequency) );
one.data$uniqueness <- as.numeric( as.character(one.data$uniqueness) );
one.data$dispensability <- as.numeric( as.character(one.data$dispensability) );
#head(one.data);


# --------------------------------------------------------------------------
# Names of the axes, sizes of the numbers and letters, names of the columns,
# etc. can be changed below

p1 <- ggplot( data = one.data );
p1 <- p1 + geom_point( aes( plot_X, plot_Y, colour = log10_p_value, size = plot_size), alpha = I(0.6) ) + scale_size_area();
p1 <- p1 + scale_colour_gradientn( colours = c("blue", "green", "yellow", "red"), limits = c( min(one.data$log10_p_value), 0) );
p1 <- p1 + geom_point( aes(plot_X, plot_Y, size = plot_size), shape = 21, fill = "transparent", colour = I (alpha ("black", 0.6) )) + scale_size_area();
p1 <- p1 + scale_size( range=c(5, 30)) + theme_bw(); # + scale_fill_gradientn(colours = heat_hcl(7), limits = c(-300, 0) );
ex <- one.data [ one.data$dispensability < 0.19, ]; 
p1 <- p1 + geom_text( data = ex, aes(plot_X, plot_Y, label = description), colour = I(alpha("black", 0.85)), size = 3 );
p1 <- p1 + labs (y = "semantic space x", x = "semantic space y");
p1 <- p1 + theme(legend.key = element_blank()) ;
one.x_range = max(one.data$plot_X) - min(one.data$plot_X);
one.y_range = max(one.data$plot_Y) - min(one.data$plot_Y);
p1 <- p1 + xlim(min(one.data$plot_X)-one.x_range/10,max(one.data$plot_X)+one.x_range/10);
p1 <- p1 + ylim(min(one.data$plot_Y)-one.y_range/10,max(one.data$plot_Y)+one.y_range/10);


cairo_ps(file = "/home/eneritz/ASCHROM1909/FIGURES0919/GO_figure/SACS7_GO.eps", onefile = FALSE, fallback_resolution = 3000 , width=6, height=8 )
p1
dev.off()


# --------------------------------------------------------------------------
# Output the plot to screen

p1;

####NOTE TO CHANGE THE PLOT LABELS NEEDS TO BE CHANGE THE DISPENSABILITY THRESHOLD


```
K4 K27 excluded from enriched are
```{r}
# --------------------------------------------------------------------------
# If you don't have the ggplot2 package installed, uncomment the following line:
# install.packages( "ggplot2" );
library( ggplot2 );
# --------------------------------------------------------------------------
# If you don't have the scales package installed, uncomment the following line:
# install.packages( "scales" );
library( scales );


# --------------------------------------------------------------------------
# Here is your data from REVIGO. Scroll down for plot configuration options.

revigo.names <- c("term_ID","description","frequency_%","plot_X","plot_Y","plot_size","log10_p_value","uniqueness","dispensability");
revigo.data <- rbind(c("GO:0007157","heterophilic cell-cell adhesion via plasma membrane cell adhesion molecules", 0.265,-0.333, 0.079, 1.672,-2.5210,0.976,0.000),
c("GO:0045637","regulation of myeloid cell differentiation", 1.010, 4.122,-6.371, 2.246,-2.2502,0.799,0.000),
c("GO:0042181","ketone biosynthetic process", 0.225,-3.782,-1.470, 1.602,-1.8182,0.883,0.033),
c("GO:0035088","establishment or maintenance of apical/basal cell polarity", 0.237,-0.759, 0.021, 1.623,-1.4972,0.955,0.033),
c("GO:0035194","posttranscriptional gene silencing by RNA", 0.369,-6.040,-0.373, 1.813,-1.4972,0.849,0.034),
c("GO:0051017","actin filament bundle assembly", 0.767, 0.129,-0.668, 2.127,-1.2030,0.934,0.037),
c("GO:0017015","regulation of transforming growth factor beta receptor signaling pathway", 0.583, 0.262, 7.521, 2.009,-2.0884,0.779,0.041),
c("GO:0043112","receptor metabolic process", 0.917,-1.355,-0.295, 2.204,-1.0193,0.944,0.057),
c("GO:0003094","glomerular filtration", 0.127, 1.074,-2.320, 1.362,-1.7317,0.925,0.093),
c("GO:0030252","growth hormone secretion", 0.092, 3.677, 2.593, 1.230,-1.9266,0.834,0.105),
c("GO:0097646","calcitonin family receptor signaling pathway", 0.138, 0.674, 4.576, 1.380,-1.8690,0.888,0.123),
c("GO:0035329","hippo signaling", 0.202, 0.558, 4.147, 1.556,-1.2889,0.897,0.128),
c("GO:0006817","phosphate ion transport", 0.150, 3.230, 1.121, 1.431,-1.6281,0.932,0.129),
c("GO:0030198","extracellular matrix organization", 1.829, 0.237,-0.448, 2.502,-0.4876,0.943,0.149),
c("GO:0016197","endosomal transport", 1.570, 3.098, 0.969, 2.436,-0.4876,0.932,0.163),
c("GO:0030162","regulation of proteolysis", 4.178,-7.634,-1.551, 2.860,-0.9858,0.823,0.165),
c("GO:0070076","histone lysine demethylation", 0.150,-4.076,-0.941, 1.431,-1.4346,0.869,0.168),
c("GO:0060055","angiogenesis involved in wound healing", 0.115, 2.908,-4.479, 1.322,-1.8182,0.867,0.178),
c("GO:0098700","neurotransmitter loading into synaptic vesicle", 0.040, 3.139, 1.227, 0.903,-1.9266,0.924,0.182),
c("GO:0050919","negative chemotaxis", 0.127,-0.090, 2.188, 1.362,-1.3475,0.939,0.184),
c("GO:2000427","positive regulation of apoptotic cell clearance", 0.046, 2.625, 0.307, 0.954,-1.8690,0.819,0.184),
c("GO:0018196","peptidyl-asparagine modification", 0.254,-4.987,-0.242, 1.653,-1.1228,0.905,0.184),
c("GO:0051092","positive regulation of NF-kappaB transcription factor activity", 0.785,-5.759,-1.929, 2.137,-0.7041,0.808,0.198),
c("GO:0043687","post-translational protein modification", 0.600,-4.674,-0.472, 2.021,-0.9086,0.912,0.207),
c("GO:0008285","negative regulation of cell proliferation", 3.716,-2.704, 1.079, 2.810,-0.3328,0.897,0.216),
c("GO:0051128","regulation of cellular component organization",13.335, 0.103,-0.497, 3.364,-0.6697,0.899,0.220),
c("GO:0010884","positive regulation of lipid storage", 0.127, 1.231,-1.052, 1.362,-1.4753,0.869,0.231),
c("GO:0006836","neurotransmitter transport", 1.114, 4.420, 1.396, 2.288,-1.5448,0.909,0.235),
c("GO:0097400","interleukin-17-mediated signaling pathway", 0.006,-0.127, 4.571, 0.301,-1.8182,0.886,0.238),
c("GO:0045446","endothelial cell differentiation", 0.565, 2.818,-4.606, 1.996,-1.6281,0.897,0.241),
c("GO:0046777","protein autophosphorylation", 1.333,-5.290,-0.673, 2.365,-0.5854,0.865,0.243),
c("GO:0006811","ion transport", 9.158, 4.596, 1.153, 3.201,-0.4071,0.920,0.266),
c("GO:0051146","striated muscle cell differentiation", 1.443, 2.813,-4.651, 2.400,-1.4753,0.905,0.266),
c("GO:0099504","synaptic vesicle cycle", 0.716, 2.814, 0.912, 2.097,-1.6281,0.931,0.295),
c("GO:0038042","dimeric G-protein coupled receptor signaling pathway", 0.006,-0.089, 2.116, 0.301,-1.8182,0.909,0.296),
c("GO:0097647","amylin receptor signaling pathway", 0.006,-0.188, 2.466, 0.301,-1.9266,0.909,0.296),
c("GO:0007229","integrin-mediated signaling pathway", 0.548, 0.181, 6.153, 1.982,-1.0193,0.871,0.299),
c("GO:0001558","regulation of cell growth", 2.250, 0.746,-0.490, 2.592,-0.5345,0.900,0.304),
c("GO:0002673","regulation of acute inflammatory response", 0.421,-0.528, 3.938, 1.869,-0.7259,0.883,0.307),
c("GO:0007249","I-kappaB kinase/NF-kappaB signaling", 1.535, 0.712, 5.523, 2.427,-1.0264,0.878,0.309),
c("GO:0010827","regulation of glucose transport", 0.600, 3.694, 1.191, 2.021,-1.5448,0.868,0.315),
c("GO:0032446","protein modification by small protein conjugation", 5.234,-5.350,-0.769, 2.958,-0.3043,0.888,0.316),
c("GO:0003073","regulation of systemic arterial blood pressure", 0.496, 1.703,-2.548, 1.940,-1.6599,0.874,0.318),
c("GO:0019933","cAMP-mediated signaling", 0.294, 0.617, 4.495, 1.716,-0.9154,0.894,0.321),
c("GO:0006895","Golgi to endosome transport", 0.115, 3.788, 1.165, 1.322,-1.4545,0.915,0.321),
c("GO:0031326","regulation of cellular biosynthetic process",24.939,-7.629,-2.118, 3.636,-1.3028,0.795,0.330),
c("GO:0030814","regulation of cAMP metabolic process", 0.791,-7.638,-1.767, 2.140,-1.6281,0.766,0.340),
c("GO:0010866","regulation of triglyceride biosynthetic process", 0.092,-5.216,-2.116, 1.230,-1.6281,0.856,0.344),
c("GO:0007420","brain development", 3.895, 3.285,-5.429, 2.830,-0.7324,0.883,0.347),
c("GO:2000257","regulation of protein activation cascade", 0.202,-6.298, 2.788, 1.556,-0.7703,0.827,0.350),
c("GO:1903818","positive regulation of voltage-gated potassium channel activity", 0.029, 1.265, 0.925, 0.778,-1.7728,0.838,0.350),
c("GO:0000902","cell morphogenesis", 5.297, 3.398,-4.559, 2.963,-1.0124,0.869,0.351),
c("GO:0035249","synaptic transmission, glutamatergic", 0.433, 1.196, 4.590, 1.881,-1.4346,0.907,0.352),
c("GO:0038061","NIK/NF-kappaB signaling", 0.716, 0.600, 4.792, 2.097,-0.8895,0.886,0.352),
c("GO:0002029","desensitization of G-protein coupled receptor protein signaling pathway", 0.092,-1.358, 6.476, 1.230,-1.8182,0.843,0.357),
c("GO:1901796","regulation of signal transduction by p53 class mediator", 0.964, 0.084, 7.089, 2.225,-0.6752,0.847,0.364),
c("GO:0034097","response to cytokine", 4.714,-0.082, 3.117, 2.913,-0.6752,0.922,0.367),
c("GO:0007223","Wnt signaling pathway, calcium modulating pathway", 0.214, 0.655, 6.014, 1.580,-1.2142,0.860,0.378),
c("GO:0001819","positive regulation of cytokine production", 2.256, 0.872,-4.987, 2.593,-1.2667,0.808,0.380),
c("GO:0016482","cytosolic transport", 0.842, 2.856, 0.927, 2.167,-0.7357,0.930,0.387),
c("GO:0007167","enzyme linked receptor protein signaling pathway", 5.713, 0.482, 6.629, 2.996,-0.7292,0.838,0.387),
c("GO:0060973","cell migration involved in heart development", 0.081, 4.493,-4.914, 1.176,-1.8182,0.861,0.399),
c("GO:0051246","regulation of protein metabolic process",14.651,-7.721,-1.873, 3.405,-1.0124,0.819,0.400),
c("GO:0048333","mesodermal cell differentiation", 0.185, 3.397,-5.567, 1.519,-1.7728,0.872,0.417),
c("GO:0008277","regulation of G-protein coupled receptor protein signaling pathway", 0.733, 0.225, 7.412, 2.107,-0.9208,0.841,0.421),
c("GO:0007175","negative regulation of epidermal growth factor-activated receptor activity", 0.063,-5.328, 3.577, 1.079,-1.6281,0.714,0.432),
c("GO:0009967","positive regulation of signal transduction", 8.309,-0.789, 5.219, 3.159,-0.5256,0.762,0.434),
c("GO:0002920","regulation of humoral immune response", 0.294,-0.191, 3.836, 1.716,-0.7526,0.881,0.437),
c("GO:0006820","anion transport", 3.422, 4.450, 1.243, 2.774,-1.0336,0.916,0.440),
c("GO:0018279","protein N-linked glycosylation via asparagine", 0.248,-6.905,-1.267, 1.643,-1.1320,0.835,0.448),
c("GO:0030521","androgen receptor signaling pathway", 0.381, 0.785, 5.959, 1.826,-1.1710,0.865,0.457),
c("GO:0010718","positive regulation of epithelial to mesenchymal transition", 0.225, 2.584,-6.595, 1.602,-1.1710,0.791,0.461),
c("GO:0007189","adenylate cyclase-activating G-protein coupled receptor signaling pathway", 0.410, 0.568, 4.706, 1.857,-0.8996,0.874,0.465),
c("GO:0003183","mitral valve morphogenesis", 0.058, 2.831,-4.677, 1.041,-1.7728,0.897,0.472),
c("GO:0010468","regulation of gene expression",25.084,-7.579,-2.064, 3.638,-0.5882,0.818,0.477),
c("GO:0070613","regulation of protein processing", 0.502,-7.015,-1.404, 1.944,-0.7041,0.844,0.482),
c("GO:0003158","endothelium development", 0.687, 2.777,-4.631, 2.079,-1.7317,0.922,0.485),
c("GO:0045995","regulation of embryonic development", 0.704, 3.397,-6.086, 2.090,-1.3028,0.842,0.487),
c("GO:0044267","cellular protein metabolic process",28.505,-6.650,-1.219, 3.694,-0.7008,0.876,0.488),
c("GO:0051050","positive regulation of transport", 5.349, 2.632,-0.196, 2.968,-1.1320,0.814,0.501),
c("GO:0006954","inflammatory response", 3.785,-0.169, 0.640, 2.818,-0.4540,0.940,0.504),
c("GO:0051240","positive regulation of multicellular organismal process", 8.327, 1.138,-5.587, 3.160,-0.5327,0.799,0.507),
c("GO:0000122","negative regulation of transcription from RNA polymerase II promoter", 4.380,-7.215,-0.960, 2.881,-0.6019,0.788,0.517),
c("GO:0006898","receptor-mediated endocytosis", 2.014, 4.265, 1.211, 2.544,-1.3883,0.890,0.524),
c("GO:0002697","regulation of immune effector process", 1.922, 0.275,-0.932, 2.524,-0.7492,0.919,0.533),
c("GO:0006338","chromatin remodeling", 0.860,-0.205,-0.659, 2.176,-0.7424,0.963,0.542),
c("GO:0098742","cell-cell adhesion via plasma-membrane adhesion molecules", 1.344,-0.335, 0.071, 2.369,-1.6104,0.978,0.542),
c("GO:0006897","endocytosis", 4.212, 4.348, 1.237, 2.864,-1.1314,0.897,0.545),
c("GO:0042327","positive regulation of phosphorylation", 5.384,-5.799,-2.117, 2.970,-0.5221,0.755,0.551),
c("GO:0031401","positive regulation of protein modification process", 6.636,-6.018,-1.741, 3.061,-0.6120,0.744,0.557),
c("GO:0016310","phosphorylation",13.018,-4.077,-0.376, 3.354,-0.3137,0.888,0.562),
c("GO:0001570","vasculogenesis", 0.415, 3.653,-5.875, 1.863,-1.2625,0.873,0.564),
c("GO:0006464","cellular protein modification process",22.683,-6.025,-0.939, 3.595,-0.6171,0.862,0.566),
c("GO:0035051","cardiocyte differentiation", 0.710, 3.620,-5.814, 2.093,-1.6281,0.867,0.570),
c("GO:0045893","positive regulation of transcription, DNA-templated", 7.905,-6.272,-2.570, 3.137,-0.5273,0.743,0.571),
c("GO:0038083","peptidyl-tyrosine autophosphorylation", 0.300,-5.121,-0.507, 1.724,-1.1228,0.866,0.572),
c("GO:0045597","positive regulation of cell differentiation", 4.893, 1.575,-5.689, 2.929,-0.5474,0.792,0.585),
c("GO:0015698","inorganic anion transport", 0.992, 3.194, 1.117, 2.238,-1.1320,0.923,0.587),
c("GO:0007178","transmembrane receptor protein serine/threonine kinase signaling pathway", 1.841, 0.213, 6.085, 2.505,-0.6589,0.832,0.591),
c("GO:1905330","regulation of morphogenesis of an epithelium", 1.039, 3.511,-5.505, 2.258,-1.7728,0.855,0.592),
c("GO:0045823","positive regulation of heart contraction", 0.190, 1.141,-5.135, 1.531,-1.4753,0.817,0.599),
c("GO:0045055","regulated exocytosis", 4.299, 4.561, 1.415, 2.873,-0.6485,0.855,0.610),
c("GO:0006511","ubiquitin-dependent protein catabolic process", 3.306,-4.836,-0.755, 2.759,-0.3527,0.894,0.619),
c("GO:0022008","neurogenesis", 8.442, 3.391,-5.632, 3.166,-1.0798,0.857,0.628),
c("GO:0030855","epithelial cell differentiation", 4.207, 3.002,-4.703, 2.863,-0.8168,0.875,0.636),
c("GO:0010575","positive regulation of vascular endothelial growth factor production", 0.156, 0.579,-4.560, 1.447,-1.3802,0.843,0.638),
c("GO:0006892","post-Golgi vesicle-mediated transport", 0.548, 3.939, 1.162, 1.982,-1.0193,0.911,0.640),
c("GO:0035567","non-canonical Wnt signaling pathway", 0.860, 0.834, 7.019, 2.176,-0.7072,0.843,0.643),
c("GO:0045595","regulation of cell differentiation", 8.707, 3.306,-5.453, 3.179,-0.7259,0.838,0.652),
c("GO:0042448","progesterone metabolic process", 0.098,-2.722,-1.406, 1.255,-1.6943,0.860,0.653),
c("GO:0010604","positive regulation of macromolecule metabolic process",16.613,-5.547,-2.556, 3.459,-0.4227,0.781,0.657),
c("GO:0031623","receptor internalization", 0.496, 1.869, 0.772, 1.940,-1.0798,0.818,0.659),
c("GO:0006911","phagocytosis, engulfment", 0.508, 3.522, 0.891, 1.949,-0.7852,0.858,0.660),
c("GO:0007417","central nervous system development", 5.182, 3.230,-5.365, 2.954,-0.5068,0.887,0.661),
c("GO:0050776","regulation of immune response", 5.799,-0.244, 4.716, 3.003,-0.4554,0.847,0.670),
c("GO:0042127","regulation of cell proliferation", 8.898,-1.015, 0.027, 3.188,-0.4429,0.914,0.676),
c("GO:0046890","regulation of lipid biosynthetic process", 0.773,-5.680,-2.006, 2.130,-1.3172,0.836,0.678),
c("GO:0017148","negative regulation of translation", 0.906,-7.196,-0.773, 2.199,-0.8514,0.776,0.678),
c("GO:0045652","regulation of megakaryocyte differentiation", 0.092, 4.173,-6.307, 1.230,-2.4861,0.836,0.679),
c("GO:0019229","regulation of vasoconstriction", 0.335, 2.542,-3.877, 1.771,-1.3028,0.843,0.682),
c("GO:0016339","calcium-dependent cell-cell adhesion via plasma membrane cell adhesion molecules", 0.162,-0.742,-0.142, 1.462,-1.1609,0.976,0.688),
c("GO:0010628","positive regulation of gene expression", 9.833,-5.687,-2.632, 3.232,-0.4204,0.784,0.692),
c("GO:0006357","regulation of transcription from RNA polymerase II promoter",10.900,-7.563,-2.119, 3.276,-0.3326,0.783,0.696),
c("GO:0006355","regulation of transcription, DNA-templated",20.929,-7.531,-2.032, 3.560,-0.5420,0.764,0.698),
c("GO:0032481","positive regulation of type I interferon production", 0.415, 0.621,-4.415, 1.863,-0.9923,0.832,0.705),
c("GO:0002576","platelet degranulation", 0.739, 3.899, 1.348, 2.111,-0.7164,0.877,0.707),
c("GO:0006487","protein N-linked glycosylation", 0.473,-6.030,-1.331, 1.919,-0.9263,0.839,0.708),
c("GO:2000272","negative regulation of receptor activity", 0.190,-1.166, 6.923, 1.531,-1.2625,0.851,0.719),
c("GO:0045745","positive regulation of G-protein coupled receptor protein signaling pathway", 0.144,-1.004, 4.559, 1.415,-1.5204,0.820,0.721),
c("GO:2000112","regulation of cellular macromolecule biosynthetic process",23.220,-7.581,-2.033, 3.605,-0.5346,0.783,0.723),
c("GO:0048738","cardiac muscle tissue development", 1.079, 3.349,-5.353, 2.274,-1.2142,0.861,0.726),
c("GO:0055117","regulation of cardiac muscle contraction", 0.410, 2.265,-4.305, 1.857,-1.2257,0.847,0.729),
c("GO:0007268","chemical synaptic transmission", 3.474, 1.219, 5.809, 2.780,-0.4071,0.889,0.737),
c("GO:0009152","purine ribonucleotide biosynthetic process", 1.443,-5.774,-1.444, 2.400,-0.9154,0.792,0.741),
c("GO:0030072","peptide hormone secretion", 1.373, 4.461, 4.678, 2.378,-1.3635,0.793,0.745),
c("GO:0035809","regulation of urine volume", 0.115, 1.687,-2.661, 1.322,-1.7317,0.894,0.746),
c("GO:0007169","transmembrane receptor protein tyrosine kinase signaling pathway", 3.976, 0.378, 6.575, 2.839,-0.3058,0.818,0.748),
c("GO:0097205","renal filtration", 0.133, 1.089,-2.357, 1.380,-1.6943,0.924,0.754),
c("GO:0007015","actin filament organization", 2.031, 0.314,-0.562, 2.548,-0.7292,0.930,0.754),
c("GO:0006351","transcription, DNA-templated",21.881,-6.750,-2.479, 3.579,-0.3389,0.834,0.766),
c("GO:0006909","phagocytosis", 1.858, 3.672, 1.160, 2.509,-0.6836,0.872,0.769),
c("GO:0018108","peptidyl-tyrosine phosphorylation", 2.135,-6.358,-0.628, 2.569,-0.8945,0.838,0.779),
c("GO:0008015","blood circulation", 2.827, 2.269,-4.028, 2.691,-1.0798,0.896,0.779),
c("GO:1903506","regulation of nucleic acid-templated transcription",21.073,-7.356,-2.013, 3.563,-0.5579,0.769,0.785),
c("GO:0061572","actin filament bundle organization", 0.785, 0.156,-0.565, 2.137,-1.2257,0.933,0.786),
c("GO:1902259","regulation of delayed rectifier potassium channel activity", 0.092, 2.852, 2.057, 1.230,-1.5708,0.858,0.794),
c("GO:0007176","regulation of epidermal growth factor-activated receptor activity", 0.150,-5.215, 3.720, 1.431,-1.3802,0.710,0.794),
c("GO:0055007","cardiac muscle cell differentiation", 0.560, 3.790,-6.126, 1.991,-1.5708,0.850,0.798),
c("GO:2000425","regulation of apoptotic cell clearance", 0.058, 3.791, 0.967, 1.041,-1.8182,0.847,0.818),
c("GO:0051091","positive regulation of sequence-specific DNA binding transcription factor activity", 1.362,-5.877,-1.959, 2.375,-0.5101,0.797,0.820),
c("GO:0090092","regulation of transmembrane receptor protein serine/threonine kinase signaling pathway", 1.223, 0.255, 7.685, 2.328,-1.3172,0.803,0.821),
c("GO:0050766","positive regulation of phagocytosis", 0.260, 2.753, 0.112, 1.663,-1.1814,0.795,0.828),
c("GO:1903844","regulation of cellular response to transforming growth factor beta stimulus", 0.594,-0.459, 5.204, 2.017,-1.4972,0.841,0.840),
c("GO:0001932","regulation of protein phosphorylation", 7.519,-7.955,-1.638, 3.115,-0.4418,0.747,0.841),
c("GO:0010717","regulation of epithelial to mesenchymal transition", 0.427, 3.888,-6.643, 1.875,-0.9549,0.818,0.842),
c("GO:0090100","positive regulation of transmembrane receptor protein serine/threonine kinase signaling pathway", 0.577,-0.957, 6.460, 2.004,-0.8796,0.773,0.845),
c("GO:0010828","positive regulation of glucose transport", 0.265, 2.347, 0.214, 1.672,-1.3635,0.841,0.846),
c("GO:0090101","negative regulation of transmembrane receptor protein serine/threonine kinase signaling pathway", 0.612,-1.203, 6.674, 2.029,-0.9048,0.796,0.849),
c("GO:0061099","negative regulation of protein tyrosine kinase activity", 0.138,-6.941,-0.147, 1.398,-1.4346,0.787,0.858),
c("GO:0030518","intracellular steroid hormone receptor signaling pathway", 0.739, 0.911, 6.451, 2.111,-0.9669,0.857,0.860),
c("GO:0045892","negative regulation of transcription, DNA-templated", 6.411,-7.534,-0.989, 3.046,-0.3921,0.774,0.861),
c("GO:0042059","negative regulation of epidermal growth factor receptor signaling pathway", 0.231,-1.171, 7.134, 1.613,-1.1710,0.817,0.862),
c("GO:0010460","positive regulation of heart rate", 0.127, 1.433,-4.723, 1.362,-1.5985,0.815,0.865),
c("GO:0030511","positive regulation of transforming growth factor beta receptor signaling pathway", 0.144,-0.854, 6.465, 1.415,-1.4753,0.769,0.872),
c("GO:1903846","positive regulation of cellular response to transforming growth factor beta stimulus", 0.144,-1.326, 3.577, 1.415,-1.4753,0.817,0.872),
c("GO:0006942","regulation of striated muscle contraction", 0.496, 1.941,-4.000, 1.940,-1.3028,0.856,0.873),
c("GO:0008016","regulation of heart contraction", 1.327, 2.598,-4.701, 2.364,-0.8209,0.834,0.876),
c("GO:0045944","positive regulation of transcription from RNA polymerase II promoter", 6.111,-6.061,-2.461, 3.025,-0.3702,0.747,0.876),
c("GO:0007188","adenylate cyclase-modulating G-protein coupled receptor signaling pathway", 0.877, 0.594, 4.866, 2.185,-0.6865,0.866,0.884),
c("GO:1900544","positive regulation of purine nucleotide metabolic process", 0.779,-5.691,-1.966, 2.134,-1.5985,0.741,0.892));

one.data <- data.frame(revigo.data);
names(one.data) <- revigo.names;
one.data <- one.data [(one.data$plot_X != "null" & one.data$plot_Y != "null"), ];
one.data$plot_X <- as.numeric( as.character(one.data$plot_X) );
one.data$plot_Y <- as.numeric( as.character(one.data$plot_Y) );
one.data$plot_size <- as.numeric( as.character(one.data$plot_size) );
one.data$log10_p_value <- as.numeric( as.character(one.data$log10_p_value) );
one.data$frequency <- as.numeric( as.character(one.data$frequency) );
one.data$uniqueness <- as.numeric( as.character(one.data$uniqueness) );
one.data$dispensability <- as.numeric( as.character(one.data$dispensability) );
#head(one.data);


# --------------------------------------------------------------------------
# Names of the axes, sizes of the numbers and letters, names of the columns,
# etc. can be changed below

p1 <- ggplot( data = one.data );
p1 <- p1 + geom_point( aes( plot_X, plot_Y, colour = log10_p_value, size = plot_size), alpha = I(0.6) ) + scale_size_area();
p1 <- p1 + scale_colour_gradientn( colours = c("blue", "green", "yellow", "red"), limits = c( min(one.data$log10_p_value), 0) );
p1 <- p1 + geom_point( aes(plot_X, plot_Y, size = plot_size), shape = 21, fill = "transparent", colour = I (alpha ("black", 0.6) )) + scale_size_area();
p1 <- p1 + scale_size( range=c(5, 30)) + theme_bw(); # + scale_fill_gradientn(colours = heat_hcl(7), limits = c(-300, 0) );
ex <- one.data [ one.data$dispensability < 0.15, ]; 
p1 <- p1 + geom_text( data = ex, aes(plot_X, plot_Y, label = description), colour = I(alpha("black", 0.85)), size = 3 );
p1 <- p1 + labs (y = "semantic space x", x = "semantic space y");
p1 <- p1 + theme(legend.key = element_blank()) ;
one.x_range = max(one.data$plot_X) - min(one.data$plot_X);
one.y_range = max(one.data$plot_Y) - min(one.data$plot_Y);
p1 <- p1 + xlim(min(one.data$plot_X)-one.x_range/10,max(one.data$plot_X)+one.x_range/10);
p1 <- p1 + ylim(min(one.data$plot_Y)-one.y_range/10,max(one.data$plot_Y)+one.y_range/10);



# --------------------------------------------------------------------------
# Output the plot to screen

p1;
 exc
##280419
##3new K27K4
#filter by pval the enricr results 
#cut -f 2,3 /home/eneritz/ASCHROM0818/GO_terms_supfig4/K4_K27_GO.filter > lala


# --------------------------------------------------------------------------
# Here is your data from REVIGO. Scroll down for plot configuration options.

revigo.names <- c("term_ID","description","frequency_%","plot_X","plot_Y","plot_size","log10_p_value","uniqueness","dispensability");
revigo.data <- rbind(c("GO:0007157","heterophilic cell-cell adhesion via plasma membrane cell adhesion molecules", 0.265, 1.176, 0.022, 1.672,-2.5210,0.958,0.000),
c("GO:0045637","regulation of myeloid cell differentiation", 1.010, 4.448, 3.711, 2.246,-2.2502,0.782,0.000),
c("GO:0042181","ketone biosynthetic process", 0.225, 2.235,-6.563, 1.602,-1.8182,0.842,0.033),
c("GO:0035088","establishment or maintenance of apical/basal cell polarity", 0.237,-2.868,-1.068, 1.623,-1.4972,0.946,0.033),
c("GO:0035194","posttranscriptional gene silencing by RNA", 0.369,-3.412,-5.918, 1.813,-1.4972,0.889,0.034),
c("GO:0017015","regulation of transforming growth factor beta receptor signaling pathway", 0.583,-4.264,-3.790, 2.009,-2.0884,0.782,0.041),
c("GO:0003094","glomerular filtration", 0.127, 0.558, 5.696, 1.362,-1.7317,0.866,0.093),
c("GO:0030252","growth hormone secretion", 0.092,-4.949, 2.273, 1.230,-1.9266,0.786,0.105),
c("GO:0097646","calcitonin family receptor signaling pathway", 0.138, 2.213,-3.152, 1.380,-1.8690,0.875,0.123),
c("GO:0006817","phosphate ion transport", 0.150,-6.954, 1.124, 1.431,-1.6281,0.909,0.129),
c("GO:0070076","histone lysine demethylation", 0.150,-1.449,-6.382, 1.431,-1.4346,0.883,0.168),
c("GO:0060055","angiogenesis involved in wound healing", 0.115, 5.980, 4.230, 1.322,-1.8182,0.823,0.178),
c("GO:0098700","neurotransmitter loading into synaptic vesicle", 0.040,-4.198, 6.088, 0.903,-1.9266,0.891,0.182),
c("GO:0050919","negative chemotaxis", 0.127, 6.342,-2.893, 1.362,-1.3475,0.925,0.184),
c("GO:2000427","positive regulation of apoptotic cell clearance", 0.046,-4.890, 3.863, 0.954,-1.8690,0.804,0.184),
c("GO:0031326","regulation of cellular biosynthetic process",24.939, 0.809,-6.656, 3.636,-1.3028,0.861,0.189),
c("GO:0010884","positive regulation of lipid storage", 0.127,-2.965, 2.818, 1.362,-1.4753,0.829,0.231),
c("GO:0006836","neurotransmitter transport", 1.114,-5.463, 4.418, 2.288,-1.5448,0.878,0.235),
c("GO:0097400","interleukin-17-mediated signaling pathway", 0.006,-5.417,-3.525, 0.301,-1.8182,0.863,0.238),
c("GO:0045446","endothelial cell differentiation", 0.565, 6.692, 0.808, 1.996,-1.6281,0.858,0.241),
c("GO:0051146","striated muscle cell differentiation", 1.443, 6.085, 1.327, 2.400,-1.4753,0.867,0.266),
c("GO:0010575","positive regulation of vascular endothelial growth factor production", 0.156, 2.131, 7.250, 1.447,-1.3802,0.813,0.288),
c("GO:0099504","synaptic vesicle cycle", 0.716,-3.422, 6.119, 2.097,-1.6281,0.903,0.295),
c("GO:0038042","dimeric G-protein coupled receptor signaling pathway", 0.006, 4.204,-3.755, 0.301,-1.8182,0.892,0.296),
c("GO:0097647","amylin receptor signaling pathway", 0.006, 3.366,-3.300, 0.301,-1.9266,0.892,0.296),
c("GO:0010827","regulation of glucose transport", 0.600,-5.336, 3.335, 2.021,-1.5448,0.829,0.315),
c("GO:0003073","regulation of systemic arterial blood pressure", 0.496,-0.160, 6.109, 1.940,-1.6599,0.815,0.318),
c("GO:0006895","Golgi to endosome transport", 0.115,-3.994, 5.891, 1.322,-1.4545,0.898,0.321),
c("GO:0030814","regulation of cAMP metabolic process", 0.791,-0.186,-6.441, 2.140,-1.6281,0.745,0.340),
c("GO:0010866","regulation of triglyceride biosynthetic process", 0.092, 2.923,-6.389, 1.230,-1.6281,0.839,0.344),
c("GO:1903818","positive regulation of voltage-gated potassium channel activity", 0.029,-5.890, 2.006, 0.778,-1.7728,0.802,0.350),
c("GO:0035249","synaptic transmission, glutamatergic", 0.433,-6.952,-1.778, 1.881,-1.4346,0.897,0.352),
c("GO:0002029","desensitization of G-protein coupled receptor protein signaling pathway", 0.092,-0.308,-3.440, 1.230,-1.8182,0.837,0.357),
c("GO:0006942","regulation of striated muscle contraction", 0.496, 1.688, 6.329, 1.940,-1.3028,0.811,0.359),
c("GO:0060973","cell migration involved in heart development", 0.081, 5.056, 4.476, 1.176,-1.8182,0.802,0.399),
c("GO:0048333","mesodermal cell differentiation", 0.185, 6.205, 2.336, 1.519,-1.7728,0.808,0.417),
c("GO:0007175","negative regulation of epidermal growth factor-activated receptor activity", 0.063,-3.163,-4.618, 1.079,-1.6281,0.755,0.432),
c("GO:0003183","mitral valve morphogenesis", 0.058, 5.391, 4.954, 1.041,-1.7728,0.844,0.472),
c("GO:0003158","endothelium development", 0.687, 6.869, 0.410, 2.079,-1.7317,0.885,0.485),
c("GO:0045995","regulation of embryonic development", 0.704, 4.610, 3.117, 2.090,-1.3028,0.800,0.487));

one.data <- data.frame(revigo.data);
names(one.data) <- revigo.names;
one.data <- one.data [(one.data$plot_X != "null" & one.data$plot_Y != "null"), ];
one.data$plot_X <- as.numeric( as.character(one.data$plot_X) );
one.data$plot_Y <- as.numeric( as.character(one.data$plot_Y) );
one.data$plot_size <- as.numeric( as.character(one.data$plot_size) );
one.data$log10_p_value <- as.numeric( as.character(one.data$log10_p_value) );
one.data$frequency <- as.numeric( as.character(one.data$frequency) );
one.data$uniqueness <- as.numeric( as.character(one.data$uniqueness) );
one.data$dispensability <- as.numeric( as.character(one.data$dispensability) );
#head(one.data);


# --------------------------------------------------------------------------
# Names of the axes, sizes of the numbers and letters, names of the columns,
# etc. can be changed below

p1 <- ggplot( data = one.data );
p1 <- p1 + geom_point( aes( plot_X, plot_Y, colour = log10_p_value, size = plot_size), alpha = I(0.6) ) + scale_size_area();
p1 <- p1 + scale_colour_gradientn( colours = c("blue", "green", "yellow", "red"), limits = c( min(one.data$log10_p_value), 0) );
p1 <- p1 + geom_point( aes(plot_X, plot_Y, size = plot_size), shape = 21, fill = "transparent", colour = I (alpha ("black", 0.6) )) + scale_size_area();
p1 <- p1 + scale_size( range=c(5, 30)) + theme_bw(); # + scale_fill_gradientn(colours = heat_hcl(7), limits = c(-300, 0) );
ex <- one.data [ one.data$dispensability < 0.15, ]; 
p1 <- p1 + geom_text( data = ex, aes(plot_X, plot_Y, label = description), colour = I(alpha("black", 0.85)), size = 3 );
p1 <- p1 + labs (y = "semantic space x", x = "semantic space y");
p1 <- p1 + theme(legend.key = element_blank()) ;
one.x_range = max(one.data$plot_X) - min(one.data$plot_X);
one.y_range = max(one.data$plot_Y) - min(one.data$plot_Y);
p1 <- p1 + xlim(min(one.data$plot_X)-one.x_range/10,max(one.data$plot_X)+one.x_range/10);
p1 <- p1 + ylim(min(one.data$plot_Y)-one.y_range/10,max(one.data$plot_Y)+one.y_range/10);



# --------------------------------------------------------------------------
# Output the plot to screen

p1;


```


K20 K79 exc
```{r}
# --------------------------------------------------------------------------
# If you don't have the ggplot2 package installed, uncomment the following line:
# install.packages( "ggplot2" );
library( ggplot2 );
# --------------------------------------------------------------------------
# If you don't have the scales package installed, uncomment the following line:
# install.packages( "scales" );
library( scales );


# --------------------------------------------------------------------------
# Here is your data from REVIGO. Scroll down for plot configuration options.

revigo.names <- c("term_ID","description","frequency_%","plot_X","plot_Y","plot_size","log10_p_value","uniqueness","dispensability");
revigo.data <- rbind(c("GO:0008150","biological_process",100.000, 4.098, 3.219, 4.239,-2.6383,1.000,0.000),
c("GO:0008380","RNA splicing", 2.418,-5.528,-1.090, 2.623,-1.9586,0.518,0.000),
c("GO:0000077","DNA damage checkpoint", 0.894,-1.938, 5.614, 2.193,-1.6778,0.393,0.002),
c("GO:0072717","cellular response to actinomycin D", 0.017, 4.027,-4.635, 0.602,-1.3010,0.764,0.081),
c("GO:0016071","mRNA metabolic process", 3.935,-6.058,-2.953, 2.834,-1.2924,0.697,0.276),
c("GO:0006396","RNA processing", 5.280,-4.897,-4.135, 2.962,-0.4437,0.692,0.310),
c("GO:0072716","response to actinomycin D", 0.023, 3.578,-5.390, 0.699,-1.0655,0.777,0.687));

one.data <- data.frame(revigo.data);
names(one.data) <- revigo.names;
one.data <- one.data [(one.data$plot_X != "null" & one.data$plot_Y != "null"), ];
one.data$plot_X <- as.numeric( as.character(one.data$plot_X) );
one.data$plot_Y <- as.numeric( as.character(one.data$plot_Y) );
one.data$plot_size <- as.numeric( as.character(one.data$plot_size) );
one.data$log10_p_value <- as.numeric( as.character(one.data$log10_p_value) );
one.data$frequency <- as.numeric( as.character(one.data$frequency) );
one.data$uniqueness <- as.numeric( as.character(one.data$uniqueness) );
one.data$dispensability <- as.numeric( as.character(one.data$dispensability) );
#head(one.data);


# --------------------------------------------------------------------------
# Names of the axes, sizes of the numbers and letters, names of the columns,
# etc. can be changed below

p1 <- ggplot( data = one.data );
p1 <- p1 + geom_point( aes( plot_X, plot_Y, colour = log10_p_value, size = plot_size), alpha = I(0.6) ) + scale_size_area();
p1 <- p1 + scale_colour_gradientn( colours = c("blue", "green", "yellow", "red"), limits = c( min(one.data$log10_p_value), 0) );
p1 <- p1 + geom_point( aes(plot_X, plot_Y, size = plot_size), shape = 21, fill = "transparent", colour = I (alpha ("black", 0.6) )) + scale_size_area();
p1 <- p1 + scale_size( range=c(5, 30)) + theme_bw(); # + scale_fill_gradientn(colours = heat_hcl(7), limits = c(-300, 0) );
ex <- one.data [ one.data$dispensability < 0.15, ]; 
p1 <- p1 + geom_text( data = ex, aes(plot_X, plot_Y, label = description), colour = I(alpha("black", 0.85)), size = 3 );
p1 <- p1 + labs (y = "semantic space x", x = "semantic space y");
p1 <- p1 + theme(legend.key = element_blank()) ;
one.x_range = max(one.data$plot_X) - min(one.data$plot_X);
one.y_range = max(one.data$plot_Y) - min(one.data$plot_Y);
p1 <- p1 + xlim(min(one.data$plot_X)-one.x_range/10,max(one.data$plot_X)+one.x_range/10);
p1 <- p1 + ylim(min(one.data$plot_Y)-one.y_range/10,max(one.data$plot_Y)+one.y_range/10);



# --------------------------------------------------------------------------
# Output the plot to screen

p1;

#####enrichr


# --------------------------------------------------------------------------
# If you don't have the ggplot2 package installed, uncomment the following line:
# install.packages( "ggplot2" );
library( ggplot2 );
# --------------------------------------------------------------------------
# If you don't have the scales package installed, uncomment the following line:
# install.packages( "scales" );
library( scales );


# --------------------------------------------------------------------------
# Here is your data from REVIGO. Scroll down for plot configuration options.

revigo.names <- c("term_ID","description","frequency_%","plot_X","plot_Y","plot_size","log10_p_value","uniqueness","dispensability");
revigo.data <- rbind(c("GO:0002478","antigen processing and presentation of exogenous peptide antigen", 0.975, 1.291, 1.367, 2.230,-2.4153,0.963,0.000),
c("GO:0008380","RNA splicing", 2.418,-6.785, 2.222, 2.623,-6.3116,0.821,0.000),
c("GO:0031648","protein destabilization", 0.219, 3.774, 1.270, 1.591,-1.7373,0.956,0.000),
c("GO:0006999","nuclear pore organization", 0.087, 4.395, 6.437, 1.204,-3.8500,0.881,0.002),
c("GO:0071479","cellular response to ionizing radiation", 0.375, 0.421, 0.191, 1.820,-3.6112,0.935,0.002),
c("GO:2000576","positive regulation of microtubule motor activity", 0.006,-1.114,-4.473, 0.301,-1.3504,0.970,0.017),
c("GO:0051196","regulation of coenzyme metabolic process", 0.306,-5.139,-2.938, 1.732,-1.5553,0.911,0.044),
c("GO:0006477","protein sulfation", 0.040,-0.442, 5.716, 0.903,-1.3504,0.907,0.064),
c("GO:0035646","endosome to melanosome transport", 0.058, 3.947,-5.181, 1.041,-2.7391,0.868,0.083),
c("GO:0035458","cellular response to interferon-beta", 0.087,-1.167,-3.600, 1.204,-2.5765,0.957,0.086),
c("GO:0010763","positive regulation of fibroblast migration", 0.069,-0.361,-5.646, 1.114,-1.3504,0.906,0.089),
c("GO:0030262","apoptotic nuclear changes", 0.167, 4.092, 3.675, 1.477,-1.9772,0.900,0.099),
c("GO:0031646","positive regulation of neurological system process", 0.127, 3.300, 0.147, 1.362,-1.2938,0.931,0.116),
c("GO:0010467","gene expression",30.542,-5.330, 0.434, 3.724,-4.7932,0.909,0.120),
c("GO:0043467","regulation of generation of precursor metabolites and energy", 0.502,-4.899,-2.185, 1.944,-1.4263,0.913,0.129),
c("GO:0010569","regulation of double-strand break repair via homologous recombination", 0.133,-6.171,-1.103, 1.380,-3.1076,0.796,0.164),
c("GO:0060119","inner ear receptor cell development", 0.225, 5.941,-1.907, 1.602,-1.3504,0.924,0.174),
c("GO:1901797","negative regulation of signal transduction by p53 class mediator", 0.162,-3.709,-3.060, 1.462,-2.3242,0.883,0.184),
c("GO:0010742","macrophage derived foam cell differentiation", 0.196, 4.507,-1.907, 1.544,-1.3504,0.945,0.191),
c("GO:0090077","foam cell differentiation", 0.196, 5.724,-3.935, 1.544,-1.3504,0.945,0.191),
c("GO:0051294","establishment of spindle orientation", 0.121, 6.483,-1.581, 1.342,-3.4231,0.781,0.192),
c("GO:0072161","mesenchymal cell differentiation involved in kidney development", 0.046, 5.932,-2.884, 0.954,-1.3504,0.944,0.198),
c("GO:0006417","regulation of translation", 2.031,-4.946, 4.659, 2.548,-4.0964,0.749,0.216),
c("GO:0006353","DNA-templated transcription, termination", 0.612,-6.956, 3.237, 2.029,-3.1725,0.869,0.217),
c("GO:1902416","positive regulation of mRNA binding", 0.035,-2.426,-5.650, 0.845,-1.2938,0.970,0.222),
c("GO:0006369","termination of RNA polymerase II transcription", 0.398,-7.360, 2.086, 1.845,-2.9735,0.870,0.223),
c("GO:0051453","regulation of intracellular pH", 0.519, 5.660, 0.611, 1.959,-1.4609,0.927,0.224),
c("GO:0006997","nucleus organization", 0.831, 5.346, 5.150, 2.161,-1.4434,0.904,0.230),
c("GO:0018205","peptidyl-lysine modification", 2.118,-2.023, 7.472, 2.566,-2.1569,0.861,0.232),
c("GO:0019083","viral transcription", 0.900,-6.172, 3.471, 2.196,-1.4167,0.836,0.234),
c("GO:0006513","protein monoubiquitination", 0.340,-1.541, 6.599, 1.778,-1.5968,0.882,0.241),
c("GO:0019941","modification-dependent protein catabolic process", 3.347,-2.821, 6.513, 2.764,-1.8571,0.853,0.247),
c("GO:0006487","protein N-linked glycosylation", 0.473,-3.393, 5.297, 1.919,-1.9053,0.835,0.249),
c("GO:0043436","oxoacid metabolic process", 6.186,-1.284, 3.822, 3.031,-1.7120,0.885,0.261),
c("GO:0051342","regulation of cyclic-nucleotide phosphodiesterase activity", 0.040, 0.031,-2.667, 0.903,-1.2938,0.969,0.265),
c("GO:0072599","establishment of protein localization to endoplasmic reticulum", 0.612, 1.375,-6.844, 2.029,-2.3242,0.880,0.274),
c("GO:0016071","mRNA metabolic process", 3.935,-7.255, 2.538, 2.834,-1.5834,0.864,0.276),
c("GO:0071499","cellular response to laminar fluid shear stress", 0.046,-1.812,-1.137, 0.954,-1.2938,0.937,0.282),
c("GO:2000271","positive regulation of fibroblast apoptotic process", 0.040, 5.027,-2.922, 0.903,-1.3504,0.932,0.283),
c("GO:1902275","regulation of chromatin organization", 0.935, 4.331, 6.546, 2.212,-2.6538,0.864,0.285),
c("GO:0051028","mRNA transport", 0.871, 0.569,-7.033, 2.182,-2.3087,0.888,0.304),
c("GO:0006396","RNA processing", 5.280,-6.738, 2.855, 2.962,-6.2457,0.847,0.310),
c("GO:0070316","regulation of G0 to G1 transition", 0.046, 2.433,-2.806, 0.954,-1.5553,0.912,0.329),
c("GO:0044267","cellular protein metabolic process",28.505,-3.619, 6.690, 3.694,-2.3503,0.854,0.333),
c("GO:1902108","regulation of mitochondrial membrane permeability involved in apoptotic process", 0.346, 6.559, 0.894, 1.785,-1.2938,0.799,0.334),
c("GO:0006888","ER to Golgi vesicle-mediated transport", 1.050, 1.703,-6.574, 2.262,-2.1791,0.871,0.336),
c("GO:0051963","regulation of synapse assembly", 0.467, 6.375, 2.475, 1.914,-1.6408,0.833,0.348),
c("GO:0007010","cytoskeleton organization", 6.861, 5.575, 5.301, 3.076,-1.3008,0.886,0.354),
c("GO:0022618","ribonucleoprotein complex assembly", 1.143, 4.142, 6.999, 2.299,-2.4384,0.883,0.365),
c("GO:0033044","regulation of chromosome organization", 0.819, 5.972, 4.963, 2.155,-1.9429,0.868,0.368),
c("GO:0019673","GDP-mannose metabolic process", 0.040,-4.382, 2.382, 0.903,-1.3504,0.908,0.369),
c("GO:0016070","RNA metabolic process",26.924,-6.528, 3.523, 3.669,-2.7923,0.835,0.398),
c("GO:0006032","chitin catabolic process", 0.017,-4.649, 1.539, 0.602,-1.2938,0.876,0.413),
c("GO:0022604","regulation of cell morphogenesis", 2.452, 6.374, 2.779, 2.629,-1.8188,0.849,0.417),
c("GO:0043470","regulation of carbohydrate catabolic process", 0.242,-4.478,-0.590, 1.633,-1.5968,0.896,0.420),
c("GO:0035456","response to interferon-beta", 0.133,-0.845,-3.018, 1.380,-2.1320,0.965,0.426),
c("GO:0000086","G2/M transition of mitotic cell cycle", 1.350, 2.692,-3.835, 2.371,-1.3008,0.925,0.429),
c("GO:0000122","negative regulation of transcription from RNA polymerase II promoter", 4.380,-6.372, 1.609, 2.881,-1.4998,0.769,0.430),
c("GO:0051017","actin filament bundle assembly", 0.767, 6.452, 4.489, 2.127,-1.5968,0.847,0.431),
c("GO:0035608","protein deglutamylation", 0.046,-1.071, 5.476, 0.954,-1.2938,0.899,0.431),
c("GO:1903578","regulation of ATP metabolic process", 0.317,-4.537, 3.159, 1.748,-1.4788,0.821,0.438),
c("GO:0016482","cytosolic transport", 0.842, 1.544,-6.342, 2.167,-2.1184,0.886,0.444),
c("GO:0009221","pyrimidine deoxyribonucleotide biosynthetic process", 0.035,-5.613, 4.761, 0.845,-1.3504,0.853,0.452),
c("GO:0048193","Golgi vesicle transport", 2.008, 0.706,-6.345, 2.543,-1.4870,0.917,0.455),
c("GO:1902855","regulation of non-motile cilium assembly", 0.075, 6.182, 4.569, 1.146,-1.3504,0.862,0.461),
c("GO:0090166","Golgi disassembly", 0.035, 5.759, 4.108, 0.845,-1.3504,0.887,0.472),
c("GO:0080135","regulation of cellular response to stress", 3.543,-2.825,-2.274, 2.789,-2.3087,0.882,0.476),
c("GO:0060253","negative regulation of glial cell proliferation", 0.075, 5.243,-1.132, 1.146,-1.3504,0.887,0.477),
c("GO:2000394","positive regulation of lamellipodium morphogenesis", 0.040, 6.124, 2.340, 0.903,-1.2938,0.878,0.483),
c("GO:0006013","mannose metabolic process", 0.058,-1.980, 2.263, 1.041,-1.2938,0.928,0.485),
c("GO:0071822","protein complex subunit organization", 9.106, 4.465, 6.661, 3.198,-1.4434,0.889,0.486),
c("GO:0051246","regulation of protein metabolic process",14.651,-3.952, 6.257, 3.405,-2.1542,0.835,0.488),
c("GO:0034720","histone H3-K4 demethylation", 0.046, 0.599, 7.593, 0.954,-1.2938,0.833,0.494),
c("GO:0001672","regulation of chromatin assembly or disassembly", 0.052, 3.295, 7.078, 1.000,-1.2938,0.890,0.499),
c("GO:1903827","regulation of cellular protein localization", 3.324, 0.739,-6.645, 2.761,-1.9099,0.853,0.499));

one.data <- data.frame(revigo.data);
names(one.data) <- revigo.names;
one.data <- one.data [(one.data$plot_X != "null" & one.data$plot_Y != "null"), ];
one.data$plot_X <- as.numeric( as.character(one.data$plot_X) );
one.data$plot_Y <- as.numeric( as.character(one.data$plot_Y) );
one.data$plot_size <- as.numeric( as.character(one.data$plot_size) );
one.data$log10_p_value <- as.numeric( as.character(one.data$log10_p_value) );
one.data$frequency <- as.numeric( as.character(one.data$frequency) );
one.data$uniqueness <- as.numeric( as.character(one.data$uniqueness) );
one.data$dispensability <- as.numeric( as.character(one.data$dispensability) );
#head(one.data);


# --------------------------------------------------------------------------
# Names of the axes, sizes of the numbers and letters, names of the columns,
# etc. can be changed below

p1 <- ggplot( data = one.data );
p1 <- p1 + geom_point( aes( plot_X, plot_Y, colour = log10_p_value, size = plot_size), alpha = I(0.6) ) + scale_size_area();
p1 <- p1 + scale_colour_gradientn( colours = c("blue", "green", "yellow", "red"), limits = c( min(one.data$log10_p_value), 0) );
p1 <- p1 + geom_point( aes(plot_X, plot_Y, size = plot_size), shape = 21, fill = "transparent", colour = I (alpha ("black", 0.6) )) + scale_size_area();
p1 <- p1 + scale_size( range=c(5, 30)) + theme_bw(); # + scale_fill_gradientn(colours = heat_hcl(7), limits = c(-300, 0) );
ex <- one.data [ one.data$dispensability < 0.15, ]; 
p1 <- p1 + geom_text( data = ex, aes(plot_X, plot_Y, label = description), colour = I(alpha("black", 0.85)), size = 3 );
p1 <- p1 + labs (y = "semantic space x", x = "semantic space y");
p1 <- p1 + theme(legend.key = element_blank()) ;
one.x_range = max(one.data$plot_X) - min(one.data$plot_X);
one.y_range = max(one.data$plot_Y) - min(one.data$plot_Y);
p1 <- p1 + xlim(min(one.data$plot_X)-one.x_range/10,max(one.data$plot_X)+one.x_range/10);
p1 <- p1 + ylim(min(one.data$plot_Y)-one.y_range/10,max(one.data$plot_Y)+one.y_range/10);



# --------------------------------------------------------------------------
# Output the plot to screen

p1;


#280419 filtered go
 #cut -f 2,3 /home/eneritz/ASCHROM0818/GO_terms_supfig4/K20_K79_GO.filter > lala
cut -f 2,3 /home/eneritz/ASCHROM0818/GO_terms_supfig4/K20_K79_GO.filter > lala
# --------------------------------------------------------------------------
# Here is your data from REVIGO. Scroll down for plot configuration options.

revigo.names <- c("term_ID","description","frequency_%","plot_X","plot_Y","plot_size","log10_p_value","uniqueness","dispensability");
revigo.data <- rbind(c("GO:0002478","antigen processing and presentation of exogenous peptide antigen", 0.975, 0.587, 3.428, 2.230,-2.4153,0.961,0.000),
c("GO:0008380","RNA splicing", 2.418,-4.606,-5.776, 2.623,-6.3116,0.809,0.000),
c("GO:0031648","protein destabilization", 0.219, 2.153,-3.019, 1.591,-1.7373,0.958,0.000),
c("GO:0006999","nuclear pore organization", 0.087, 2.373,-7.380, 1.204,-3.8500,0.882,0.002),
c("GO:0071479","cellular response to ionizing radiation", 0.375, 1.255,-0.643, 1.820,-3.6112,0.929,0.002),
c("GO:2000576","positive regulation of microtubule motor activity", 0.006, 1.908, 1.608, 0.301,-1.3504,0.975,0.017),
c("GO:0051196","regulation of coenzyme metabolic process", 0.306,-3.396, 1.971, 1.732,-1.5553,0.909,0.044),
c("GO:0006477","protein sulfation", 0.040,-4.929, 4.579, 0.903,-1.3504,0.903,0.064),
c("GO:0035646","endosome to melanosome transport", 0.058, 4.959, 4.176, 1.041,-2.7391,0.861,0.083),
c("GO:0035458","cellular response to interferon-beta", 0.087,-1.854, 5.182, 1.204,-2.5765,0.954,0.086),
c("GO:0010763","positive regulation of fibroblast migration", 0.069,-1.734, 6.638, 1.114,-1.3504,0.902,0.089),
c("GO:0030262","apoptotic nuclear changes", 0.167, 1.083,-6.421, 1.477,-1.9772,0.899,0.099),
c("GO:0010467","gene expression",30.542,-5.895,-0.466, 3.724,-4.7932,0.906,0.120),
c("GO:0043467","regulation of generation of precursor metabolites and energy", 0.502,-2.917,-0.929, 1.944,-1.4263,0.912,0.129),
c("GO:0010569","regulation of double-strand break repair via homologous recombination", 0.133,-3.388,-3.674, 1.380,-3.1076,0.794,0.164),
c("GO:0060119","inner ear receptor cell development", 0.225, 6.980, 0.036, 1.602,-1.3504,0.934,0.174),
c("GO:1901797","negative regulation of signal transduction by p53 class mediator", 0.162,-0.931,-0.061, 1.462,-2.3242,0.880,0.184),
c("GO:0010742","macrophage derived foam cell differentiation", 0.196, 6.393, 1.674, 1.544,-1.3504,0.945,0.191),
c("GO:0090077","foam cell differentiation", 0.196, 6.346, 2.733, 1.544,-1.3504,0.945,0.191),
c("GO:0051294","establishment of spindle orientation", 0.121, 4.251, 1.224, 1.342,-3.4231,0.772,0.192),
c("GO:0072161","mesenchymal cell differentiation involved in kidney development", 0.046, 6.966, 1.543, 0.954,-1.3504,0.947,0.198),
c("GO:0006417","regulation of translation", 2.031,-6.138,-1.683, 2.548,-4.0964,0.746,0.216),
c("GO:0006353","DNA-templated transcription, termination", 0.612,-4.823,-6.198, 2.029,-3.1725,0.863,0.217),
c("GO:0006369","termination of RNA polymerase II transcription", 0.398,-4.106,-6.483, 1.845,-2.9735,0.864,0.223),
c("GO:0051453","regulation of intracellular pH", 0.519, 2.602,-3.229, 1.959,-1.4609,0.930,0.224),
c("GO:0006997","nucleus organization", 0.831, 4.114,-6.067, 2.161,-1.4434,0.906,0.230),
c("GO:0018205","peptidyl-lysine modification", 2.118,-6.358, 3.154, 2.566,-2.1569,0.858,0.232),
c("GO:0019083","viral transcription", 0.900,-5.014,-5.488, 2.196,-1.4167,0.832,0.234),
c("GO:0006513","protein monoubiquitination", 0.340,-5.623, 3.913, 1.778,-1.5968,0.877,0.241),
c("GO:0019941","modification-dependent protein catabolic process", 3.347,-6.374, 2.377, 2.764,-1.8571,0.850,0.247),
c("GO:0006487","protein N-linked glycosylation", 0.473,-6.477,-0.411, 1.919,-1.9053,0.839,0.249),
c("GO:0043436","oxoacid metabolic process", 6.186,-1.540,-4.503, 3.031,-1.7120,0.886,0.261),
c("GO:0072599","establishment of protein localization to endoplasmic reticulum", 0.612, 1.995, 6.175, 2.029,-2.3242,0.867,0.274),
c("GO:0016071","mRNA metabolic process", 3.935,-5.709,-5.090, 2.834,-1.5834,0.857,0.276),
c("GO:2000271","positive regulation of fibroblast apoptotic process", 0.040, 3.758, 0.182, 0.903,-1.3504,0.934,0.283),
c("GO:1902275","regulation of chromatin organization", 0.935, 3.569,-6.934, 2.212,-2.6538,0.873,0.285),
c("GO:0051028","mRNA transport", 0.871, 0.762, 6.697, 2.182,-2.3087,0.876,0.304),
c("GO:0006396","RNA processing", 5.280,-5.733,-4.710, 2.962,-6.2457,0.839,0.310),
c("GO:0070316","regulation of G0 to G1 transition", 0.046, 4.054, 3.291, 0.954,-1.5553,0.910,0.329),
c("GO:0044267","cellular protein metabolic process",28.505,-6.848, 0.683, 3.694,-2.3503,0.848,0.333),
c("GO:0006888","ER to Golgi vesicle-mediated transport", 1.050, 2.328, 5.952, 2.262,-2.1791,0.857,0.336),
c("GO:0051963","regulation of synapse assembly", 0.467, 5.533,-3.307, 1.914,-1.6408,0.844,0.348),
c("GO:0022618","ribonucleoprotein complex assembly", 1.143, 2.949,-7.267, 2.299,-2.4384,0.890,0.365),
c("GO:0033044","regulation of chromosome organization", 0.819, 4.805,-5.523, 2.155,-1.9429,0.872,0.368),
c("GO:0019673","GDP-mannose metabolic process", 0.040,-2.482,-6.835, 0.903,-1.3504,0.908,0.369),
c("GO:0016070","RNA metabolic process",26.924,-6.127,-3.492, 3.669,-2.7923,0.827,0.398),
c("GO:0022604","regulation of cell morphogenesis", 2.452, 6.000,-2.627, 2.629,-1.8188,0.858,0.417),
c("GO:0043470","regulation of carbohydrate catabolic process", 0.242,-5.033, 2.231, 1.633,-1.5968,0.901,0.420),
c("GO:0035456","response to interferon-beta", 0.133,-1.491, 4.985, 1.380,-2.1320,0.962,0.426),
c("GO:0000122","negative regulation of transcription from RNA polymerase II promoter", 4.380,-5.294,-2.940, 2.881,-1.4998,0.766,0.430),
c("GO:0051017","actin filament bundle assembly", 0.767, 4.746,-5.140, 2.127,-1.5968,0.846,0.431),
c("GO:1903578","regulation of ATP metabolic process", 0.317,-4.804,-3.218, 1.748,-1.4788,0.824,0.438),
c("GO:0016482","cytosolic transport", 0.842, 2.450, 5.697, 2.167,-2.1184,0.874,0.444),
c("GO:0009221","pyrimidine deoxyribonucleotide biosynthetic process", 0.035,-5.366,-3.813, 0.845,-1.3504,0.855,0.452),
c("GO:0048193","Golgi vesicle transport", 2.008, 0.703, 6.285, 2.543,-1.4870,0.909,0.455),
c("GO:1902855","regulation of non-motile cilium assembly", 0.075, 5.131,-5.162, 1.146,-1.3504,0.867,0.461),
c("GO:0031643","positive regulation of myelination", 0.063, 6.877,-1.307, 1.079,-1.3504,0.915,0.470),
c("GO:0090166","Golgi disassembly", 0.035, 3.484,-5.430, 0.845,-1.3504,0.885,0.472),
c("GO:0080135","regulation of cellular response to stress", 3.543,-0.692, 0.657, 2.789,-2.3087,0.879,0.476),
c("GO:0060253","negative regulation of glial cell proliferation", 0.075, 6.466,-0.582, 1.146,-1.3504,0.894,0.477),
c("GO:0071822","protein complex subunit organization", 9.106, 2.770,-7.102, 3.198,-1.4434,0.896,0.486),
c("GO:0051246","regulation of protein metabolic process",14.651,-6.182, 1.168, 3.405,-2.1542,0.832,0.488),
c("GO:1903827","regulation of cellular protein localization", 3.324, 1.459, 6.145, 2.761,-1.9099,0.840,0.499));

one.data <- data.frame(revigo.data);
names(one.data) <- revigo.names;
one.data <- one.data [(one.data$plot_X != "null" & one.data$plot_Y != "null"), ];
one.data$plot_X <- as.numeric( as.character(one.data$plot_X) );
one.data$plot_Y <- as.numeric( as.character(one.data$plot_Y) );
one.data$plot_size <- as.numeric( as.character(one.data$plot_size) );
one.data$log10_p_value <- as.numeric( as.character(one.data$log10_p_value) );
one.data$frequency <- as.numeric( as.character(one.data$frequency) );
one.data$uniqueness <- as.numeric( as.character(one.data$uniqueness) );
one.data$dispensability <- as.numeric( as.character(one.data$dispensability) );
#head(one.data);


# --------------------------------------------------------------------------
# Names of the axes, sizes of the numbers and letters, names of the columns,
# etc. can be changed below

p1 <- ggplot( data = one.data );
p1 <- p1 + geom_point( aes( plot_X, plot_Y, colour = log10_p_value, size = plot_size), alpha = I(0.6) ) + scale_size_area();
p1 <- p1 + scale_colour_gradientn( colours = c("blue", "green", "yellow", "red"), limits = c( min(one.data$log10_p_value), 0) );
p1 <- p1 + geom_point( aes(plot_X, plot_Y, size = plot_size), shape = 21, fill = "transparent", colour = I (alpha ("black", 0.6) )) + scale_size_area();
p1 <- p1 + scale_size( range=c(5, 30)) + theme_bw(); # + scale_fill_gradientn(colours = heat_hcl(7), limits = c(-300, 0) );
ex <- one.data [ one.data$dispensability < 0.15, ]; 
p1 <- p1 + geom_text( data = ex, aes(plot_X, plot_Y, label = description), colour = I(alpha("black", 0.85)), size = 3 );
p1 <- p1 + labs (y = "semantic space x", x = "semantic space y");
p1 <- p1 + theme(legend.key = element_blank()) ;
one.x_range = max(one.data$plot_X) - min(one.data$plot_X);
one.y_range = max(one.data$plot_Y) - min(one.data$plot_Y);
p1 <- p1 + xlim(min(one.data$plot_X)-one.x_range/10,max(one.data$plot_X)+one.x_range/10);
p1 <- p1 + ylim(min(one.data$plot_Y)-one.y_range/10,max(one.data$plot_Y)+one.y_range/10);



# --------------------------------------------------------------------------
# Output the plot to screen

p1;

######more filtered
cut -f 2,4 K20_K79_GO.filter_2  > lala


# --------------------------------------------------------------------------
# If you don't have the ggplot2 package installed, uncomment the following line:
# install.packages( "ggplot2" );
library( ggplot2 );
# --------------------------------------------------------------------------
# If you don't have the scales package installed, uncomment the following line:
# install.packages( "scales" );
library( scales );


# --------------------------------------------------------------------------
# Here is your data from REVIGO. Scroll down for plot configuration options.

revigo.names <- c("term_ID","description","frequency_%","plot_X","plot_Y","plot_size","log10_p_value","uniqueness","dispensability");
revigo.data <- rbind(c("GO:0008380","RNA splicing", 2.418,-3.962,-0.787, 2.623,-3.6039,0.391,0.000),
c("GO:0006999","nuclear pore organization", 0.087, 4.964,-1.979, 1.204,-1.8106,0.834,0.002),
c("GO:0071479","cellular response to ionizing radiation", 0.375, 3.174, 4.493, 1.820,-1.6229,0.848,0.002),
c("GO:0010467","gene expression",30.542,-3.107, 5.536, 3.724,-2.6067,0.752,0.120),
c("GO:0010569","regulation of double-strand break repair via homologous recombination", 0.133,-3.370,-6.065, 1.380,-1.3412,0.650,0.164),
c("GO:0051294","establishment of spindle orientation", 0.121, 4.128,-4.859, 1.342,-1.4806,0.828,0.192),
c("GO:0006417","regulation of translation", 2.031,-4.640,-3.435, 2.548,-1.9990,0.543,0.216),
c("GO:0006353","   DNA-templated transcription, termination", 0.612,-7.140,-1.970, 2.029,-1.3761,0.614,0.217),
c("GO:0006396","RNA processing", 5.280,-5.381, 0.147, 2.962,-3.6039,0.560,0.288));

one.data <- data.frame(revigo.data);
names(one.data) <- revigo.names;
one.data <- one.data [(one.data$plot_X != "null" & one.data$plot_Y != "null"), ];
one.data$plot_X <- as.numeric( as.character(one.data$plot_X) );
one.data$plot_Y <- as.numeric( as.character(one.data$plot_Y) );
one.data$plot_size <- as.numeric( as.character(one.data$plot_size) );
one.data$log10_p_value <- as.numeric( as.character(one.data$log10_p_value) );
one.data$frequency <- as.numeric( as.character(one.data$frequency) );
one.data$uniqueness <- as.numeric( as.character(one.data$uniqueness) );
one.data$dispensability <- as.numeric( as.character(one.data$dispensability) );
#head(one.data);


# --------------------------------------------------------------------------
# Names of the axes, sizes of the numbers and letters, names of the columns,
# etc. can be changed below

p1 <- ggplot( data = one.data );
p1 <- p1 + geom_point( aes( plot_X, plot_Y, colour = log10_p_value, size = plot_size), alpha = I(0.6) ) + scale_size_area();
p1 <- p1 + scale_colour_gradientn( colours = c("blue", "green", "yellow", "red"), limits = c( min(one.data$log10_p_value), 0) );
p1 <- p1 + geom_point( aes(plot_X, plot_Y, size = plot_size), shape = 21, fill = "transparent", colour = I (alpha ("black", 0.6) )) + scale_size_area();
p1 <- p1 + scale_size( range=c(5, 30)) + theme_bw(); # + scale_fill_gradientn(colours = heat_hcl(7), limits = c(-300, 0) );
ex <- one.data [ one.data$dispensability < 0.3, ]; 
p1 <- p1 + geom_text_repel( data = ex, aes(plot_X, plot_Y, label = description), colour = I(alpha("black", 0.85)), size = 4 );
p1 <- p1 + labs (y = "semantic space x", x = "semantic space y");
p1 <- p1 + theme(legend.key = element_blank()) ;
one.x_range = max(one.data$plot_X) - min(one.data$plot_X);
one.y_range = max(one.data$plot_Y) - min(one.data$plot_Y);
p1 <- p1 + xlim(min(one.data$plot_X)-one.x_range/10,max(one.data$plot_X)+one.x_range/10);
p1 <- p1 + ylim(min(one.data$plot_Y)-one.y_range/10,max(one.data$plot_Y)+one.y_range/10);



# --------------------------------------------------------------------------
# Output the plot to screen

p1;



```
K9 5mC excluded
```{r}


# --------------------------------------------------------------------------
# If you don't have the ggplot2 package installed, uncomment the following line:
# install.packages( "ggplot2" );
library( ggplot2 );
# --------------------------------------------------------------------------
# If you don't have the scales package installed, uncomment the following line:
# install.packages( "scales" );
library( scales );


# --------------------------------------------------------------------------
# Here is your data from REVIGO. Scroll down for plot configuration options.

revigo.names <- c("term_ID","description","frequency_%","plot_X","plot_Y","plot_size","log10_p_value","uniqueness","dispensability");
revigo.data <- rbind(c("GO:0090403","oxidative stress-induced premature senescence", 0.023, 0.139, 1.000, 0.699,-1.6021,0.005,0.000),
c("GO:0030913","paranodal junction assembly", 0.040, 1.000, 0.139, 0.903,-1.0655,0.128,0.139));

one.data <- data.frame(revigo.data);
names(one.data) <- revigo.names;
one.data <- one.data [(one.data$plot_X != "null" & one.data$plot_Y != "null"), ];
one.data$plot_X <- as.numeric( as.character(one.data$plot_X) );
one.data$plot_Y <- as.numeric( as.character(one.data$plot_Y) );
one.data$plot_size <- as.numeric( as.character(one.data$plot_size) );
one.data$log10_p_value <- as.numeric( as.character(one.data$log10_p_value) );
one.data$frequency <- as.numeric( as.character(one.data$frequency) );
one.data$uniqueness <- as.numeric( as.character(one.data$uniqueness) );
one.data$dispensability <- as.numeric( as.character(one.data$dispensability) );
#head(one.data);


# --------------------------------------------------------------------------
# Names of the axes, sizes of the numbers and letters, names of the columns,
# etc. can be changed below

p1 <- ggplot( data = one.data );
p1 <- p1 + geom_point( aes( plot_X, plot_Y, colour = log10_p_value, size = plot_size), alpha = I(0.6) ) + scale_size_area();
p1 <- p1 + scale_colour_gradientn( colours = c("blue", "green", "yellow", "red"), limits = c( min(one.data$log10_p_value), 0) );
p1 <- p1 + geom_point( aes(plot_X, plot_Y, size = plot_size), shape = 21, fill = "transparent", colour = I (alpha ("black", 0.6) )) + scale_size_area();
p1 <- p1 + scale_size( range=c(5, 30)) + theme_bw(); # + scale_fill_gradientn(colours = heat_hcl(7), limits = c(-300, 0) );
ex <- one.data [ one.data$dispensability < 0.15, ]; 
p1 <- p1 + geom_text( data = ex, aes(plot_X, plot_Y, label = description), colour = I(alpha("black", 0.85)), size = 3 );
p1 <- p1 + labs (y = "semantic space x", x = "semantic space y");
p1 <- p1 + theme(legend.key = element_blank()) ;
one.x_range = max(one.data$plot_X) - min(one.data$plot_X);
one.y_range = max(one.data$plot_Y) - min(one.data$plot_Y);
p1 <- p1 + xlim(min(one.data$plot_X)-one.x_range/10,max(one.data$plot_X)+one.x_range/10);
p1 <- p1 + ylim(min(one.data$plot_Y)-one.y_range/10,max(one.data$plot_Y)+one.y_range/10);

 plot.new()
> 
> legend("top", inset=.03, title=" ",       c( "Excluded" ,  "H4K20me1+H3K79me2" ,  "H3K9me3+5mC" , "H3K9ac+H3K14ac" , "Included" , "H4K20me1+H4K91ac" , "H3K9me3+5mC" , "Constitutives"  ), fill =c("lightgray", "dodgerblue", "dodgerblue" , "dodgerblue4" , "lightgray" , "orange", "firebrick3" , "lightgrey" ) , border=c("blue", "dodgerblue", "dodgerblue", "dodgerblue4" , "red", "orange", "firebrick3" , "black" ) , lwd=6, cex=1.4, horiz = F , box.col="white"  )
> 

# --------------------------------------------------------------------------
# Output the plot to screen

p1;


###enrichr


# --------------------------------------------------------------------------
# If you don't have the ggplot2 package installed, uncomment the following line:
# install.packages( "ggplot2" );
library( ggplot2 );
# --------------------------------------------------------------------------
# If you don't have the scales package installed, uncomment the following line:
# install.packages( "scales" );
library( scales );


# --------------------------------------------------------------------------
# Here is your data from REVIGO. Scroll down for plot configuration options.

revigo.names <- c("term_ID","description","frequency_%","plot_X","plot_Y","plot_size","log10_p_value","uniqueness","dispensability");
revigo.data <- rbind(c("GO:0061001","regulation of dendritic spine morphogenesis", 0.196, 5.409, 2.948, 1.544,-3.8984,0.729,0.000),
c("GO:0035690","cellular response to drug", 0.381, 3.234,-2.604, 1.826,-2.1458,0.956,0.002),
c("GO:0032092","positive regulation of protein binding", 0.462,-5.448,-3.039, 1.908,-1.5851,0.919,0.023),
c("GO:0035308","negative regulation of protein dephosphorylation", 0.133,-4.925, 5.919, 1.380,-2.5664,0.809,0.031),
c("GO:0051282","regulation of sequestering of calcium ion", 0.600, 3.631,-6.095, 2.021,-3.1358,0.821,0.035),
c("GO:0016101","diterpenoid metabolic process", 0.554, 0.290, 1.681, 1.987,-1.5591,0.917,0.036),
c("GO:0043068","positive regulation of programmed cell death", 3.428,-4.518, 0.313, 2.775,-2.3785,0.851,0.047),
c("GO:0044772","mitotic cell cycle phase transition", 3.064, 1.319, 3.343, 2.726,-1.8804,0.914,0.051),
c("GO:0010212","response to ionizing radiation", 0.854, 1.653, 4.169, 2.173,-1.3893,0.973,0.069),
c("GO:0007185","transmembrane receptor protein tyrosine phosphatase signaling pathway", 0.040, 1.313,-1.432, 0.903,-1.5474,0.924,0.081),
c("GO:0000045","autophagosome assembly", 0.410,-3.489,-5.478, 1.857,-1.6120,0.896,0.113),
c("GO:0034394","protein localization to cell surface", 0.306, 0.450,-7.514, 1.732,-2.4009,0.917,0.120),
c("GO:0007398","ectoderm development", 0.121, 6.953, 2.920, 1.342,-1.3546,0.940,0.122),
c("GO:0010646","regulation of cell communication",17.651, 1.258,-2.830, 3.486,-1.4525,0.895,0.133),
c("GO:0016197","endosomal transport", 1.570, 1.656,-6.230, 2.436,-1.8291,0.936,0.141),
c("GO:0042753","positive regulation of circadian rhythm", 0.104,-2.585, 1.008, 1.279,-1.3546,0.922,0.158),
c("GO:0070933","histone H4 deacetylation", 0.052,-6.689,-0.615, 1.000,-1.4400,0.868,0.162),
c("GO:0090400","stress-induced premature senescence", 0.046, 3.209, 3.845, 0.954,-3.2315,0.882,0.162),
c("GO:0003230","cardiac atrium development", 0.202, 4.245, 6.424, 1.556,-1.4903,0.920,0.162),
c("GO:0006563","L-serine metabolic process", 0.063,-0.773, 4.260, 1.079,-1.3177,0.915,0.169),
c("GO:0006511","ubiquitin-dependent protein catabolic process", 3.306,-5.321, 2.422, 2.759,-2.5503,0.877,0.179),
c("GO:0070936","protein K48-linked ubiquitination", 0.283,-4.945, 3.450, 1.699,-1.7635,0.919,0.193),
c("GO:0008285","negative regulation of cell proliferation", 3.716,-1.020, 7.779, 2.810,-1.7713,0.853,0.210),
c("GO:0006497","protein lipidation", 0.842,-5.387, 4.132, 2.167,-1.8038,0.890,0.219),
c("GO:0051095","regulation of helicase activity", 0.052,-4.745,-2.373, 1.000,-1.3951,0.936,0.224),
c("GO:0045714","regulation of low-density lipoprotein particle receptor biosynthetic process", 0.052,-5.698, 5.404, 1.000,-1.4903,0.862,0.231),
c("GO:2000322","regulation of glucocorticoid receptor signaling pathway", 0.046, 4.731,-3.591, 0.954,-1.4903,0.899,0.236),
c("GO:0007399","nervous system development",12.770, 5.317, 5.711, 3.345,-1.4046,0.890,0.244),
c("GO:0031532","actin cytoskeleton reorganization", 0.502,-1.328,-5.504, 1.944,-1.5720,0.879,0.245),
c("GO:0010882","regulation of cardiac muscle contraction by calcium ion signaling", 0.127, 6.469, 1.069, 1.362,-2.4393,0.823,0.247),
c("GO:0090091","positive regulation of extracellular matrix disassembly", 0.040,-3.544, 0.866, 0.903,-1.4903,0.833,0.249),
c("GO:0042634","regulation of hair cycle", 0.144, 4.762, 3.893, 1.415,-1.3951,0.885,0.249),
c("GO:0051974","negative regulation of telomerase activity", 0.069,-5.988, 3.165, 1.114,-1.4400,0.817,0.250),
c("GO:0016192","vesicle-mediated transport",11.385, 1.978,-6.599, 3.295,-1.5702,0.924,0.275),
c("GO:2000112","regulation of cellular macromolecule biosynthetic process",23.220,-6.166, 4.815, 3.605,-2.3616,0.812,0.279),
c("GO:0060333","interferon-gamma-mediated signaling pathway", 0.496, 4.447,-1.178, 1.940,-1.4636,0.884,0.287),
c("GO:0007043","cell-cell junction assembly", 0.537,-2.967,-3.981, 1.973,-1.3695,0.890,0.292),
c("GO:0072331","signal transduction by p53 class mediator", 1.518, 6.210,-3.020, 2.422,-1.9315,0.894,0.296),
c("GO:0007422","peripheral nervous system development", 0.415, 4.628, 5.382, 1.863,-2.4009,0.893,0.306),
c("GO:0006888","ER to Golgi vesicle-mediated transport", 1.050, 0.760,-7.153, 2.262,-1.4160,0.904,0.313),
c("GO:2000739","regulation of mesenchymal stem cell differentiation", 0.035, 6.628, 4.760, 0.845,-1.5474,0.863,0.314),
c("GO:1901019","regulation of calcium ion transmembrane transporter activity", 0.450,-1.999,-6.598, 1.898,-2.2323,0.812,0.315),
c("GO:0038180","nerve growth factor signaling pathway", 0.046, 3.697,-3.336, 0.954,-1.3546,0.915,0.319),
c("GO:1902959","regulation of aspartic-type endopeptidase activity involved in amyloid precursor protein catabolic process", 0.012,-7.481, 0.517, 0.477,-1.4400,0.855,0.327),
c("GO:1901985","positive regulation of protein acetylation", 0.202,-6.664, 3.571, 1.556,-1.5474,0.821,0.329),
c("GO:0007010","cytoskeleton organization", 6.861,-1.734,-4.672, 3.076,-1.8129,0.895,0.331),
c("GO:2001014","regulation of skeletal muscle cell differentiation", 0.098, 5.926, 3.984, 1.255,-1.3951,0.836,0.339),
c("GO:0042552","myelination", 0.629, 4.865, 5.690, 2.041,-1.9743,0.865,0.340),
c("GO:0001558","regulation of cell growth", 2.250,-1.181,-3.572, 2.592,-1.2984,0.830,0.348),
c("GO:0045581","negative regulation of T cell differentiation", 0.196, 3.353, 6.162, 1.544,-1.3177,0.780,0.357),
c("GO:0032933","SREBP signaling pathway", 0.046,-2.703, 4.146, 0.954,-1.5474,0.805,0.366),
c("GO:0080135","regulation of cellular response to stress", 3.543, 3.548, 0.400, 2.789,-2.0351,0.878,0.366),
c("GO:0010968","regulation of microtubule nucleation", 0.029,-3.319,-4.716, 0.778,-1.5474,0.822,0.371),
c("GO:1990440","positive regulation of transcription from RNA polymerase II promoter in response to endoplasmic reticulum stress", 0.069,-3.073, 5.617, 1.114,-1.3177,0.819,0.380),
c("GO:0010628","positive regulation of gene expression", 9.833,-6.862, 3.386, 3.232,-1.4096,0.811,0.383),
c("GO:0022603","regulation of anatomical structure morphogenesis", 5.430, 5.842, 5.076, 2.974,-1.3407,0.833,0.389),
c("GO:0001561","fatty acid alpha-oxidation", 0.040,-0.893, 2.458, 0.903,-1.4400,0.904,0.398),
c("GO:2000637","positive regulation of gene silencing by miRNA", 0.115,-4.770, 5.352, 1.322,-2.4795,0.762,0.398),
c("GO:0050730","regulation of peptidyl-tyrosine phosphorylation", 1.287,-6.280, 4.211, 2.350,-1.3130,0.828,0.399),
c("GO:0050796","regulation of insulin secretion", 0.975, 2.850,-5.966, 2.230,-2.1049,0.788,0.402),
c("GO:0048523","negative regulation of cellular process",24.951,-1.464, 7.761, 3.636,-1.6465,0.834,0.411),
c("GO:0080134","regulation of response to stress", 7.657, 3.029, 0.716, 3.123,-1.3546,0.895,0.412),
c("GO:0042147","retrograde transport, endosome to Golgi", 0.496, 0.666,-7.005, 1.940,-1.3407,0.909,0.420),
c("GO:1903689","regulation of wound healing, spreading of epidermal cells", 0.040, 6.237, 0.769, 0.903,-1.5474,0.789,0.425),
c("GO:2000074","regulation of type B pancreatic cell development", 0.046, 6.235, 4.000, 0.954,-1.4400,0.827,0.426),
c("GO:0070243","regulation of thymocyte apoptotic process", 0.069,-0.678,-0.265, 1.114,-1.5474,0.918,0.430),
c("GO:0072332","intrinsic apoptotic signaling pathway by p53 class mediator", 0.473, 3.994,-1.873, 1.919,-1.8523,0.882,0.444),
c("GO:0072393","microtubule anchoring at microtubule organizing center", 0.046,-2.443,-5.582, 0.954,-1.3177,0.883,0.445),
c("GO:0043547","positive regulation of GTPase activity", 3.803,-5.088,-2.529, 2.820,-1.3468,0.907,0.454),
c("GO:0072657","protein localization to membrane", 2.954, 0.714,-6.572, 2.710,-2.3617,0.802,0.456),
c("GO:0045899","positive regulation of RNA polymerase II transcriptional preinitiation complex assembly", 0.058,-5.945, 0.207, 1.041,-1.4400,0.765,0.476),
c("GO:0000122","negative regulation of transcription from RNA polymerase II promoter", 4.380,-3.908, 6.309, 2.881,-2.0679,0.761,0.482),
c("GO:0048710","regulation of astrocyte differentiation", 0.144, 5.587, 4.528, 1.415,-1.3546,0.800,0.490),
c("GO:0035458","cellular response to interferon-beta", 0.087, 4.815,-3.152, 1.204,-1.3177,0.956,0.493),
c("GO:0051865","protein autoubiquitination", 0.317,-4.910, 3.227, 1.748,-1.7469,0.918,0.500));

one.data <- data.frame(revigo.data);
names(one.data) <- revigo.names;
one.data <- one.data [(one.data$plot_X != "null" & one.data$plot_Y != "null"), ];
one.data$plot_X <- as.numeric( as.character(one.data$plot_X) );
one.data$plot_Y <- as.numeric( as.character(one.data$plot_Y) );
one.data$plot_size <- as.numeric( as.character(one.data$plot_size) );
one.data$log10_p_value <- as.numeric( as.character(one.data$log10_p_value) );
one.data$frequency <- as.numeric( as.character(one.data$frequency) );
one.data$uniqueness <- as.numeric( as.character(one.data$uniqueness) );
one.data$dispensability <- as.numeric( as.character(one.data$dispensability) );
#head(one.data);


# --------------------------------------------------------------------------
# Names of the axes, sizes of the numbers and letters, names of the columns,
# etc. can be changed below

p1 <- ggplot( data = one.data );
p1 <- p1 + geom_point( aes( plot_X, plot_Y, colour = log10_p_value, size = plot_size), alpha = I(0.6) ) + scale_size_area();
p1 <- p1 + scale_colour_gradientn( colours = c("blue", "green", "yellow", "red"), limits = c( min(one.data$log10_p_value), 0) );
p1 <- p1 + geom_point( aes(plot_X, plot_Y, size = plot_size), shape = 21, fill = "transparent", colour = I (alpha ("black", 0.6) )) + scale_size_area();
p1 <- p1 + scale_size( range=c(5, 30)) + theme_bw(); # + scale_fill_gradientn(colours = heat_hcl(7), limits = c(-300, 0) );
ex <- one.data [ one.data$dispensability < 0.15, ]; 
p1 <- p1 + geom_text( data = ex, aes(plot_X, plot_Y, label = description), colour = I(alpha("black", 0.85)), size = 3 );
p1 <- p1 + labs (y = "semantic space x", x = "semantic space y");
p1 <- p1 + theme(legend.key = element_blank()) ;
one.x_range = max(one.data$plot_X) - min(one.data$plot_X);
one.y_range = max(one.data$plot_Y) - min(one.data$plot_Y);
p1 <- p1 + xlim(min(one.data$plot_X)-one.x_range/10,max(one.data$plot_X)+one.x_range/10);
p1 <- p1 + ylim(min(one.data$plot_Y)-one.y_range/10,max(one.data$plot_Y)+one.y_range/10);



# --------------------------------------------------------------------------
# Output the plot to screen

p1;


##filtered enrichr
 #cut -f 2,3 /home/eneritz/ASCHROM0818/GO_terms_supfig4/K9_5mC_E_GO.filter > lal

# --------------------------------------------------------------------------
# Here is your data from REVIGO. Scroll down for plot configuration options.

revigo.names <- c("term_ID","description","frequency_%","plot_X","plot_Y","plot_size","log10_p_value","uniqueness","dispensability");
revigo.data <- rbind(c("GO:0061001","regulation of dendritic spine morphogenesis", 0.196, 5.318,-3.556, 1.544,-3.8984,0.732,0.000),
c("GO:0035690","cellular response to drug", 0.381, 3.602, 2.850, 1.826,-2.1458,0.956,0.002),
c("GO:0032092","positive regulation of protein binding", 0.462,-5.715, 3.323, 1.908,-1.5851,0.919,0.023),
c("GO:0035308","negative regulation of protein dephosphorylation", 0.133,-5.024,-5.825, 1.380,-2.5664,0.809,0.031),
c("GO:0051282","regulation of sequestering of calcium ion", 0.600, 2.815, 6.542, 2.021,-3.1358,0.822,0.035),
c("GO:0016101","diterpenoid metabolic process", 0.554, 0.465,-1.644, 1.987,-1.5591,0.917,0.036),
c("GO:0043068","positive regulation of programmed cell death", 3.428,-4.118,-0.400, 2.775,-2.3785,0.851,0.047),
c("GO:0044772","mitotic cell cycle phase transition", 3.064, 1.586,-3.075, 2.726,-1.8804,0.915,0.051),
c("GO:0010212","response to ionizing radiation", 0.854, 1.934,-3.850, 2.173,-1.3893,0.973,0.069),
c("GO:0007185","transmembrane receptor protein tyrosine phosphatase signaling pathway", 0.040, 1.688, 1.653, 0.903,-1.5474,0.924,0.081),
c("GO:0000045","autophagosome assembly", 0.410,-3.433, 4.721, 1.857,-1.6120,0.898,0.113),
c("GO:0034394","protein localization to cell surface", 0.306,-0.334, 7.586, 1.732,-2.4009,0.917,0.120),
c("GO:0007398","ectoderm development", 0.121, 7.016,-2.442, 1.342,-1.3546,0.940,0.122),
c("GO:0010646","regulation of cell communication",17.651, 1.542, 3.880, 3.486,-1.4525,0.896,0.133),
c("GO:0016197","endosomal transport", 1.570,-0.261, 6.678, 2.436,-1.8291,0.936,0.141),
c("GO:0042753","positive regulation of circadian rhythm", 0.104,-2.758,-0.812, 1.279,-1.3546,0.923,0.158),
c("GO:0070933","histone H4 deacetylation", 0.052,-6.326, 0.228, 1.000,-1.4400,0.868,0.162),
c("GO:0090400","stress-induced premature senescence", 0.046, 3.553,-3.165, 0.954,-3.2315,0.882,0.162),
c("GO:0003230","cardiac atrium development", 0.202, 4.270,-6.369, 1.556,-1.4903,0.920,0.162),
c("GO:0006563","L-serine metabolic process", 0.063,-0.803,-3.634, 1.079,-1.3177,0.915,0.169),
c("GO:0006511","ubiquitin-dependent protein catabolic process", 3.306,-5.003,-2.319, 2.759,-2.5503,0.871,0.179),
c("GO:0070936","protein K48-linked ubiquitination", 0.283,-5.407,-2.839, 1.699,-1.7635,0.918,0.193),
c("GO:0008285","negative regulation of cell proliferation", 3.716,-1.004,-7.789, 2.810,-1.7713,0.853,0.210),
c("GO:0006497","protein lipidation", 0.842,-5.174,-4.236, 2.167,-1.8038,0.889,0.219),
c("GO:0051095","regulation of helicase activity", 0.052,-5.161, 2.630, 1.000,-1.3951,0.936,0.224),
c("GO:0045714","regulation of low-density lipoprotein particle receptor biosynthetic process", 0.052,-5.720,-5.324, 1.000,-1.4903,0.862,0.231),
c("GO:2000322","regulation of glucocorticoid receptor signaling pathway", 0.046, 4.477, 4.051, 0.954,-1.4903,0.899,0.236),
c("GO:0007399","nervous system development",12.770, 5.334,-5.676, 3.345,-1.4046,0.890,0.244),
c("GO:0031532","actin cytoskeleton reorganization", 0.502,-0.081, 4.886, 1.944,-1.5720,0.880,0.245),
c("GO:0010882","regulation of cardiac muscle contraction by calcium ion signaling", 0.127, 6.576,-0.859, 1.362,-2.4393,0.823,0.247),
c("GO:0090091","positive regulation of extracellular matrix disassembly", 0.040,-1.299,-5.084, 0.903,-1.4903,0.836,0.249),
c("GO:0042634","regulation of hair cycle", 0.144, 4.684,-3.989, 1.415,-1.3951,0.885,0.249),
c("GO:0051974","negative regulation of telomerase activity", 0.069,-6.711,-2.209, 1.114,-1.4400,0.816,0.250),
c("GO:0016192","vesicle-mediated transport",11.385, 0.544, 7.011, 3.295,-1.5702,0.924,0.275),
c("GO:2000112","regulation of cellular macromolecule biosynthetic process",23.220,-6.243,-4.671, 3.605,-2.3616,0.812,0.279),
c("GO:0060333","interferon-gamma-mediated signaling pathway", 0.496, 4.579, 1.663, 1.940,-1.4636,0.885,0.287),
c("GO:0007043","cell-cell junction assembly", 0.537,-2.053, 2.944, 1.973,-1.3695,0.891,0.292),
c("GO:0072331","signal transduction by p53 class mediator", 1.518, 5.805, 4.099, 2.422,-1.9315,0.895,0.296),
c("GO:0007422","peripheral nervous system development", 0.415, 4.681,-5.337, 1.863,-2.4009,0.893,0.306),
c("GO:0006888","ER to Golgi vesicle-mediated transport", 1.050,-0.661, 7.213, 2.262,-1.4160,0.904,0.313),
c("GO:2000739","regulation of mesenchymal stem cell differentiation", 0.035, 6.692,-4.603, 0.845,-1.5474,0.863,0.314),
c("GO:1901019","regulation of calcium ion transmembrane transporter activity", 0.450,-2.696, 6.207, 1.898,-2.2323,0.812,0.315),
c("GO:0038180","nerve growth factor signaling pathway", 0.046, 3.733, 3.911, 0.954,-1.3546,0.915,0.319),
c("GO:1902959","regulation of aspartic-type endopeptidase activity involved in amyloid precursor protein catabolic process", 0.012,-7.429, 0.157, 0.477,-1.4400,0.855,0.327),
c("GO:1901985","positive regulation of protein acetylation", 0.202,-6.524,-3.724, 1.556,-1.5474,0.820,0.329),
c("GO:0007010","cytoskeleton organization", 6.861,-1.080, 3.788, 3.076,-1.8129,0.897,0.331),
c("GO:2001014","regulation of skeletal muscle cell differentiation", 0.098, 5.974,-3.875, 1.255,-1.3951,0.836,0.339),
c("GO:0042552","myelination", 0.629, 4.890,-5.707, 2.041,-1.9743,0.866,0.340),
c("GO:0045581","negative regulation of T cell differentiation", 0.196, 3.411,-6.091, 1.544,-1.3177,0.780,0.357),
c("GO:0032933","SREBP signaling pathway", 0.046,-2.967,-3.601, 0.954,-1.5474,0.804,0.366),
c("GO:0080135","regulation of cellular response to stress", 3.543, 3.821, 0.150, 2.789,-2.0351,0.878,0.366),
c("GO:0010968","regulation of microtubule nucleation", 0.029,-1.907, 2.957, 0.778,-1.5474,0.825,0.371),
c("GO:1990440","positive regulation of transcription from RNA polymerase II promoter in response to endoplasmic reticulum stress", 0.069,-3.203,-5.462, 1.114,-1.3177,0.818,0.380),
c("GO:0010628","positive regulation of gene expression", 9.833,-6.684,-3.599, 3.232,-1.4096,0.811,0.383),
c("GO:0022603","regulation of anatomical structure morphogenesis", 5.430, 5.901,-5.009, 2.974,-1.3407,0.834,0.389),
c("GO:0001561","fatty acid alpha-oxidation", 0.040,-0.888,-2.186, 0.903,-1.4400,0.901,0.398),
c("GO:2000637","positive regulation of gene silencing by miRNA", 0.115,-4.825,-5.260, 1.322,-2.4795,0.762,0.398),
c("GO:0050730","regulation of peptidyl-tyrosine phosphorylation", 1.287,-6.163,-4.149, 2.350,-1.3130,0.828,0.399),
c("GO:0050796","regulation of insulin secretion", 0.975, 1.954, 6.369, 2.230,-2.1049,0.788,0.402),
c("GO:0048523","negative regulation of cellular process",24.951,-1.480,-7.772, 3.636,-1.6465,0.835,0.411),
c("GO:0080134","regulation of response to stress", 7.657, 3.288,-0.160, 3.123,-1.3546,0.896,0.412),
c("GO:0042147","retrograde transport, endosome to Golgi", 0.496,-1.082, 7.116, 1.940,-1.3407,0.909,0.420),
c("GO:1903689","regulation of wound healing, spreading of epidermal cells", 0.040, 6.139,-0.544, 0.903,-1.5474,0.790,0.425),
c("GO:2000074","regulation of type B pancreatic cell development", 0.046, 6.285,-3.898, 0.954,-1.4400,0.827,0.426),
c("GO:0070243","regulation of thymocyte apoptotic process", 0.069,-0.548, 0.230, 1.114,-1.5474,0.918,0.430),
c("GO:0072332","intrinsic apoptotic signaling pathway by p53 class mediator", 0.473, 4.471, 2.854, 1.919,-1.8523,0.882,0.444),
c("GO:0072393","microtubule anchoring at microtubule organizing center", 0.046,-1.861, 4.512, 0.954,-1.3177,0.884,0.445),
c("GO:0043547","positive regulation of GTPase activity", 3.803,-5.457, 2.820, 2.820,-1.3468,0.908,0.454),
c("GO:0072657","protein localization to membrane", 2.954, 0.312, 6.669, 2.710,-2.3617,0.804,0.456),
c("GO:0045899","positive regulation of RNA polymerase II transcriptional preinitiation complex assembly", 0.058,-5.865,-1.161, 1.041,-1.4400,0.767,0.476),
c("GO:0000122","negative regulation of transcription from RNA polymerase II promoter", 4.380,-4.016,-6.222, 2.881,-2.0679,0.759,0.482),
c("GO:0048710","regulation of astrocyte differentiation", 0.144, 5.636,-4.461, 1.415,-1.3546,0.801,0.490),
c("GO:0035458","cellular response to interferon-beta", 0.087, 4.849, 3.646, 1.204,-1.3177,0.956,0.493),
c("GO:0051865","protein autoubiquitination", 0.317,-5.247,-2.714, 1.748,-1.7469,0.918,0.500));

one.data <- data.frame(revigo.data);
names(one.data) <- revigo.names;
one.data <- one.data [(one.data$plot_X != "null" & one.data$plot_Y != "null"), ];
one.data$plot_X <- as.numeric( as.character(one.data$plot_X) );
one.data$plot_Y <- as.numeric( as.character(one.data$plot_Y) );
one.data$plot_size <- as.numeric( as.character(one.data$plot_size) );
one.data$log10_p_value <- as.numeric( as.character(one.data$log10_p_value) );
one.data$frequency <- as.numeric( as.character(one.data$frequency) );
one.data$uniqueness <- as.numeric( as.character(one.data$uniqueness) );
one.data$dispensability <- as.numeric( as.character(one.data$dispensability) );
#head(one.data);


# --------------------------------------------------------------------------
# Names of the axes, sizes of the numbers and letters, names of the columns,
# etc. can be changed below

p1 <- ggplot( data = one.data );
p1 <- p1 + geom_point( aes( plot_X, plot_Y, colour = log10_p_value, size = plot_size), alpha = I(0.6) ) + scale_size_area();
p1 <- p1 + scale_colour_gradientn( colours = c("blue", "green", "yellow", "red"), limits = c( min(one.data$log10_p_value), 0) );
p1 <- p1 + geom_point( aes(plot_X, plot_Y, size = plot_size), shape = 21, fill = "transparent", colour = I (alpha ("black", 0.6) )) + scale_size_area();
p1 <- p1 + scale_size( range=c(5, 30)) + theme_bw(); # + scale_fill_gradientn(colours = heat_hcl(7), limits = c(-300, 0) );
ex <- one.data [ one.data$dispensability < 0.15, ]; 
p1 <- p1 + geom_text( data = ex, aes(plot_X, plot_Y, label = description), colour = I(alpha("black", 0.85)), size = 3 );
p1 <- p1 + labs (y = "semantic space x", x = "semantic space y");
p1 <- p1 + theme(legend.key = element_blank()) ;
one.x_range = max(one.data$plot_X) - min(one.data$plot_X);
one.y_range = max(one.data$plot_Y) - min(one.data$plot_Y);
p1 <- p1 + xlim(min(one.data$plot_X)-one.x_range/10,max(one.data$plot_X)+one.x_range/10);
p1 <- p1 + ylim(min(one.data$plot_Y)-one.y_range/10,max(one.data$plot_Y)+one.y_range/10);



# --------------------------------------------------------------------------
# Output the plot to screen

p1;
```

cut -f 2,4 K14K9_GO.filter_2  > lala
```{r}

# --------------------------------------------------------------------------
# Here is your data from REVIGO. Scroll down for plot configuration options.

revigo.names <- c("term_ID","description","frequency_%","plot_X","plot_Y","plot_size","log10_p_value","uniqueness","dispensability");
revigo.data <- rbind(c("GO:0043547","positive regulation of GTPase activity", 3.803, 2.747,-5.880, 2.820,-0.6448,0.820,0.000),
c("GO:1902850","microtubule cytoskeleton organization involved in mitosis", 0.583, 3.931,-0.981, 2.009,-0.6448,0.639,0.000),
c("GO:0006468","protein phosphorylation",10.854, 5.426, 3.645, 3.275,-0.6448,0.864,0.003),
c("GO:0048048","embryonic eye morphogenesis", 0.185, 1.175, 6.142, 1.519,-0.6448,0.791,0.009),
c("GO:0045921","positive regulation of exocytosis", 0.444,-6.544, 0.855, 1.892,-0.6448,0.691,0.033),
c("GO:0060070","canonical Wnt signaling pathway", 1.685,-2.250,-6.134, 2.467,-0.6448,0.758,0.042),
c("GO:0099515","actin filament-based transport", 0.035,-5.071, 5.300, 0.845,-0.6448,0.799,0.130),
c("GO:1903551","regulation of extracellular exosome assembly", 0.035, 5.600,-1.588, 0.845,-0.6448,0.790,0.201),
c("GO:1901185","negative regulation of ERBB signaling pathway", 0.300,-3.580,-4.939, 1.724,-0.6448,0.749,0.313),
c("GO:0030336","negative regulation of cell migration", 1.229,-5.657,-1.003, 2.330,-0.6448,0.674,0.338),
c("GO:0022603","regulation of anatomical structure morphogenesis", 5.430, 0.356, 6.644, 2.974,-0.6448,0.762,0.370),
c("GO:0010959","regulation of metal ion transport", 1.887,-5.736, 1.297, 2.516,-0.6448,0.693,0.429));

one.data <- data.frame(revigo.data);
names(one.data) <- revigo.names;
one.data <- one.data [(one.data$plot_X != "null" & one.data$plot_Y != "null"), ];
one.data$plot_X <- as.numeric( as.character(one.data$plot_X) );
one.data$plot_Y <- as.numeric( as.character(one.data$plot_Y) );
one.data$plot_size <- as.numeric( as.character(one.data$plot_size) );
one.data$log10_p_value <- as.numeric( as.character(one.data$log10_p_value) );
one.data$frequency <- as.numeric( as.character(one.data$frequency) );
one.data$uniqueness <- as.numeric( as.character(one.data$uniqueness) );
one.data$dispensability <- as.numeric( as.character(one.data$dispensability) );
#head(one.data);


# --------------------------------------------------------------------------
# Names of the axes, sizes of the numbers and letters, names of the columns,
# etc. can be changed below

p1 <- ggplot( data = one.data );
p1 <- p1 + geom_point( aes( plot_X, plot_Y, colour = log10_p_value, size = plot_size), alpha = I(0.6) ) + scale_size_area();
p1 <- p1 + scale_colour_gradientn( colours = c("blue", "green", "yellow", "red"), limits = c( min(one.data$log10_p_value), 0) );
p1 <- p1 + geom_point( aes(plot_X, plot_Y, size = plot_size), shape = 21, fill = "transparent", colour = I (alpha ("black", 0.6) )) + scale_size_area();
p1 <- p1 + scale_size( range=c(5, 30)) + theme_bw(); # + scale_fill_gradientn(colours = heat_hcl(7), limits = c(-300, 0) );
ex <- one.data [ one.data$dispensability < 0.3, ]; 
p1 <- p1 + geom_text_repel( data = ex, aes(plot_X, plot_Y, label = description), colour = I(alpha("black", 0.85)), size = 4 );
p1 <- p1 + labs (y = "semantic space x", x = "semantic space y");
p1 <- p1 + theme(legend.key = element_blank()) ;
one.x_range = max(one.data$plot_X) - min(one.data$plot_X);
one.y_range = max(one.data$plot_Y) - min(one.data$plot_Y);
p1 <- p1 + xlim(min(one.data$plot_X)-one.x_range/10,max(one.data$plot_X)+one.x_range/10);
p1 <- p1 + ylim(min(one.data$plot_Y)-one.y_range/10,max(one.data$plot_Y)+one.y_range/10);



# --------------------------------------------------------------------------
# Output the plot to screen

p1;


```
K95mC excluded
```{r}
cut -f 2,4 K9me3_5mC_E_GO.filter_2  > lala




# --------------------------------------------------------------------------
# If you don't have the ggplot2 package installed, uncomment the following line:
# install.packages( "ggplot2" );
library( ggplot2 );
# --------------------------------------------------------------------------
# If you don't have the scales package installed, uncomment the following line:
# install.packages( "scales" );
library( scales );


# --------------------------------------------------------------------------
# Here is your data from REVIGO. Scroll down for plot configuration options.

revigo.names <- c("term_ID","description","frequency_%","plot_X","plot_Y","plot_size","log10_p_value","uniqueness","dispensability");
revigo.data <- rbind(c("GO:0048814","regulation of dendrite morphogenesis", 0.433, 3.910,-3.866, 1.881,-0.9618,0.404,0.000),
c("GO:2000637","positive regulation of gene silencing by miRNA", 0.115,-5.391,-1.965, 1.322,-0.7878,0.708,0.033),
c("GO:0051282","regulation of sequestering of calcium ion", 0.600,-1.690, 5.429, 2.021,-0.9618,0.723,0.038),
c("GO:0034394","protein localization to cell surface", 0.306, 2.086, 5.307, 1.732,-0.7878,0.790,0.120),
c("GO:0090400","stress-induced premature senescence", 0.046, 4.798, 1.188, 0.954,-0.9618,0.741,0.172),
c("GO:0006511","ubiquitin-dependent protein catabolic process", 3.306,-3.947,-6.144, 2.759,-0.7878,0.754,0.176),
c("GO:0043068","positive regulation of programmed cell death", 3.428,-5.822, 2.146, 2.775,-0.7878,0.770,0.183),
c("GO:0010882","regulation of cardiac muscle contraction by calcium ion signaling", 0.127, 6.032,-2.984, 1.362,-0.7878,0.615,0.263),
c("GO:0007422","peripheral nervous system development", 0.415, 2.433,-6.508, 1.863,-0.7878,0.653,0.329),
c("GO:0035308","negative regulation of protein dephosphorylation", 0.133,-5.294,-3.723, 1.380,-0.7878,0.758,0.398));

one.data <- data.frame(revigo.data);
names(one.data) <- revigo.names;
one.data <- one.data [(one.data$plot_X != "null" & one.data$plot_Y != "null"), ];
one.data$plot_X <- as.numeric( as.character(one.data$plot_X) );
one.data$plot_Y <- as.numeric( as.character(one.data$plot_Y) );
one.data$plot_size <- as.numeric( as.character(one.data$plot_size) );
one.data$log10_p_value <- as.numeric( as.character(one.data$log10_p_value) );
one.data$frequency <- as.numeric( as.character(one.data$frequency) );
one.data$uniqueness <- as.numeric( as.character(one.data$uniqueness) );
one.data$dispensability <- as.numeric( as.character(one.data$dispensability) );
#head(one.data);


# --------------------------------------------------------------------------
# Names of the axes, sizes of the numbers and letters, names of the columns,
# etc. can be changed below

p1 <- ggplot( data = one.data );
p1 <- p1 + geom_point( aes( plot_X, plot_Y, colour = log10_p_value, size = plot_size), alpha = I(0.6) ) + scale_size_area();
p1 <- p1 + scale_colour_gradientn( colours = c("blue", "green", "yellow", "red"), limits = c( min(one.data$log10_p_value), 0) );
p1 <- p1 + geom_point( aes(plot_X, plot_Y, size = plot_size), shape = 21, fill = "transparent", colour = I (alpha ("black", 0.6) )) + scale_size_area();
p1 <- p1 + scale_size( range=c(5, 30)) + theme_bw(); # + scale_fill_gradientn(colours = heat_hcl(7), limits = c(-300, 0) );
ex <- one.data [ one.data$dispensability < 0.3, ]; 
p1 <- p1 + geom_text_repel( data = ex, aes(plot_X, plot_Y, label = description), colour = I(alpha("black", 0.85)), size = 4 );
p1 <- p1 + labs (y = "semantic space x", x = "semantic space y");
p1 <- p1 + theme(legend.key = element_blank()) ;
one.x_range = max(one.data$plot_X) - min(one.data$plot_X);
one.y_range = max(one.data$plot_Y) - min(one.data$plot_Y);
p1 <- p1 + xlim(min(one.data$plot_X)-one.x_range/10,max(one.data$plot_X)+one.x_range/10);
p1 <- p1 + ylim(min(one.data$plot_Y)-one.y_range/10,max(one.data$plot_Y)+one.y_range/10);



# --------------------------------------------------------------------------
# Output the plot to screen

p1;

```
 cut -f 2,4 K4K27_GO.filter_2  > lala
```{r}

# --------------------------------------------------------------------------
# Here is your data from REVIGO. Scroll down for plot configuration options.

revigo.names <- c("term_ID","description","frequency_%","plot_X","plot_Y","plot_size","log10_p_value","uniqueness","dispensability");
revigo.data <- rbind(c("GO:0007157","heterophilic cell-cell adhesion via plasma membrane cell adhesion molecules", 0.265,-6.940, 0.581, 1.672,-0.9060,0.879,0.000),
c("GO:0060973","cell migration involved in heart development", 0.081,-1.549, 6.108, 1.176,-0.9060,0.738,0.000),
c("GO:0002029","desensitization of G-protein coupled receptor protein signaling pathway", 0.092, 5.782, 0.537, 1.230,-0.9060,0.699,0.008),
c("GO:0042181","ketone biosynthetic process", 0.225,-3.892, 0.288, 1.602,-0.9060,0.857,0.026),
c("GO:0097400","interleukin-17-mediated signaling pathway", 0.006, 4.553,-5.625, 0.301,-0.9060,0.681,0.086),
c("GO:2000427","positive regulation of apoptotic cell clearance", 0.046,-2.318,-5.854, 0.954,-0.9060,0.713,0.088),
c("GO:0098700","neurotransmitter loading into synaptic vesicle", 0.040,-3.561,-4.319, 0.903,-0.9060,0.800,0.174),
c("GO:0030252","growth hormone secretion", 0.092,-0.655,-5.427, 1.230,-0.9060,0.738,0.184),
c("GO:0045637","regulation of myeloid cell differentiation", 1.010,-3.172, 5.295, 2.246,-0.9060,0.721,0.232),
c("GO:0017015","regulation of transforming growth factor beta receptor signaling pathway", 0.583, 5.322,-3.498, 2.009,-0.9060,0.725,0.255),
c("GO:0038042","dimeric G-protein coupled receptor signaling pathway", 0.006, 5.011, 3.559, 0.301,-0.9060,0.734,0.289),
c("GO:0097647","amylin receptor signaling pathway", 0.006, 6.317, 2.982, 0.301,-0.9060,0.734,0.289),
c("GO:0097646","calcitonin family receptor signaling pathway", 0.138, 5.528, 1.836, 1.380,-0.9060,0.710,0.357),
c("GO:0060055","angiogenesis involved in wound healing", 0.115,-0.672, 6.591, 1.322,-0.9060,0.755,0.399));

one.data <- data.frame(revigo.data);
names(one.data) <- revigo.names;
one.data <- one.data [(one.data$plot_X != "null" & one.data$plot_Y != "null"), ];
one.data$plot_X <- as.numeric( as.character(one.data$plot_X) );
one.data$plot_Y <- as.numeric( as.character(one.data$plot_Y) );
one.data$plot_size <- as.numeric( as.character(one.data$plot_size) );
one.data$log10_p_value <- as.numeric( as.character(one.data$log10_p_value) );
one.data$frequency <- as.numeric( as.character(one.data$frequency) );
one.data$uniqueness <- as.numeric( as.character(one.data$uniqueness) );
one.data$dispensability <- as.numeric( as.character(one.data$dispensability) );
#head(one.data);


# --------------------------------------------------------------------------
# Names of the axes, sizes of the numbers and letters, names of the columns,
# etc. can be changed below

p1 <- ggplot( data = one.data );
p1 <- p1 + geom_point( aes( plot_X, plot_Y, colour = log10_p_value, size = plot_size), alpha = I(0.6) ) + scale_size_area();
p1 <- p1 + scale_colour_gradientn( colours = c("blue", "green", "yellow", "red"), limits = c( min(one.data$log10_p_value), 0) );
p1 <- p1 + geom_point( aes(plot_X, plot_Y, size = plot_size), shape = 21, fill = "transparent", colour = I (alpha ("black", 0.6) )) + scale_size_area();
p1 <- p1 + scale_size( range=c(5, 30)) + theme_bw(); # + scale_fill_gradientn(colours = heat_hcl(7), limits = c(-300, 0) );
ex <- one.data [ one.data$dispensability < 0.3, ]; 
p1 <- p1 + geom_text_repel( data = ex, aes(plot_X, plot_Y, label = description), colour = I(alpha("black", 0.85)), size = 4 );
p1 <- p1 + labs (y = "semantic space x", x = "semantic space y");
p1 <- p1 + theme(legend.key = element_blank()) ;
one.x_range = max(one.data$plot_X) - min(one.data$plot_X);
one.y_range = max(one.data$plot_Y) - min(one.data$plot_Y);
p1 <- p1 + xlim(min(one.data$plot_X)-one.x_range/10,max(one.data$plot_X)+one.x_range/10);
p1 <- p1 + ylim(min(one.data$plot_Y)-one.y_range/10,max(one.data$plot_Y)+one.y_range/10);



# --------------------------------------------------------------------------
# Output the plot to screen

p1;


```

K20 K91 included revigo
```{r}

# --------------------------------------------------------------------------
# If you don't have the ggplot2 package installed, uncomment the following line:
# install.packages( "ggplot2" );
library( ggplot2 );
# --------------------------------------------------------------------------
# If you don't have the scales package installed, uncomment the following line:
# install.packages( "scales" );
library( scales );


# --------------------------------------------------------------------------
# Here is your data from REVIGO. Scroll down for plot configuration options.

revigo.names <- c("term_ID","description","frequency_%","plot_X","plot_Y","plot_size","log10_p_value","uniqueness","dispensability");
revigo.data <- rbind(c("GO:0051017","actin filament bundle assembly", 0.045, 1.000, 0.829, 3.762,-0.4202,0.613,0.000));

one.data <- data.frame(revigo.data);
names(one.data) <- revigo.names;
one.data <- one.data [(one.data$plot_X != "null" & one.data$plot_Y != "null"), ];
one.data$plot_X <- as.numeric( as.character(one.data$plot_X) );
one.data$plot_Y <- as.numeric( as.character(one.data$plot_Y) );
one.data$plot_size <- as.numeric( as.character(one.data$plot_size) );
one.data$log10_p_value <- as.numeric( as.character(one.data$log10_p_value) );
one.data$frequency <- as.numeric( as.character(one.data$frequency) );
one.data$uniqueness <- as.numeric( as.character(one.data$uniqueness) );
one.data$dispensability <- as.numeric( as.character(one.data$dispensability) );
#head(one.data);


# --------------------------------------------------------------------------
# Names of the axes, sizes of the numbers and letters, names of the columns,
# etc. can be changed below

p1 <- ggplot( data = one.data );
p1 <- p1 + geom_point( aes( plot_X, plot_Y, colour = log10_p_value, size = plot_size), alpha = I(0.6) ) + scale_size_area();
p1 <- p1 + scale_colour_gradientn( colours = c("blue", "green", "yellow", "red"), limits = c( min(one.data$log10_p_value), 0) );
p1 <- p1 + geom_point( aes(plot_X, plot_Y, size = plot_size), shape = 21, fill = "transparent", colour = I (alpha ("black", 0.6) )) + scale_size_area();
p1 <- p1 + scale_size( range=c(5, 30)) + theme_bw(); # + scale_fill_gradientn(colours = heat_hcl(7), limits = c(-300, 0) );
ex <- one.data [ one.data$dispensability < 0.15, ]; 
p1 <- p1 + geom_text( data = ex, aes(plot_X, plot_Y, label = description), colour = I(alpha("black", 0.85)), size = 3 );
p1 <- p1 + labs (y = "semantic space x", x = "semantic space y");
p1 <- p1 + theme(legend.key = element_blank()) ;
one.x_range = max(one.data$plot_X) - min(one.data$plot_X);
one.y_range = max(one.data$plot_Y) - min(one.data$plot_Y);
p1 <- p1 + xlim(min(one.data$plot_X)-one.x_range/10,max(one.data$plot_X)+one.x_range/10);
p1 <- p1 + ylim(min(one.data$plot_Y)-one.y_range/10,max(one.data$plot_Y)+one.y_range/10);



# --------------------------------------------------------------------------
# Output the plot to screen

p1;

##enrichr

# --------------------------------------------------------------------------
# If you don't have the ggplot2 package installed, uncomment the following line:
# install.packages( "ggplot2" );
library( ggplot2 );
# --------------------------------------------------------------------------
# If you don't have the scales package installed, uncomment the following line:
# install.packages( "scales" );
library( scales );


# --------------------------------------------------------------------------
# Here is your data from REVIGO. Scroll down for plot configuration options.

revigo.names <- c("term_ID","description","frequency_%","plot_X","plot_Y","plot_size","log10_p_value","uniqueness","dispensability");
revigo.data <- rbind(c("GO:0007160","cell-matrix adhesion", 1.148,-0.119,-0.706, 2.301,-2.7381,0.966,0.000),
c("GO:0046578","regulation of Ras protein signal transduction", 1.200,-7.081,-0.928, 2.320,-4.8697,0.849,0.000),
c("GO:0019079","viral genome replication", 0.589,-2.717, 1.277, 2.013,-2.1229,0.953,0.002),
c("GO:0010498","proteasomal protein catabolic process", 2.383,-1.421,-5.724, 2.617,-3.3640,0.915,0.002),
c("GO:0031032","actomyosin structure organization", 0.975, 5.643, 4.692, 2.230,-3.1167,0.850,0.012),
c("GO:0042117","monocyte activation", 0.046,-3.348, 2.123, 0.954,-1.2997,0.966,0.028),
c("GO:0070344","regulation of fat cell proliferation", 0.063, 0.066, 0.670, 1.079,-1.4064,0.951,0.034),
c("GO:0043547","positive regulation of GTPase activity", 3.803,-1.219,-2.964, 2.820,-2.3015,0.925,0.034),
c("GO:2000021","regulation of ion homeostasis", 1.131,-5.835, 3.036, 2.294,-2.0867,0.905,0.037),
c("GO:0051924","regulation of calcium ion transport", 1.206,-1.349, 7.098, 2.322,-3.6264,0.819,0.038),
c("GO:0090342","regulation of cell aging", 0.208, 3.696, 1.022, 1.568,-2.5525,0.904,0.038),
c("GO:0046856","phosphatidylinositol dephosphorylation", 0.138, 3.612,-5.745, 1.398,-2.6887,0.873,0.041),
c("GO:0010564","regulation of cell cycle process", 3.393,-1.328, 0.938, 2.770,-1.8170,0.898,0.051),
c("GO:0043153","entrainment of circadian clock by photoperiod", 0.110,-3.052, 4.001, 1.301,-2.2006,0.937,0.064),
c("GO:0008156","negative regulation of DNA replication", 0.208,-6.096,-3.763, 1.568,-2.3828,0.824,0.074),
c("GO:0042060","wound healing", 3.058,-4.599, 1.182, 2.725,-1.4653,0.947,0.091),
c("GO:0006337","nucleosome disassembly", 0.104, 6.077, 1.678, 1.279,-2.2424,0.903,0.115),
c("GO:0003230","cardiac atrium development", 0.202, 4.192,-0.153, 1.556,-1.3496,0.943,0.118),
c("GO:0045815","positive regulation of gene expression, epigenetic", 0.317,-4.307,-5.283, 1.748,-1.5137,0.891,0.121),
c("GO:0031529","ruffle organization", 0.248, 4.944, 3.905, 1.643,-2.3828,0.911,0.125),
c("GO:0045216","cell-cell junction organization", 1.293, 2.592, 3.708, 2.352,-2.0758,0.906,0.148),
c("GO:0002903","negative regulation of B cell apoptotic process", 0.069,-6.653, 1.434, 1.114,-1.3496,0.895,0.155),
c("GO:0008333","endosome to lysosome transport", 0.254,-0.290, 6.411, 1.653,-1.4972,0.911,0.168),
c("GO:0005979","regulation of glycogen biosynthetic process", 0.162, 2.452,-6.843, 1.462,-2.2006,0.837,0.178),
c("GO:0031323","regulation of cellular metabolic process",33.837,-4.421,-5.976, 3.768,-1.6215,0.858,0.194),
c("GO:0008643","carbohydrate transport", 1.091, 0.086, 7.716, 2.279,-1.4064,0.900,0.195),
c("GO:0006084","acetyl-CoA metabolic process", 0.196, 2.362,-5.432, 1.544,-2.4919,0.934,0.213),
c("GO:0048194","Golgi vesicle budding", 0.052, 2.659, 6.378, 1.000,-2.8518,0.878,0.217),
c("GO:0035810","positive regulation of urine volume", 0.075,-5.324, 2.226, 1.146,-1.3496,0.917,0.219),
c("GO:0061668","mitochondrial ribosome assembly", 0.023, 6.543, 2.561, 0.699,-1.4064,0.923,0.219),
c("GO:0030575","nuclear body organization", 0.063, 4.747, 5.585, 1.079,-1.3496,0.934,0.220),
c("GO:0097499","protein localization to non-motile cilium", 0.058,-0.993, 5.723, 1.041,-1.4064,0.920,0.241),
c("GO:0018107","peptidyl-threonine phosphorylation", 0.635, 0.206,-6.834, 2.045,-2.1432,0.901,0.241),
c("GO:0048523","negative regulation of cellular process",24.951,-8.297, 0.550, 3.636,-1.4745,0.840,0.254),
c("GO:0002831","regulation of response to biotic stimulus", 0.802,-6.538,-1.832, 2.146,-1.3496,0.903,0.260),
c("GO:0016192","vesicle-mediated transport",11.385,-0.071, 7.404, 3.295,-3.2363,0.905,0.264),
c("GO:0007517","muscle organ development", 2.083, 3.856,-0.464, 2.559,-1.3242,0.932,0.270),
c("GO:0045879","negative regulation of smoothened signaling pathway", 0.144,-7.886,-0.528, 1.415,-2.2424,0.830,0.282),
c("GO:1900120","regulation of receptor binding", 0.121,-1.685,-2.946, 1.342,-1.2997,0.948,0.291),
c("GO:0006101","citrate metabolic process", 0.202, 5.296,-5.350, 1.556,-1.4064,0.909,0.294),
c("GO:0031397","negative regulation of protein ubiquitination", 0.785,-4.829,-4.369, 2.137,-1.4653,0.818,0.316),
c("GO:0016311","dephosphorylation", 2.418, 2.760,-5.335, 2.623,-1.4594,0.927,0.318),
c("GO:0008277","regulation of G-protein coupled receptor protein signaling pathway", 0.733,-7.588,-0.918, 2.107,-2.0437,0.872,0.331),
c("GO:0048193","Golgi vesicle transport", 2.008, 0.457, 6.857, 2.543,-2.3282,0.913,0.336),
c("GO:0030330","DNA damage response, signal transduction by p53 class mediator", 0.635,-6.032, 0.085, 2.045,-1.9241,0.876,0.339),
c("GO:0009221","pyrimidine deoxyribonucleotide biosynthetic process", 0.035, 3.282,-6.322, 0.845,-1.4064,0.875,0.343),
c("GO:0033043","regulation of organelle organization", 6.082, 5.923, 3.585, 3.023,-1.5611,0.842,0.356),
c("GO:0043951","negative regulation of cAMP-mediated signaling", 0.040,-7.743,-0.016, 0.903,-1.4064,0.840,0.357),
c("GO:0015884","folic acid transport", 0.046,-0.490, 7.239, 0.954,-1.3496,0.898,0.359),
c("GO:0030033","microvillus assembly", 0.127, 3.672, 3.539, 1.362,-1.2997,0.901,0.368),
c("GO:0060292","long term synaptic depression", 0.133,-7.532, 0.621, 1.380,-1.4064,0.822,0.374),
c("GO:0006695","cholesterol biosynthetic process", 0.294, 4.637,-5.715, 1.716,-1.7476,0.869,0.375),
c("GO:0032528","microvillus organization", 0.173, 4.435, 4.133, 1.491,-1.2997,0.913,0.377),
c("GO:0050767","regulation of neurogenesis", 3.855, 4.738,-0.548, 2.825,-2.3535,0.842,0.377),
c("GO:1902959","regulation of aspartic-type endopeptidase activity involved in amyloid precursor protein catabolic process", 0.012,-2.649,-6.262, 0.477,-1.2997,0.898,0.378),
c("GO:2000146","negative regulation of cell motility", 1.310,-4.916, 5.230, 2.358,-1.7318,0.768,0.379),
c("GO:0030213","hyaluronan biosynthetic process", 0.075, 0.126,-4.958, 1.146,-1.3496,0.946,0.392),
c("GO:0046627","negative regulation of insulin receptor signaling pathway", 0.173,-7.949,-0.212, 1.491,-2.0190,0.805,0.397),
c("GO:0006107","oxaloacetate metabolic process", 0.069, 5.494,-5.143, 1.114,-1.3496,0.915,0.405),
c("GO:0002029","desensitization of G-protein coupled receptor protein signaling pathway", 0.092,-7.819,-0.519, 1.230,-1.2997,0.839,0.407),
c("GO:0072659","protein localization to plasma membrane", 1.287, 1.668, 6.609, 2.350,-2.1716,0.825,0.407),
c("GO:0098751","bone cell development", 0.190, 3.218,-0.094, 1.531,-1.3496,0.916,0.412),
c("GO:0001843","neural tube closure", 0.496, 3.624,-0.720, 1.940,-1.7947,0.910,0.419),
c("GO:0006720","isoprenoid metabolic process", 0.733, 5.132,-4.503, 2.107,-1.2997,0.897,0.451),
c("GO:0043648","dicarboxylic acid metabolic process", 0.571, 5.048,-5.646, 2.000,-1.3371,0.902,0.480),
c("GO:0035065","regulation of histone acetylation", 0.312, 1.589,-3.292, 1.740,-1.9871,0.803,0.483),
c("GO:0070316","regulation of G0 to G1 transition", 0.046,-1.243, 1.280, 0.954,-1.6612,0.919,0.483),
c("GO:0006468","protein phosphorylation",10.854, 0.466,-6.308, 3.275,-1.7371,0.882,0.487),
c("GO:0050708","regulation of protein secretion", 2.291,-0.406, 7.132, 2.600,-2.4680,0.789,0.492),
c("GO:0018210","peptidyl-threonine modification", 0.658,-1.296,-6.780, 2.061,-1.8298,0.929,0.493));

one.data <- data.frame(revigo.data);
names(one.data) <- revigo.names;
one.data <- one.data [(one.data$plot_X != "null" & one.data$plot_Y != "null"), ];
one.data$plot_X <- as.numeric( as.character(one.data$plot_X) );
one.data$plot_Y <- as.numeric( as.character(one.data$plot_Y) );
one.data$plot_size <- as.numeric( as.character(one.data$plot_size) );
one.data$log10_p_value <- as.numeric( as.character(one.data$log10_p_value) );
one.data$frequency <- as.numeric( as.character(one.data$frequency) );
one.data$uniqueness <- as.numeric( as.character(one.data$uniqueness) );
one.data$dispensability <- as.numeric( as.character(one.data$dispensability) );
#head(one.data);


# --------------------------------------------------------------------------
# Names of the axes, sizes of the numbers and letters, names of the columns,
# etc. can be changed below

p1 <- ggplot( data = one.data );
p1 <- p1 + geom_point( aes( plot_X, plot_Y, colour = log10_p_value, size = plot_size), alpha = I(0.6) ) + scale_size_area();
p1 <- p1 + scale_colour_gradientn( colours = c("blue", "green", "yellow", "red"), limits = c( min(one.data$log10_p_value), 0) );
p1 <- p1 + geom_point( aes(plot_X, plot_Y, size = plot_size), shape = 21, fill = "transparent", colour = I (alpha ("black", 0.6) )) + scale_size_area();
p1 <- p1 + scale_size( range=c(5, 30)) + theme_bw(); # + scale_fill_gradientn(colours = heat_hcl(7), limits = c(-300, 0) );
ex <- one.data [ one.data$dispensability < 0.15, ]; 
p1 <- p1 + geom_text( data = ex, aes(plot_X, plot_Y, label = description), colour = I(alpha("black", 0.85)), size = 3 );
p1 <- p1 + labs (y = "semantic space x", x = "semantic space y");
p1 <- p1 + theme(legend.key = element_blank()) ;
one.x_range = max(one.data$plot_X) - min(one.data$plot_X);
one.y_range = max(one.data$plot_Y) - min(one.data$plot_Y);
p1 <- p1 + xlim(min(one.data$plot_X)-one.x_range/10,max(one.data$plot_X)+one.x_range/10);
p1 <- p1 + ylim(min(one.data$plot_Y)-one.y_range/10,max(one.data$plot_Y)+one.y_range/10);



# --------------------------------------------------------------------------
# Output the plot to screen

p1;

###filter go enrichr
#cut -f 2,3 /home/eneritz/ASCHROM0818/GO_terms_supfig4/K20_K91_GO.filter > lala

# --------------------------------------------------------------------------
# Here is your data from REVIGO. Scroll down for plot configuration options.

revigo.names <- c("term_ID","description","frequency_%","plot_X","plot_Y","plot_size","log10_p_value","uniqueness","dispensability");
revigo.data <- rbind(c("GO:0007160","cell-matrix adhesion", 1.148,-1.071, 1.149, 2.301,-2.7381,0.962,0.000),
c("GO:0046578","regulation of Ras protein signal transduction", 1.200,-4.926, 5.041, 2.320,-4.8697,0.844,0.000),
c("GO:0019079","viral genome replication", 0.589, 1.047, 5.215, 2.013,-2.1229,0.954,0.002),
c("GO:0010498","proteasomal protein catabolic process", 2.383,-6.498,-2.167, 2.617,-3.3640,0.916,0.002),
c("GO:0031032","actomyosin structure organization", 0.975, 6.612,-2.425, 2.230,-3.1167,0.847,0.012),
c("GO:0070344","regulation of fat cell proliferation", 0.063,-0.126, 2.760, 1.079,-1.4064,0.949,0.034),
c("GO:0043547","positive regulation of GTPase activity", 3.803, 1.984, 1.789, 2.820,-2.3015,0.925,0.034),
c("GO:2000021","regulation of ion homeostasis", 1.131,-0.389, 7.932, 2.294,-2.0867,0.899,0.037),
c("GO:0051924","regulation of calcium ion transport", 1.206, 4.626, 5.546, 2.322,-3.6264,0.814,0.038),
c("GO:0090342","regulation of cell aging", 0.208, 3.516,-3.681, 1.568,-2.5525,0.913,0.038),
c("GO:0046856","phosphatidylinositol dephosphorylation", 0.138,-2.456,-6.383, 1.398,-2.6887,0.874,0.041),
c("GO:0010564","regulation of cell cycle process", 3.393,-1.842,-0.557, 2.770,-1.8170,0.899,0.051),
c("GO:0043153","entrainment of circadian clock by photoperiod", 0.110, 1.364, 7.075, 1.301,-2.2006,0.939,0.064),
c("GO:0008156","negative regulation of DNA replication", 0.208,-6.378, 1.815, 1.568,-2.3828,0.822,0.074),
c("GO:0042060","wound healing", 3.058,-1.275, 4.751, 2.725,-1.4653,0.949,0.091),
c("GO:0006337","nucleosome disassembly", 0.104, 5.075,-5.008, 1.279,-2.2424,0.910,0.115),
c("GO:0003230","cardiac atrium development", 0.202, 2.405,-3.641, 1.556,-1.3496,0.946,0.118),
c("GO:0045815","positive regulation of gene expression, epigenetic", 0.317,-7.326,-0.122, 1.748,-1.5137,0.886,0.121),
c("GO:0031529","ruffle organization", 0.248, 6.014,-0.493, 1.643,-2.3828,0.922,0.125),
c("GO:0045216","cell-cell junction organization", 1.293, 5.474,-1.448, 2.352,-2.0758,0.906,0.148),
c("GO:0002903","negative regulation of B cell apoptotic process", 0.069,-3.684, 3.540, 1.114,-1.3496,0.894,0.155),
c("GO:0008333","endosome to lysosome transport", 0.254, 4.986, 3.498, 1.653,-1.4972,0.912,0.168),
c("GO:0005979","regulation of glycogen biosynthetic process", 0.162,-1.850,-5.427, 1.462,-2.2006,0.833,0.178),
c("GO:0031323","regulation of cellular metabolic process",33.837,-7.335,-0.231, 3.768,-1.6215,0.852,0.194),
c("GO:0008643","carbohydrate transport", 1.091, 6.120, 4.020, 2.279,-1.4064,0.896,0.195),
c("GO:0006084","acetyl-CoA metabolic process", 0.196,-4.203,-5.768, 1.544,-2.4919,0.930,0.213),
c("GO:0048194","Golgi vesicle budding", 0.052, 6.734, 0.914, 1.000,-2.8518,0.876,0.217),
c("GO:0035810","positive regulation of urine volume", 0.075,-0.633, 7.383, 1.146,-1.3496,0.919,0.219),
c("GO:0061668","mitochondrial ribosome assembly", 0.023, 5.890,-4.035, 0.699,-1.4064,0.925,0.219),
c("GO:0030575","nuclear body organization", 0.063, 7.151,-2.011, 1.079,-1.3496,0.933,0.220),
c("GO:0097499","protein localization to non-motile cilium", 0.058, 3.936, 4.049, 1.041,-1.4064,0.916,0.241),
c("GO:0018107","peptidyl-threonine phosphorylation", 0.635,-5.423,-4.154, 2.045,-2.1432,0.898,0.241),
c("GO:0048523","negative regulation of cellular process",24.951,-6.124, 4.097, 3.636,-1.4745,0.836,0.254),
c("GO:0002831","regulation of response to biotic stimulus", 0.802,-5.867, 3.508, 2.146,-1.3496,0.899,0.260),
c("GO:0016192","vesicle-mediated transport",11.385, 5.610, 4.140, 3.295,-3.2363,0.901,0.264),
c("GO:0007517","muscle organ development", 2.083, 2.005,-4.185, 2.559,-1.3242,0.936,0.270),
c("GO:0045879","negative regulation of smoothened signaling pathway", 0.144,-4.873, 5.209, 1.415,-2.2424,0.825,0.282),
c("GO:0006101","citrate metabolic process", 0.202, 0.206,-7.019, 1.556,-1.4064,0.909,0.294),
c("GO:2000465","regulation of glycogen (starch) synthase activity", 0.046, 1.889, 1.153, 0.954,-1.3496,0.950,0.309),
c("GO:0031397","negative regulation of protein ubiquitination", 0.785,-5.937, 0.608, 2.137,-1.4653,0.815,0.316),
c("GO:0016311","dephosphorylation", 2.418,-4.552,-5.602, 2.623,-1.4594,0.923,0.318),
c("GO:0008277","regulation of G-protein coupled receptor protein signaling pathway", 0.733,-5.230, 5.620, 2.107,-2.0437,0.869,0.331),
c("GO:0048193","Golgi vesicle transport", 2.008, 6.214, 3.140, 2.543,-2.3282,0.910,0.336),
c("GO:0030330","DNA damage response, signal transduction by p53 class mediator", 0.635,-2.827, 5.555, 2.045,-1.9241,0.881,0.339),
c("GO:0009221","pyrimidine deoxyribonucleotide biosynthetic process", 0.035,-2.431,-6.017, 0.845,-1.4064,0.873,0.343),
c("GO:0033043","regulation of organelle organization", 6.082, 6.115,-2.477, 3.023,-1.5611,0.840,0.356),
c("GO:0043951","negative regulation of cAMP-mediated signaling", 0.040,-4.199, 5.768, 0.903,-1.4064,0.836,0.357),
c("GO:0015884","folic acid transport", 0.046, 5.535, 5.215, 0.954,-1.3496,0.893,0.359),
c("GO:0060292","long term synaptic depression", 0.133,-3.550, 6.359, 1.380,-1.4064,0.819,0.374),
c("GO:0006695","cholesterol biosynthetic process", 0.294,-0.729,-6.791, 1.716,-1.7476,0.873,0.375),
c("GO:0050767","regulation of neurogenesis", 3.855, 2.628,-4.612, 2.825,-2.3535,0.851,0.377),
c("GO:2000146","negative regulation of cell motility", 1.310, 1.791, 6.126, 2.358,-1.7318,0.771,0.379),
c("GO:0030213","hyaluronan biosynthetic process", 0.075,-4.833,-2.538, 1.146,-1.3496,0.943,0.392),
c("GO:0046627","negative regulation of insulin receptor signaling pathway", 0.173,-4.623, 5.544, 1.491,-2.0190,0.799,0.397),
c("GO:0006107","oxaloacetate metabolic process", 0.069, 0.549,-7.046, 1.114,-1.3496,0.915,0.405),
c("GO:0072659","protein localization to plasma membrane", 1.287, 5.896, 2.099, 2.350,-2.1716,0.823,0.407),
c("GO:0098751","bone cell development", 0.190, 1.944,-3.184, 1.531,-1.3496,0.922,0.412),
c("GO:0001843","neural tube closure", 0.496, 1.086,-3.653, 1.940,-1.7947,0.914,0.419),
c("GO:0043648","dicarboxylic acid metabolic process", 0.571,-0.107,-7.007, 2.000,-1.3371,0.902,0.480),
c("GO:0035065","regulation of histone acetylation", 0.312,-3.642,-2.470, 1.740,-1.9871,0.802,0.483),
c("GO:0070316","regulation of G0 to G1 transition", 0.046,-2.137,-0.629, 0.954,-1.6612,0.920,0.483),
c("GO:0006468","protein phosphorylation",10.854,-5.224,-4.096, 3.275,-1.7371,0.877,0.487),
c("GO:0050708","regulation of protein secretion", 2.291, 4.941, 4.682, 2.600,-2.4680,0.783,0.492),
c("GO:0018210","peptidyl-threonine modification", 0.658,-6.527,-2.856, 2.061,-1.8298,0.928,0.493));

one.data <- data.frame(revigo.data);
names(one.data) <- revigo.names;
one.data <- one.data [(one.data$plot_X != "null" & one.data$plot_Y != "null"), ];
one.data$plot_X <- as.numeric( as.character(one.data$plot_X) );
one.data$plot_Y <- as.numeric( as.character(one.data$plot_Y) );
one.data$plot_size <- as.numeric( as.character(one.data$plot_size) );
one.data$log10_p_value <- as.numeric( as.character(one.data$log10_p_value) );
one.data$frequency <- as.numeric( as.character(one.data$frequency) );
one.data$uniqueness <- as.numeric( as.character(one.data$uniqueness) );
one.data$dispensability <- as.numeric( as.character(one.data$dispensability) );
#head(one.data);


# --------------------------------------------------------------------------
# Names of the axes, sizes of the numbers and letters, names of the columns,
# etc. can be changed below

p1 <- ggplot( data = one.data );
p1 <- p1 + geom_point( aes( plot_X, plot_Y, colour = log10_p_value, size = plot_size), alpha = I(0.6) ) + scale_size_area();
p1 <- p1 + scale_colour_gradientn( colours = c("blue", "green", "yellow", "red"), limits = c( min(one.data$log10_p_value), 0) );
p1 <- p1 + geom_point( aes(plot_X, plot_Y, size = plot_size), shape = 21, fill = "transparent", colour = I (alpha ("black", 0.6) )) + scale_size_area();
p1 <- p1 + scale_size( range=c(5, 30)) + theme_bw(); # + scale_fill_gradientn(colours = heat_hcl(7), limits = c(-300, 0) );
ex <- one.data [ one.data$dispensability < 0.15, ]; 
p1 <- p1 + geom_text( data = ex, aes(plot_X, plot_Y, label = description), colour = I(alpha("black", 0.85)), size = 3 );
p1 <- p1 + labs (y = "semantic space x", x = "semantic space y");
p1 <- p1 + theme(legend.key = element_blank()) ;
one.x_range = max(one.data$plot_X) - min(one.data$plot_X);
one.y_range = max(one.data$plot_Y) - min(one.data$plot_Y);
p1 <- p1 + xlim(min(one.data$plot_X)-one.x_range/10,max(one.data$plot_X)+one.x_range/10);
p1 <- p1 + ylim(min(one.data$plot_Y)-one.y_range/10,max(one.data$plot_Y)+one.y_range/10);



# --------------------------------------------------------------------------
# Output the plot to screen

p1;

###sig more filter
 cut -f 2,4 K20_K91_GO.filter_2  > lala



# --------------------------------------------------------------------------
# Here is your data from REVIGO. Scroll down for plot configuration options.

revigo.names <- c("term_ID","description","frequency_%","plot_X","plot_Y","plot_size","log10_p_value","uniqueness","dispensability");
revigo.data <- rbind(c("GO:0046578","regulation of Ras protein signal transduction", 1.200,-2.873, 6.191, 2.320,-1.8484,0.587,0.000),
c("GO:0010498","proteasomal protein catabolic process", 2.383, 3.208,-4.752, 2.617,-1.0498,0.744,0.002),
c("GO:0031032","actomyosin structure organization", 0.975, 5.797, 0.436, 2.230,-1.0498,0.627,0.012),
c("GO:0051345","positive regulation of hydrolase activity", 5.499, 2.817, 5.210, 2.980,-1.0498,0.739,0.036),
c("GO:0051924","regulation of calcium ion transport", 1.206,-6.112, 0.398, 2.322,-1.0498,0.673,0.038),
c("GO:2000113","negative regulation of cellular macromolecule biosynthetic process", 7.576,-1.308,-6.466, 3.119,-1.0498,0.704,0.117),
c("GO:0016192","vesicle-mediated transport",11.385,-6.080,-1.539, 3.295,-1.0498,0.733,0.264));

one.data <- data.frame(revigo.data);
names(one.data) <- revigo.names;
one.data <- one.data [(one.data$plot_X != "null" & one.data$plot_Y != "null"), ];
one.data$plot_X <- as.numeric( as.character(one.data$plot_X) );
one.data$plot_Y <- as.numeric( as.character(one.data$plot_Y) );
one.data$plot_size <- as.numeric( as.character(one.data$plot_size) );
one.data$log10_p_value <- as.numeric( as.character(one.data$log10_p_value) );
one.data$frequency <- as.numeric( as.character(one.data$frequency) );
one.data$uniqueness <- as.numeric( as.character(one.data$uniqueness) );
one.data$dispensability <- as.numeric( as.character(one.data$dispensability) );
#head(one.data);


# --------------------------------------------------------------------------
# Names of the axes, sizes of the numbers and letters, names of the columns,
# etc. can be changed below

p1 <- ggplot( data = one.data );
p1 <- p1 + geom_point( aes( plot_X, plot_Y, colour = log10_p_value, size = plot_size), alpha = I(0.6) ) + scale_size_area();
p1 <- p1 + scale_colour_gradientn( colours = c("blue", "green", "yellow", "red"), limits = c( min(one.data$log10_p_value), 0) );
p1 <- p1 + geom_point( aes(plot_X, plot_Y, size = plot_size), shape = 21, fill = "transparent", colour = I (alpha ("black", 0.6) )) + scale_size_area();
p1 <- p1 + scale_size( range=c(5, 30)) + theme_bw(); # + scale_fill_gradientn(colours = heat_hcl(7), limits = c(-300, 0) );
ex <- one.data [ one.data$dispensability < 0.3, ]; 
p1 <- p1 + geom_text_repel( data = ex, aes(plot_X, plot_Y, label = description), colour = I(alpha("black", 0.85)), size = 4 );
p1 <- p1 + labs (y = "semantic space x", x = "semantic space y");
p1 <- p1 + theme(legend.key = element_blank()) ;
one.x_range = max(one.data$plot_X) - min(one.data$plot_X);
one.y_range = max(one.data$plot_Y) - min(one.data$plot_Y);
p1 <- p1 + xlim(min(one.data$plot_X)-one.x_range/10,max(one.data$plot_X)+one.x_range/10);
p1 <- p1 + ylim(min(one.data$plot_Y)-one.y_range/10,max(one.data$plot_Y)+one.y_range/10);



# --------------------------------------------------------------------------
# Output the plot to screen

p1;



```
K4me1 k4me2 included
```{r}

# --------------------------------------------------------------------------
# If you don't have the ggplot2 package installed, uncomment the following line:
# install.packages( "ggplot2" );
library( ggplot2 );
# --------------------------------------------------------------------------
# If you don't have the scales package installed, uncomment the following line:
# install.packages( "scales" );
library( scales );


# --------------------------------------------------------------------------
# Here is your data from REVIGO. Scroll down for plot configuration options.

revigo.names <- c("term_ID","description","frequency_%","plot_X","plot_Y","plot_size","log10_p_value","uniqueness","dispensability");
revigo.data <- rbind(c("GO:0019320","hexose catabolic process", 0.043,-4.963,-0.724, 3.745,-1.1024,0.021,0.000),
c("GO:0045214","sarcomere organization", 0.013, 5.013,-0.030, 3.220,-0.4202,0.202,0.049),
c("GO:0046365","monosaccharide catabolic process", 0.115,-4.954, 0.784, 4.167,-0.6198,0.020,0.663));

one.data <- data.frame(revigo.data);
names(one.data) <- revigo.names;
one.data <- one.data [(one.data$plot_X != "null" & one.data$plot_Y != "null"), ];
one.data$plot_X <- as.numeric( as.character(one.data$plot_X) );
one.data$plot_Y <- as.numeric( as.character(one.data$plot_Y) );
one.data$plot_size <- as.numeric( as.character(one.data$plot_size) );
one.data$log10_p_value <- as.numeric( as.character(one.data$log10_p_value) );
one.data$frequency <- as.numeric( as.character(one.data$frequency) );
one.data$uniqueness <- as.numeric( as.character(one.data$uniqueness) );
one.data$dispensability <- as.numeric( as.character(one.data$dispensability) );
#head(one.data);


# --------------------------------------------------------------------------
# Names of the axes, sizes of the numbers and letters, names of the columns,
# etc. can be changed below

p1 <- ggplot( data = one.data );
p1 <- p1 + geom_point( aes( plot_X, plot_Y, colour = log10_p_value, size = plot_size), alpha = I(0.6) ) + scale_size_area();
p1 <- p1 + scale_colour_gradientn( colours = c("blue", "green", "yellow", "red"), limits = c( min(one.data$log10_p_value), 0) );
p1 <- p1 + geom_point( aes(plot_X, plot_Y, size = plot_size), shape = 21, fill = "transparent", colour = I (alpha ("black", 0.6) )) + scale_size_area();
p1 <- p1 + scale_size( range=c(5, 30)) + theme_bw(); # + scale_fill_gradientn(colours = heat_hcl(7), limits = c(-300, 0) );
ex <- one.data [ one.data$dispensability < 0.15, ]; 
p1 <- p1 + geom_text( data = ex, aes(plot_X, plot_Y, label = description), colour = I(alpha("black", 0.85)), size = 3 );
p1 <- p1 + labs (y = "semantic space x", x = "semantic space y");
p1 <- p1 + theme(legend.key = element_blank()) ;
one.x_range = max(one.data$plot_X) - min(one.data$plot_X);
one.y_range = max(one.data$plot_Y) - min(one.data$plot_Y);
p1 <- p1 + xlim(min(one.data$plot_X)-one.x_range/10,max(one.data$plot_X)+one.x_range/10);
p1 <- p1 + ylim(min(one.data$plot_Y)-one.y_range/10,max(one.data$plot_Y)+one.y_range/10);



# --------------------------------------------------------------------------
# Output the plot to screen

p1;

###enrichr

--------------------------------------------------------------------
# If you don't have the ggplot2 package installed, uncomment the following line:
# install.packages( "ggplot2" );
library( ggplot2 );
# --------------------------------------------------------------------------
# If you don't have the scales package installed, uncomment the following line:
# install.packages( "scales" );
library( scales );


# --------------------------------------------------------------------------
# Here is your data from REVIGO. Scroll down for plot configuration options.

revigo.names <- c("term_ID","description","frequency_%","plot_X","plot_Y","plot_size","log10_p_value","uniqueness","dispensability");
revigo.data <- rbind(c("GO:0031119","tRNA pseudouridine synthesis", 0.023, 6.179,-4.390, 0.699,-4.1688,0.805,0.000),
c("GO:0090075","relaxation of muscle", 0.173,-2.561,-1.320, 1.491,-2.1427,0.970,0.000),
c("GO:0006614","SRP-dependent cotranslational protein targeting to membrane", 0.537,-3.809, 3.792, 1.973,-3.3525,0.793,0.001),
c("GO:0071312","cellular response to alkaloid", 0.225,-2.661,-7.440, 1.602,-1.6295,0.946,0.002),
c("GO:0016032","viral process", 4.091,-1.068,-3.885, 2.851,-1.9741,0.939,0.002),
c("GO:0006094","gluconeogenesis", 0.462, 1.746, 6.309, 1.908,-2.9617,0.777,0.029),
c("GO:0051402","neuron apoptotic process", 1.269,-6.457,-0.145, 2.344,-1.8442,0.946,0.036),
c("GO:0010467","gene expression",30.542, 3.557,-0.860, 3.724,-2.7246,0.904,0.061),
c("GO:0032479","regulation of type I interferon production", 0.681,-0.742, 0.454, 2.076,-1.8438,0.937,0.092),
c("GO:0045842","positive regulation of mitotic metaphase/anaphase transition", 0.035, 1.048, 2.805, 0.845,-2.3966,0.842,0.100),
c("GO:0050862","positive regulation of T cell receptor signaling pathway", 0.063,-6.060,-1.919, 1.079,-2.2188,0.882,0.121),
c("GO:0007249","I-kappaB kinase/NF-kappaB signaling", 1.535,-5.881,-4.288, 2.427,-1.6001,0.929,0.125),
c("GO:0043631","RNA polyadenylation", 0.248, 3.497,-6.953, 1.643,-1.4060,0.871,0.133),
c("GO:0006333","chromatin assembly or disassembly", 0.906, 4.110, 3.717, 2.199,-2.0094,0.906,0.133),
c("GO:0006412","translation", 3.693, 5.926,-2.363, 2.807,-2.3953,0.746,0.148),
c("GO:0042984","regulation of amyloid precursor protein biosynthetic process", 0.040, 5.874, 1.326, 0.903,-2.3028,0.857,0.157),
c("GO:2000147","positive regulation of cell motility", 2.395,-5.366, 3.256, 2.619,-1.3301,0.854,0.167),
c("GO:0007417","central nervous system development", 5.182,-2.214, 1.013, 2.954,-1.4953,0.962,0.171),
c("GO:0051865","protein autoubiquitination", 0.317, 7.532,-0.338, 1.748,-1.7699,0.880,0.198),
c("GO:0046834","lipid phosphorylation", 0.652,-0.517, 6.407, 2.057,-2.1605,0.855,0.204),
c("GO:0006353","DNA-templated transcription, termination", 0.612, 5.095,-5.571, 2.029,-1.5336,0.849,0.212),
c("GO:0006369","termination of RNA polymerase II transcription", 0.398, 4.461,-6.115, 1.845,-1.4074,0.855,0.223),
c("GO:0008213","protein alkylation", 1.016, 7.789,-0.066, 2.248,-1.3791,0.880,0.227),
c("GO:0043161","proteasome-mediated ubiquitin-dependent protein catabolic process", 2.222, 7.220, 1.663, 2.587,-2.3242,0.803,0.253),
c("GO:0033344","cholesterol efflux", 0.231,-6.009, 2.362, 1.613,-1.5583,0.905,0.278),
c("GO:0034614","cellular response to reactive oxygen species", 0.854,-2.052,-6.884, 2.173,-1.4561,0.923,0.287),
c("GO:0001558","regulation of cell growth", 2.250, 2.992, 4.150, 2.592,-2.1773,0.862,0.297),
c("GO:0032418","lysosome localization", 0.375,-4.382, 4.609, 1.820,-1.3791,0.903,0.300),
c("GO:0050687","negative regulation of defense response to virus", 0.121,-4.201,-4.398, 1.342,-2.1427,0.907,0.323),
c("GO:1901222","regulation of NIK/NF-kappaB signaling", 0.329,-6.196,-3.196, 1.763,-1.4060,0.924,0.324),
c("GO:0033209","tumor necrosis factor-mediated signaling pathway", 0.964,-3.699,-6.205, 2.225,-1.3347,0.907,0.332),
c("GO:0044267","cellular protein metabolic process",28.505, 6.539,-0.412, 3.694,-1.7643,0.853,0.340),
c("GO:0006090","pyruvate metabolic process", 0.606, 0.618, 6.649, 2.025,-1.8420,0.831,0.355),
c("GO:0006376","mRNA splice site selection", 0.150, 4.615,-3.671, 1.431,-2.6145,0.717,0.375),
c("GO:0010565","regulation of cellular ketone metabolic process", 0.964, 1.096, 6.579, 2.225,-1.5253,0.817,0.381),
c("GO:0031293","membrane protein intracellular domain proteolysis", 0.104, 6.901, 3.140, 1.279,-1.5252,0.904,0.382),
c("GO:0006576","cellular biogenic amine metabolic process", 0.306, 1.755,-7.959, 1.732,-2.2188,0.835,0.384),
c("GO:0006265","DNA topological change", 0.052, 1.814,-3.739, 1.000,-2.3028,0.827,0.414),
c("GO:0031113","regulation of microtubule polymerization", 0.196, 2.811, 2.730, 1.544,-2.1605,0.825,0.423),
c("GO:0035967","cellular response to topologically incorrect protein", 0.854,-2.436,-6.697, 2.173,-1.4340,0.931,0.424),
c("GO:0051168","nuclear export", 1.143,-4.643, 3.886, 2.299,-1.7307,0.876,0.427),
c("GO:0090208","positive regulation of triglyceride metabolic process", 0.110,-1.251, 5.934, 1.301,-1.8442,0.844,0.437),
c("GO:0031123","RNA 3'-end processing", 0.675, 4.576,-5.049, 2.072,-1.5253,0.794,0.485),
c("GO:0006281","DNA repair", 2.977, 0.889,-5.805, 2.713,-1.4118,0.803,0.489),
c("GO:2001233","regulation of apoptotic signaling pathway", 2.170,-5.549,-2.842, 2.576,-1.4935,0.893,0.491));

one.data <- data.frame(revigo.data);
names(one.data) <- revigo.names;
one.data <- one.data [(one.data$plot_X != "null" & one.data$plot_Y != "null"), ];
one.data$plot_X <- as.numeric( as.character(one.data$plot_X) );
one.data$plot_Y <- as.numeric( as.character(one.data$plot_Y) );
one.data$plot_size <- as.numeric( as.character(one.data$plot_size) );
one.data$log10_p_value <- as.numeric( as.character(one.data$log10_p_value) );
one.data$frequency <- as.numeric( as.character(one.data$frequency) );
one.data$uniqueness <- as.numeric( as.character(one.data$uniqueness) );
one.data$dispensability <- as.numeric( as.character(one.data$dispensability) );
#head(one.data);


# --------------------------------------------------------------------------
# Names of the axes, sizes of the numbers and letters, names of the columns,
# etc. can be changed below

p1 <- ggplot( data = one.data );
p1 <- p1 + geom_point( aes( plot_X, plot_Y, colour = log10_p_value, size = plot_size), alpha = I(0.6) ) + scale_size_area();
p1 <- p1 + scale_colour_gradientn( colours = c("blue", "green", "yellow", "red"), limits = c( min(one.data$log10_p_value), 0) );
p1 <- p1 + geom_point( aes(plot_X, plot_Y, size = plot_size), shape = 21, fill = "transparent", colour = I (alpha ("black", 0.6) )) + scale_size_area();
p1 <- p1 + scale_size( range=c(5, 30)) + theme_bw(); # + scale_fill_gradientn(colours = heat_hcl(7), limits = c(-300, 0) );
ex <- one.data [ one.data$dispensability < 0.15, ]; 
p1 <- p1 + geom_text( data = ex, aes(plot_X, plot_Y, label = description), colour = I(alpha("black", 0.85)), size = 3 );
p1 <- p1 + labs (y = "semantic space x", x = "semantic space y");
p1 <- p1 + theme(legend.key = element_blank()) ;
one.x_range = max(one.data$plot_X) - min(one.data$plot_X);
one.y_range = max(one.data$plot_Y) - min(one.data$plot_Y);
p1 <- p1 + xlim(min(one.data$plot_X)-one.x_range/10,max(one.data$plot_X)+one.x_range/10);
p1 <- p1 + ylim(min(one.data$plot_Y)-one.y_range/10,max(one.data$plot_Y)+one.y_range/10);



# --------------------------------------------------------------------------
# Output the plot to screen

p1;

##filter enrichr
#cut -f 2,3 /home/eneritz/ASCHROM0818/GO_terms_supfig4/K4me1_K4me2_GO.filter > lala

# --------------------------------------------------------------------------
# Here is your data from REVIGO. Scroll down for plot configuration options.

revigo.names <- c("term_ID","description","frequency_%","plot_X","plot_Y","plot_size","log10_p_value","uniqueness","dispensability");
revigo.data <- rbind(c("GO:0031119","tRNA pseudouridine synthesis", 0.023,-6.159,-4.181, 0.699,-4.1688,0.806,0.000),
c("GO:0090075","relaxation of muscle", 0.173, 2.223,-1.602, 1.491,-2.1427,0.970,0.000),
c("GO:0006614","SRP-dependent cotranslational protein targeting to membrane", 0.537, 3.626, 3.711, 1.973,-3.3525,0.791,0.001),
c("GO:0071312","cellular response to alkaloid", 0.225, 2.739,-7.169, 1.602,-1.6295,0.945,0.002),
c("GO:0016032","viral process", 4.091, 1.044,-3.877, 2.851,-1.9741,0.939,0.002),
c("GO:0006094","gluconeogenesis", 0.462,-1.742, 6.431, 1.908,-2.9617,0.776,0.029),
c("GO:0051402","neuron apoptotic process", 1.269, 5.743,-0.521, 2.344,-1.8442,0.946,0.036),
c("GO:0010467","gene expression",30.542,-3.463,-1.265, 3.724,-2.7246,0.905,0.061),
c("GO:0032479","regulation of type I interferon production", 0.681, 0.487, 0.218, 2.076,-1.8438,0.937,0.092),
c("GO:0045842","positive regulation of mitotic metaphase/anaphase transition", 0.035,-0.859, 3.049, 0.845,-2.3966,0.840,0.100),
c("GO:0050862","positive regulation of T cell receptor signaling pathway", 0.063, 6.467,-1.610, 1.079,-2.2188,0.881,0.121),
c("GO:0007249","I-kappaB kinase/NF-kappaB signaling", 1.535, 5.775,-4.063, 2.427,-1.6001,0.929,0.125),
c("GO:0043631","RNA polyadenylation", 0.248,-3.698,-6.927, 1.643,-1.4060,0.871,0.133),
c("GO:0006333","chromatin assembly or disassembly", 0.906,-3.695, 3.270, 2.199,-2.0094,0.905,0.133),
c("GO:0006412","translation", 3.693,-6.105,-2.138, 2.807,-2.3953,0.748,0.148),
c("GO:0042984","regulation of amyloid precursor protein biosynthetic process", 0.040,-5.768, 1.474, 0.903,-2.3028,0.858,0.157),
c("GO:2000147","positive regulation of cell motility", 2.395, 5.594, 3.424, 2.619,-1.3301,0.853,0.167),
c("GO:0007417","central nervous system development", 5.182, 1.877, 0.727, 2.954,-1.4953,0.962,0.171),
c("GO:0051865","protein autoubiquitination", 0.317,-7.331, 0.222, 1.748,-1.7699,0.889,0.198),
c("GO:0046834","lipid phosphorylation", 0.652, 0.665, 6.413, 2.057,-2.1605,0.855,0.204),
c("GO:0006353","DNA-templated transcription, termination", 0.612,-5.373,-5.420, 2.029,-1.5336,0.850,0.212),
c("GO:0006369","termination of RNA polymerase II transcription", 0.398,-4.952,-6.049, 1.845,-1.4074,0.855,0.223),
c("GO:0008213","protein alkylation", 1.016,-7.733, 0.346, 2.248,-1.3791,0.884,0.227),
c("GO:0043161","proteasome-mediated ubiquitin-dependent protein catabolic process", 2.222,-7.167, 1.971, 2.587,-2.3242,0.806,0.253),
c("GO:0033344","cholesterol efflux", 0.231, 5.812, 2.334, 1.613,-1.5583,0.904,0.278),
c("GO:0034614","cellular response to reactive oxygen species", 0.854, 2.022,-7.024, 2.173,-1.4561,0.922,0.287),
c("GO:0001558","regulation of cell growth", 2.250,-2.799, 4.253, 2.592,-2.1773,0.861,0.297),
c("GO:0032418","lysosome localization", 0.375, 4.344, 4.177, 1.820,-1.3791,0.903,0.300),
c("GO:0050687","negative regulation of defense response to virus", 0.121, 4.205,-4.613, 1.342,-2.1427,0.906,0.323),
c("GO:1901222","regulation of NIK/NF-kappaB signaling", 0.329, 6.346,-3.035, 1.763,-1.4060,0.923,0.324),
c("GO:0033209","tumor necrosis factor-mediated signaling pathway", 0.964, 3.859,-6.201, 2.225,-1.3347,0.906,0.332),
c("GO:0044267","cellular protein metabolic process",28.505,-6.740,-0.357, 3.694,-1.7643,0.857,0.340),
c("GO:0006090","pyruvate metabolic process", 0.606,-0.235, 6.760, 2.025,-1.8420,0.831,0.355),
c("GO:0006376","mRNA splice site selection", 0.150,-4.928,-3.591, 1.431,-2.6145,0.716,0.375),
c("GO:0010565","regulation of cellular ketone metabolic process", 0.964,-0.877, 6.653, 2.225,-1.5253,0.817,0.381),
c("GO:0031293","membrane protein intracellular domain proteolysis", 0.104,-6.786, 3.375, 1.279,-1.5252,0.905,0.382),
c("GO:0006576","cellular biogenic amine metabolic process", 0.306,-1.949,-7.591, 1.732,-2.2188,0.834,0.384),
c("GO:0006265","DNA topological change", 0.052,-1.929,-4.043, 1.000,-2.3028,0.827,0.414),
c("GO:0031113","regulation of microtubule polymerization", 0.196,-2.709, 2.861, 1.544,-2.1605,0.823,0.423),
c("GO:0035967","cellular response to topologically incorrect protein", 0.854, 2.430,-6.950, 2.173,-1.4340,0.930,0.424),
c("GO:0051168","nuclear export", 1.143, 4.560, 3.550, 2.299,-1.7307,0.875,0.427),
c("GO:0090208","positive regulation of triglyceride metabolic process", 0.110, 1.693, 6.051, 1.301,-1.8442,0.844,0.437),
c("GO:0031123","RNA 3'-end processing", 0.675,-4.656,-5.083, 2.072,-1.5253,0.794,0.485),
c("GO:0006281","DNA repair", 2.977,-0.873,-6.029, 2.713,-1.4118,0.803,0.489),
c("GO:2001233","regulation of apoptotic signaling pathway", 2.170, 5.673,-3.085, 2.576,-1.4935,0.893,0.491));

one.data <- data.frame(revigo.data);
names(one.data) <- revigo.names;
one.data <- one.data [(one.data$plot_X != "null" & one.data$plot_Y != "null"), ];
one.data$plot_X <- as.numeric( as.character(one.data$plot_X) );
one.data$plot_Y <- as.numeric( as.character(one.data$plot_Y) );
one.data$plot_size <- as.numeric( as.character(one.data$plot_size) );
one.data$log10_p_value <- as.numeric( as.character(one.data$log10_p_value) );
one.data$frequency <- as.numeric( as.character(one.data$frequency) );
one.data$uniqueness <- as.numeric( as.character(one.data$uniqueness) );
one.data$dispensability <- as.numeric( as.character(one.data$dispensability) );
#head(one.data);


# --------------------------------------------------------------------------
# Names of the axes, sizes of the numbers and letters, names of the columns,
# etc. can be changed below

p1 <- ggplot( data = one.data );
p1 <- p1 + geom_point( aes( plot_X, plot_Y, colour = log10_p_value, size = plot_size), alpha = I(0.6) ) + scale_size_area();
p1 <- p1 + scale_colour_gradientn( colours = c("blue", "green", "yellow", "red"), limits = c( min(one.data$log10_p_value), 0) );
p1 <- p1 + geom_point( aes(plot_X, plot_Y, size = plot_size), shape = 21, fill = "transparent", colour = I (alpha ("black", 0.6) )) + scale_size_area();
p1 <- p1 + scale_size( range=c(5, 30)) + theme_bw(); # + scale_fill_gradientn(colours = heat_hcl(7), limits = c(-300, 0) );
ex <- one.data [ one.data$dispensability < 0.15, ]; 
p1 <- p1 + geom_text( data = ex, aes(plot_X, plot_Y, label = description), colour = I(alpha("black", 0.85)), size = 3 );
p1 <- p1 + labs (y = "semantic space x", x = "semantic space y");
p1 <- p1 + theme(legend.key = element_blank()) ;
one.x_range = max(one.data$plot_X) - min(one.data$plot_X);
one.y_range = max(one.data$plot_Y) - min(one.data$plot_Y);
p1 <- p1 + xlim(min(one.data$plot_X)-one.x_range/10,max(one.data$plot_X)+one.x_range/10);
p1 <- p1 + ylim(min(one.data$plot_Y)-one.y_range/10,max(one.data$plot_Y)+one.y_range/10);



# --------------------------------------------------------------------------
# Output the plot to screen

p1;

 cut -f 2,4 K4me1_K4me2_GO.filter_2  > lala



# --------------------------------------------------------------------------
# Here is your data from REVIGO. Scroll down for plot configuration options.

revigo.names <- c("term_ID","description","frequency_%","plot_X","plot_Y","plot_size","log10_p_value","uniqueness","dispensability");
revigo.data <- rbind(c("GO:0031119","tRNA pseudouridine synthesis", 0.023,-3.396, 5.382, 0.699,-1.0197,0.346,0.000),
c("GO:0006614","SRP-dependent cotranslational protein targeting to membrane", 0.537, 2.999,-4.856, 1.973,-0.6804,0.689,0.001),
c("GO:0006094","gluconeogenesis", 0.462,-5.317,-2.934, 1.908,-0.5603,0.415,0.029),
c("GO:0010467","gene expression",30.542, 4.921, 3.460, 3.724,-0.5603,0.688,0.061));

one.data <- data.frame(revigo.data);
names(one.data) <- revigo.names;
one.data <- one.data [(one.data$plot_X != "null" & one.data$plot_Y != "null"), ];
one.data$plot_X <- as.numeric( as.character(one.data$plot_X) );
one.data$plot_Y <- as.numeric( as.character(one.data$plot_Y) );
one.data$plot_size <- as.numeric( as.character(one.data$plot_size) );
one.data$log10_p_value <- as.numeric( as.character(one.data$log10_p_value) );
one.data$frequency <- as.numeric( as.character(one.data$frequency) );
one.data$uniqueness <- as.numeric( as.character(one.data$uniqueness) );
one.data$dispensability <- as.numeric( as.character(one.data$dispensability) );
#head(one.data);


# --------------------------------------------------------------------------
# Names of the axes, sizes of the numbers and letters, names of the columns,
# etc. can be changed below

p1 <- ggplot( data = one.data );
p1 <- p1 + geom_point( aes( plot_X, plot_Y, colour = log10_p_value, size = plot_size), alpha = I(0.6) ) + scale_size_area();
p1 <- p1 + scale_colour_gradientn( colours = c("blue", "green", "yellow", "red"), limits = c( min(one.data$log10_p_value), 0) );
p1 <- p1 + geom_point( aes(plot_X, plot_Y, size = plot_size), shape = 21, fill = "transparent", colour = I (alpha ("black", 0.6) )) + scale_size_area();
p1 <- p1 + scale_size( range=c(5, 30)) + theme_bw(); # + scale_fill_gradientn(colours = heat_hcl(7), limits = c(-300, 0) );
ex <- one.data [ one.data$dispensability < 0.3, ]; 
p1 <- p1 + geom_text_repel( data = ex, aes(plot_X, plot_Y, label = description), colour = I(alpha("black", 0.85)), size = 4 );
p1 <- p1 + labs (y = "semantic space x", x = "semantic space y");
p1 <- p1 + theme(legend.key = element_blank()) ;
one.x_range = max(one.data$plot_X) - min(one.data$plot_X);
one.y_range = max(one.data$plot_Y) - min(one.data$plot_Y);
p1 <- p1 + xlim(min(one.data$plot_X)-one.x_range/10,max(one.data$plot_X)+one.x_range/10);
p1 <- p1 + ylim(min(one.data$plot_Y)-one.y_range/10,max(one.data$plot_Y)+one.y_range/10);



# --------------------------------------------------------------------------
# Output the plot to screen

p1;




```

K9 5mC included
```{r}






##enrichr



# --------------------------------------------------------------------------
# If you don't have the ggplot2 package installed, uncomment the following line:
# install.packages( "ggplot2" );
library( ggplot2 );
# --------------------------------------------------------------------------
# If you don't have the scales package installed, uncomment the following line:
# install.packages( "scales" );
library( scales );


# --------------------------------------------------------------------------
# Here is your data from REVIGO. Scroll down for plot configuration options.

revigo.names <- c("term_ID","description","frequency_%","plot_X","plot_Y","plot_size","log10_p_value","uniqueness","dispensability");
revigo.data <- rbind(c("GO:0007160","cell-matrix adhesion", 1.148,-2.229,-0.524, 2.301,-1.8479,0.953,0.000),
c("GO:0010561","negative regulation of glycoprotein biosynthetic process", 0.058, 1.545, 7.098, 1.041,-5.0395,0.865,0.000),
c("GO:0031032","actomyosin structure organization", 0.975, 1.785,-6.177, 2.230,-3.1602,0.880,0.002),
c("GO:0001944","vasculature development", 3.480, 2.535,-2.938, 2.781,-2.4582,0.910,0.013),
c("GO:0048145","regulation of fibroblast proliferation", 0.490, 0.048,-0.335, 1.934,-3.8750,0.925,0.031),
c("GO:0070243","regulation of thymocyte apoptotic process", 0.069, 1.299,-0.279, 1.114,-3.2010,0.905,0.032),
c("GO:0042113","B cell activation", 1.598,-4.473,-2.529, 2.444,-2.0913,0.889,0.040),
c("GO:0048017","inositol lipid-mediated signaling", 1.212,-8.204, 2.450, 2.324,-2.8491,0.877,0.041),
c("GO:0030334","regulation of cell migration", 3.912,-1.482,-7.233, 2.832,-2.7081,0.851,0.052),
c("GO:0010564","regulation of cell cycle process", 3.393,-2.626,-1.633, 2.770,-1.8479,0.890,0.060),
c("GO:0043161","proteasome-mediated ubiquitin-dependent protein catabolic process", 2.222, 3.146, 5.950, 2.587,-2.9113,0.895,0.065),
c("GO:0000375","RNA splicing, via transesterification reactions", 1.858, 0.371, 6.525, 2.509,-2.0412,0.922,0.095),
c("GO:0036092","phosphatidylinositol-3-phosphate biosynthetic process", 0.289, 0.639, 2.148, 1.708,-2.9701,0.870,0.098),
c("GO:1900223","positive regulation of beta-amyloid clearance", 0.006, 5.356,-0.584, 0.301,-1.4178,0.903,0.104),
c("GO:2000739","regulation of mesenchymal stem cell differentiation", 0.035, 3.198,-2.619, 0.845,-1.4178,0.887,0.129),
c("GO:0090400","stress-induced premature senescence", 0.046,-3.771, 3.008, 0.954,-1.3110,0.873,0.144),
c("GO:0045714","regulation of low-density lipoprotein particle receptor biosynthetic process", 0.052, 2.464, 6.680, 1.000,-1.3610,0.889,0.145),
c("GO:0015914","phospholipid transport", 0.340,-1.390,-6.007, 1.778,-2.3649,0.922,0.146),
c("GO:0031334","positive regulation of protein complex assembly", 1.344, 5.621,-3.438, 2.369,-2.8162,0.793,0.148),
c("GO:0090557","establishment of endothelial intestinal barrier", 0.075, 3.273,-2.957, 1.146,-1.4178,0.895,0.155),
c("GO:0030198","extracellular matrix organization", 1.829, 0.872,-4.798, 2.502,-2.0356,0.906,0.160),
c("GO:0010507","negative regulation of autophagy", 0.387,-0.480, 5.363, 1.833,-2.6461,0.893,0.160),
c("GO:0051972","regulation of telomerase activity", 0.283, 3.175, 6.923, 1.699,-1.6239,0.867,0.172),
c("GO:0009059","macromolecule biosynthetic process",30.127, 2.630, 7.622, 3.718,-2.2649,0.922,0.182),
c("GO:0019221","cytokine-mediated signaling pathway", 3.260,-7.442, 0.595, 2.753,-2.5549,0.846,0.185),
c("GO:0032846","positive regulation of homeostatic process", 1.218, 5.244, 2.871, 2.326,-1.9413,0.900,0.186),
c("GO:0018149","peptide cross-linking", 0.340, 2.449, 4.822, 1.778,-1.5350,0.929,0.189),
c("GO:0060272","embryonic skeletal joint morphogenesis", 0.069, 1.323,-2.191, 1.114,-1.4178,0.923,0.190),
c("GO:0051169","nuclear transport", 2.752,-0.508,-5.296, 2.679,-2.1453,0.926,0.191),
c("GO:2000379","positive regulation of reactive oxygen species metabolic process", 0.502, 4.488, 3.895, 1.944,-1.6828,0.873,0.193),
c("GO:0033234","negative regulation of protein sumoylation", 0.052, 3.101, 5.846, 1.000,-2.8747,0.839,0.232),
c("GO:0034163","regulation of toll-like receptor 9 signaling pathway", 0.023,-4.868, 3.588, 0.699,-1.3610,0.844,0.241),
c("GO:0060192","negative regulation of lipase activity", 0.087, 1.164, 3.324, 1.204,-1.3110,0.952,0.262),
c("GO:0006446","regulation of translational initiation", 0.415, 3.470, 7.119, 1.863,-2.2850,0.841,0.263),
c("GO:0042053","regulation of dopamine metabolic process", 0.104, 1.819, 7.742, 1.279,-1.4178,0.909,0.283),
c("GO:0035024","negative regulation of Rho protein signal transduction", 0.098,-6.022, 5.141, 1.255,-2.5146,0.842,0.283),
c("GO:0051091","positive regulation of sequence-specific DNA binding transcription factor activity", 1.362, 4.627, 5.109, 2.375,-1.4896,0.833,0.286),
c("GO:0051788","response to misfolded protein", 0.133,-5.734, 1.405, 1.380,-1.3610,0.931,0.299),
c("GO:0016192","vesicle-mediated transport",11.385,-1.098,-6.746, 3.295,-2.0953,0.943,0.302),
c("GO:0035329","hippo signaling", 0.202,-7.850, 2.229, 1.556,-1.8934,0.894,0.302),
c("GO:0048205","COPI coating of Golgi vesicle", 0.023, 0.977,-5.645, 0.699,-1.4178,0.870,0.314),
c("GO:0010912","positive regulation of isomerase activity", 0.029, 1.426, 3.228, 0.778,-1.3610,0.954,0.316),
c("GO:0019673","GDP-mannose metabolic process", 0.040,-0.655, 2.154, 0.903,-1.4178,0.955,0.318),
c("GO:1902416","positive regulation of mRNA binding", 0.035, 1.653, 3.542, 0.845,-1.3610,0.952,0.320),
c("GO:0038128","ERBB2 signaling pathway", 0.225,-6.426, 0.446, 1.602,-1.6828,0.904,0.325),
c("GO:0098700","neurotransmitter loading into synaptic vesicle", 0.040,-0.373,-4.902, 0.903,-1.4178,0.935,0.329),
c("GO:0051897","positive regulation of protein kinase B signaling", 0.490,-5.524, 4.552, 1.934,-1.4627,0.838,0.338),
c("GO:0042770","signal transduction in response to DNA damage", 0.716,-6.578, 3.684, 2.097,-2.0412,0.851,0.343),
c("GO:0050771","negative regulation of axonogenesis", 0.289, 5.258,-2.317, 1.708,-2.1090,0.746,0.344),
c("GO:1901216","positive regulation of neuron death", 0.398, 5.450, 1.780, 1.845,-2.8812,0.856,0.346),
c("GO:0033690","positive regulation of osteoblast proliferation", 0.063, 3.187, 1.017, 1.079,-1.4178,0.906,0.348),
c("GO:0046184","aldehyde biosynthetic process", 0.087, 0.295, 2.370, 1.204,-1.3110,0.911,0.349),
c("GO:0036003","positive regulation of transcription from RNA polymerase II promoter in response to stress", 0.144,-0.807, 7.327, 1.415,-1.9208,0.810,0.351),
c("GO:0034329","cell junction assembly", 1.137, 2.302,-7.190, 2.297,-1.3579,0.885,0.356),
c("GO:0048015","phosphatidylinositol-mediated signaling", 1.194,-8.146, 2.488, 2.318,-2.4072,0.872,0.363),
c("GO:0032933","SREBP signaling pathway", 0.046,-2.147, 6.969, 0.954,-1.4178,0.811,0.369),
c("GO:0010665","regulation of cardiac muscle cell apoptotic process", 0.167, 1.425,-0.415, 1.477,-1.4178,0.904,0.372),
c("GO:0072331","signal transduction by p53 class mediator", 1.518,-8.244, 2.585, 2.422,-1.6828,0.874,0.373),
c("GO:0036294","cellular response to decreased oxygen levels", 1.114,-6.910,-1.045, 2.288,-1.2955,0.922,0.394),
c("GO:0036109","alpha-linolenic acid metabolic process", 0.075,-0.871, 0.720, 1.146,-2.5752,0.904,0.397),
c("GO:0010604","positive regulation of macromolecule metabolic process",16.613, 5.109, 5.606, 3.459,-1.7167,0.824,0.402),
c("GO:0001738","morphogenesis of a polarized epithelium", 0.779, 1.777,-2.192, 2.134,-1.3610,0.914,0.409),
c("GO:0009267","cellular response to starvation", 0.727,-5.972, 3.276, 2.104,-1.6517,0.900,0.409),
c("GO:1901796","regulation of signal transduction by p53 class mediator", 0.964,-7.666, 2.792, 2.225,-1.3732,0.864,0.416),
c("GO:2000427","positive regulation of apoptotic cell clearance", 0.046, 4.555,-3.958, 0.954,-1.3610,0.828,0.416),
c("GO:0046825","regulation of protein export from nucleus", 0.185,-0.028,-5.325, 1.519,-1.8670,0.882,0.419),
c("GO:0033622","integrin activation", 0.104, 1.521,-5.313, 1.279,-1.3110,0.917,0.419),
c("GO:0036159","inner dynein arm assembly", 0.075, 2.518,-6.098, 1.146,-2.4054,0.843,0.422),
c("GO:0030518","intracellular steroid hormone receptor signaling pathway", 0.739,-7.340,-0.080, 2.111,-2.2105,0.874,0.427),
c("GO:0001558","regulation of cell growth", 2.250, 2.857,-4.790, 2.592,-2.2674,0.840,0.431),
c("GO:0007157","heterophilic cell-cell adhesion via plasma membrane cell adhesion molecules", 0.265,-2.135,-0.516, 1.672,-1.5350,0.954,0.434),
c("GO:0018212","peptidyl-tyrosine modification", 2.147, 4.207, 5.751, 2.572,-1.4710,0.908,0.436),
c("GO:0031399","regulation of protein modification process", 9.683, 4.041, 6.917, 3.225,-1.6239,0.827,0.438),
c("GO:0072672","neutrophil extravasation", 0.046,-3.331,-4.118, 0.954,-1.4178,0.896,0.439),
c("GO:0034599","cellular response to oxidative stress", 1.448,-5.992, 1.856, 2.401,-1.5715,0.888,0.441),
c("GO:0016055","Wnt signaling pathway", 2.724,-7.991, 1.158, 2.675,-2.4810,0.870,0.441),
c("GO:0001774","microglial cell activation", 0.104,-2.966,-1.735, 1.279,-1.3610,0.908,0.449),
c("GO:2000587","negative regulation of platelet-derived growth factor receptor-beta signaling pathway", 0.046,-5.193, 3.829, 0.954,-1.3610,0.864,0.450),
c("GO:0051382","kinetochore assembly", 0.081, 1.948,-6.891, 1.176,-1.3110,0.894,0.453),
c("GO:0035754","B cell chemotaxis", 0.046,-5.835,-2.818, 0.954,-1.4178,0.849,0.460),
c("GO:0038114","interleukin-21-mediated signaling pathway", 0.006,-6.293,-0.787, 0.301,-1.3110,0.897,0.462),
c("GO:0097400","interleukin-17-mediated signaling pathway", 0.006,-6.840,-1.310, 0.301,-1.3110,0.897,0.462),
c("GO:2000113","negative regulation of cellular macromolecule biosynthetic process", 7.576, 2.673, 7.363, 3.119,-1.6147,0.801,0.463),
c("GO:0031330","negative regulation of cellular catabolic process", 0.525, 3.136, 7.935, 1.964,-1.3982,0.852,0.478),
c("GO:0045843","negative regulation of striated muscle tissue development", 0.225, 5.680,-0.291, 1.602,-1.3610,0.836,0.487),
c("GO:0033034","positive regulation of myeloid cell apoptotic process", 0.052, 4.202, 1.188, 1.000,-1.3610,0.880,0.489),
c("GO:0034983","peptidyl-lysine deacetylation", 0.040, 2.998, 5.379, 0.903,-1.3610,0.930,0.491),
c("GO:2000643","positive regulation of early endosome to late endosome transport", 0.046, 5.028,-4.219, 0.954,-1.3110,0.870,0.491),
c("GO:0051291","protein heterooligomerization", 0.565, 2.131,-7.075, 1.996,-1.5023,0.906,0.494),
c("GO:1902531","regulation of intracellular signal transduction",10.006,-7.180, 2.856, 3.239,-2.0305,0.833,0.495),
c("GO:0002052","positive regulation of neuroblast proliferation", 0.115, 4.910,-0.498, 1.322,-1.3610,0.805,0.496),
c("GO:0000281","mitotic cytokinesis", 0.214,-3.084,-2.316, 1.580,-1.3579,0.934,0.496));

one.data <- data.frame(revigo.data);
names(one.data) <- revigo.names;
one.data <- one.data [(one.data$plot_X != "null" & one.data$plot_Y != "null"), ];
one.data$plot_X <- as.numeric( as.character(one.data$plot_X) );
one.data$plot_Y <- as.numeric( as.character(one.data$plot_Y) );
one.data$plot_size <- as.numeric( as.character(one.data$plot_size) );
one.data$log10_p_value <- as.numeric( as.character(one.data$log10_p_value) );
one.data$frequency <- as.numeric( as.character(one.data$frequency) );
one.data$uniqueness <- as.numeric( as.character(one.data$uniqueness) );
one.data$dispensability <- as.numeric( as.character(one.data$dispensability) );
#head(one.data);


# --------------------------------------------------------------------------
# Names of the axes, sizes of the numbers and letters, names of the columns,
# etc. can be changed below

p1 <- ggplot( data = one.data );
p1 <- p1 + geom_point( aes( plot_X, plot_Y, colour = log10_p_value, size = plot_size), alpha = I(0.6) ) + scale_size_area();
p1 <- p1 + scale_colour_gradientn( colours = c("blue", "green", "yellow", "red"), limits = c( min(one.data$log10_p_value), 0) );
p1 <- p1 + geom_point( aes(plot_X, plot_Y, size = plot_size), shape = 21, fill = "transparent", colour = I (alpha ("black", 0.6) )) + scale_size_area();
p1 <- p1 + scale_size( range=c(5, 30)) + theme_bw(); # + scale_fill_gradientn(colours = heat_hcl(7), limits = c(-300, 0) );
ex <- one.data [ one.data$dispensability < 0.15, ]; 
p1 <- p1 + geom_text( data = ex, aes(plot_X, plot_Y, label = description), colour = I(alpha("black", 0.85)), size = 3 );
p1 <- p1 + labs (y = "semantic space x", x = "semantic space y");
p1 <- p1 + theme(legend.key = element_blank()) ;
one.x_range = max(one.data$plot_X) - min(one.data$plot_X);
one.y_range = max(one.data$plot_Y) - min(one.data$plot_Y);
p1 <- p1 + xlim(min(one.data$plot_X)-one.x_range/10,max(one.data$plot_X)+one.x_range/10);
p1 <- p1 + ylim(min(one.data$plot_Y)-one.y_range/10,max(one.data$plot_Y)+one.y_range/10);



# --------------------------------------------------------------------------
# Output the plot to screen

p1;

##filter enrichr
#cut -f 2,3 /home/eneritz/ASCHROM0818/GO_terms_supfig4/K9_5mC_I_GO.filter > lala


# --------------------------------------------------------------------------
# Here is your data from REVIGO. Scroll down for plot configuration options.

revigo.names <- c("term_ID","description","frequency_%","plot_X","plot_Y","plot_size","log10_p_value","uniqueness","dispensability");
revigo.data <- rbind(c("GO:0007160","cell-matrix adhesion", 1.148,-2.894,-0.776, 2.301,-1.8479,0.952,0.000),
c("GO:0010561","negative regulation of glycoprotein biosynthetic process", 0.058, 0.695, 7.200, 1.041,-5.0395,0.865,0.000),
c("GO:0031032","actomyosin structure organization", 0.975, 1.740,-6.313, 2.230,-3.1602,0.880,0.002),
c("GO:0001944","vasculature development", 3.480, 2.948,-2.873, 2.781,-2.4582,0.910,0.013),
c("GO:0048145","regulation of fibroblast proliferation", 0.490,-0.138,-0.645, 1.934,-3.8750,0.925,0.031),
c("GO:0070243","regulation of thymocyte apoptotic process", 0.069, 1.647,-0.174, 1.114,-3.2010,0.905,0.032),
c("GO:0042113","B cell activation", 1.598,-5.341,-3.438, 2.444,-2.0913,0.888,0.040),
c("GO:0048017","inositol lipid-mediated signaling", 1.212,-7.798, 2.514, 2.324,-2.8491,0.877,0.041),
c("GO:0030334","regulation of cell migration", 3.912,-2.403,-6.635, 2.832,-2.7081,0.851,0.052),
c("GO:0010564","regulation of cell cycle process", 3.393,-3.363,-2.450, 2.770,-1.8479,0.890,0.060),
c("GO:0043161","proteasome-mediated ubiquitin-dependent protein catabolic process", 2.222, 3.929, 6.210, 2.587,-2.9113,0.895,0.065),
c("GO:0000375","RNA splicing, via transesterification reactions", 1.858,-0.775, 7.326, 2.509,-2.0412,0.921,0.095),
c("GO:0036092","phosphatidylinositol-3-phosphate biosynthetic process", 0.289, 0.377, 2.216, 1.708,-2.9701,0.870,0.098),
c("GO:1900223","positive regulation of beta-amyloid clearance", 0.006, 6.692,-0.577, 0.301,-1.4178,0.903,0.104),
c("GO:2000739","regulation of mesenchymal stem cell differentiation", 0.035, 4.137,-2.358, 0.845,-1.4178,0.886,0.129),
c("GO:0090400","stress-induced premature senescence", 0.046,-3.644, 3.677, 0.954,-1.3110,0.874,0.144),
c("GO:0045714","regulation of low-density lipoprotein particle receptor biosynthetic process", 0.052, 2.324, 6.823, 1.000,-1.3610,0.889,0.145),
c("GO:0015914","phospholipid transport", 0.340,-2.284,-6.118, 1.778,-2.3649,0.921,0.146),
c("GO:0031334","positive regulation of protein complex assembly", 1.344, 5.416,-3.053, 2.369,-2.8162,0.793,0.148),
c("GO:0090557","establishment of endothelial intestinal barrier", 0.075, 3.955,-2.737, 1.146,-1.4178,0.895,0.155),
c("GO:0030198","extracellular matrix organization", 1.829, 0.776,-4.928, 2.502,-2.0356,0.905,0.160),
c("GO:0010507","negative regulation of autophagy", 0.387,-1.417, 6.057, 1.833,-2.6461,0.892,0.160),
c("GO:0051972","regulation of telomerase activity", 0.283, 2.366, 7.080, 1.699,-1.6239,0.867,0.172),
c("GO:0009059","macromolecule biosynthetic process",30.127, 1.963, 7.885, 3.718,-2.2649,0.922,0.182),
c("GO:0019221","cytokine-mediated signaling pathway", 3.260,-7.085, 0.477, 2.753,-2.5549,0.848,0.185),
c("GO:0032846","positive regulation of homeostatic process", 1.218, 6.116, 2.524, 2.326,-1.9413,0.899,0.186),
c("GO:0018149","peptide cross-linking", 0.340, 2.992, 4.985, 1.778,-1.5350,0.929,0.189),
c("GO:0060272","embryonic skeletal joint morphogenesis", 0.069, 1.523,-2.373, 1.114,-1.4178,0.923,0.190),
c("GO:0051169","nuclear transport", 2.752,-1.320,-5.603, 2.679,-2.1453,0.925,0.191),
c("GO:2000379","positive regulation of reactive oxygen species metabolic process", 0.502, 5.088, 3.424, 1.944,-1.6828,0.873,0.193),
c("GO:0033234","negative regulation of protein sumoylation", 0.052, 3.219, 6.016, 1.000,-2.8747,0.839,0.232),
c("GO:0034163","regulation of toll-like receptor 9 signaling pathway", 0.023,-4.723, 3.872, 0.699,-1.3610,0.844,0.241),
c("GO:0060192","negative regulation of lipase activity", 0.087, 1.070, 3.782, 1.204,-1.3110,0.952,0.262),
c("GO:0006446","regulation of translational initiation", 0.415, 3.268, 7.022, 1.863,-2.2850,0.841,0.263),
c("GO:0042053","regulation of dopamine metabolic process", 0.104, 0.741, 8.392, 1.279,-1.4178,0.909,0.283),
c("GO:0035024","negative regulation of Rho protein signal transduction", 0.098,-5.761, 5.239, 1.255,-2.5146,0.842,0.283),
c("GO:0051091","positive regulation of sequence-specific DNA binding transcription factor activity", 1.362, 4.308, 4.526, 2.375,-1.4896,0.832,0.286),
c("GO:0051788","response to misfolded protein", 0.133,-5.357, 1.601, 1.380,-1.3610,0.933,0.299),
c("GO:0016192","vesicle-mediated transport",11.385,-1.914,-6.572, 3.295,-2.0953,0.943,0.302),
c("GO:0035329","hippo signaling", 0.202,-7.898, 1.980, 1.556,-1.8934,0.894,0.302),
c("GO:0048205","COPI coating of Golgi vesicle", 0.023, 0.412,-5.845, 0.699,-1.4178,0.870,0.314),
c("GO:0010912","positive regulation of isomerase activity", 0.029, 1.606, 3.498, 0.778,-1.3610,0.954,0.316),
c("GO:0019673","GDP-mannose metabolic process", 0.040,-1.345, 2.671, 0.903,-1.4178,0.954,0.318),
c("GO:1902416","positive regulation of mRNA binding", 0.035, 1.637, 3.962, 0.845,-1.3610,0.952,0.320),
c("GO:0038128","ERBB2 signaling pathway", 0.225,-6.866,-0.072, 1.602,-1.6828,0.904,0.325),
c("GO:0098700","neurotransmitter loading into synaptic vesicle", 0.040,-1.193,-5.197, 0.903,-1.4178,0.935,0.329),
c("GO:0051897","positive regulation of protein kinase B signaling", 0.490,-5.254, 4.682, 1.934,-1.4627,0.838,0.338),
c("GO:0042770","signal transduction in response to DNA damage", 0.716,-6.373, 3.849, 2.097,-2.0412,0.852,0.343),
c("GO:0050771","negative regulation of axonogenesis", 0.289, 5.213,-1.992, 1.708,-2.1090,0.745,0.344),
c("GO:1901216","positive regulation of neuron death", 0.398, 5.956, 1.557, 1.845,-2.8812,0.856,0.346),
c("GO:0033690","positive regulation of osteoblast proliferation", 0.063, 3.968, 0.985, 1.079,-1.4178,0.906,0.348),
c("GO:0046184","aldehyde biosynthetic process", 0.087,-0.053, 2.623, 1.204,-1.3110,0.911,0.349),
c("GO:0036003","positive regulation of transcription from RNA polymerase II promoter in response to stress", 0.144,-1.451, 6.907, 1.415,-1.9208,0.810,0.351),
c("GO:0034329","cell junction assembly", 1.137, 2.399,-6.770, 2.297,-1.3579,0.884,0.356),
c("GO:0048015","phosphatidylinositol-mediated signaling", 1.194,-7.704, 2.591, 2.318,-2.4072,0.872,0.363),
c("GO:0032933","SREBP signaling pathway", 0.046,-2.477, 6.597, 0.954,-1.4178,0.811,0.369),
c("GO:0010665","regulation of cardiac muscle cell apoptotic process", 0.167, 1.882,-0.300, 1.477,-1.4178,0.903,0.372),
c("GO:0072331","signal transduction by p53 class mediator", 1.518,-7.733, 2.758, 2.422,-1.6828,0.875,0.373),
c("GO:0036109","alpha-linolenic acid metabolic process", 0.075,-1.208, 0.816, 1.146,-2.5752,0.903,0.397),
c("GO:0010604","positive regulation of macromolecule metabolic process",16.613, 5.128, 4.879, 3.459,-1.7167,0.824,0.402),
c("GO:0001738","morphogenesis of a polarized epithelium", 0.779, 2.248,-2.298, 2.134,-1.3610,0.914,0.409),
c("GO:0009267","cellular response to starvation", 0.727,-5.774, 3.649, 2.104,-1.6517,0.900,0.409),
c("GO:1901796","regulation of signal transduction by p53 class mediator", 0.964,-7.258, 3.088, 2.225,-1.3732,0.864,0.416),
c("GO:2000427","positive regulation of apoptotic cell clearance", 0.046, 4.699,-3.767, 0.954,-1.3610,0.827,0.416),
c("GO:0046825","regulation of protein export from nucleus", 0.185,-0.859,-5.687, 1.519,-1.8670,0.881,0.419),
c("GO:0033622","integrin activation", 0.104, 1.881,-5.687, 1.279,-1.3110,0.916,0.419),
c("GO:0036159","inner dynein arm assembly", 0.075, 2.711,-6.059, 1.146,-2.4054,0.842,0.422),
c("GO:0030518","intracellular steroid hormone receptor signaling pathway", 0.739,-7.277,-0.306, 2.111,-2.2105,0.877,0.427),
c("GO:0001558","regulation of cell growth", 2.250, 3.618,-4.900, 2.592,-2.2674,0.839,0.431),
c("GO:0007157","heterophilic cell-cell adhesion via plasma membrane cell adhesion molecules", 0.265,-2.743,-0.757, 1.672,-1.5350,0.954,0.434),
c("GO:0018212","peptidyl-tyrosine modification", 2.147, 5.022, 5.766, 2.572,-1.4710,0.908,0.436),
c("GO:0031399","regulation of protein modification process", 9.683, 3.951, 6.619, 3.225,-1.6239,0.826,0.438),
c("GO:0072672","neutrophil extravasation", 0.046,-4.392,-4.739, 0.954,-1.4178,0.896,0.439),
c("GO:0034599","cellular response to oxidative stress", 1.448,-5.713, 2.124, 2.401,-1.5715,0.890,0.441),
c("GO:0016055","Wnt signaling pathway", 2.724,-7.609, 0.771, 2.675,-2.4810,0.871,0.441),
c("GO:0001774","microglial cell activation", 0.104,-3.908,-2.424, 1.279,-1.3610,0.907,0.449),
c("GO:2000587","negative regulation of platelet-derived growth factor receptor-beta signaling pathway", 0.046,-5.143, 4.267, 0.954,-1.3610,0.864,0.450),
c("GO:0051382","kinetochore assembly", 0.081, 1.870,-6.748, 1.176,-1.3110,0.894,0.453),
c("GO:0035754","B cell chemotaxis", 0.046,-6.039,-3.090, 0.954,-1.4178,0.851,0.460),
c("GO:0038114","interleukin-21-mediated signaling pathway", 0.006,-6.539,-1.093, 0.301,-1.3110,0.898,0.462),
c("GO:0097400","interleukin-17-mediated signaling pathway", 0.006,-7.002,-1.577, 0.301,-1.3110,0.899,0.462),
c("GO:2000113","negative regulation of cellular macromolecule biosynthetic process", 7.576, 1.856, 7.148, 3.119,-1.6147,0.800,0.463),
c("GO:0031330","negative regulation of cellular catabolic process", 0.525, 2.666, 7.874, 1.964,-1.3982,0.851,0.478),
c("GO:0045843","negative regulation of striated muscle tissue development", 0.225, 6.018,-0.484, 1.602,-1.3610,0.835,0.487),
c("GO:0033034","positive regulation of myeloid cell apoptotic process", 0.052, 5.098, 1.198, 1.000,-1.3610,0.879,0.489),
c("GO:0034983","peptidyl-lysine deacetylation", 0.040, 3.835, 5.255, 0.903,-1.3610,0.929,0.491),
c("GO:2000643","positive regulation of early endosome to late endosome transport", 0.046, 5.032,-3.973, 0.954,-1.3110,0.870,0.491),
c("GO:0051291","protein heterooligomerization", 0.565, 2.143,-6.820, 1.996,-1.5023,0.905,0.494),
c("GO:1902531","regulation of intracellular signal transduction",10.006,-6.899, 3.030, 3.239,-2.0305,0.833,0.495),
c("GO:0002052","positive regulation of neuroblast proliferation", 0.115, 5.345,-0.501, 1.322,-1.3610,0.804,0.496),
c("GO:0000281","mitotic cytokinesis", 0.214,-4.067,-3.270, 1.580,-1.3579,0.933,0.496));

one.data <- data.frame(revigo.data);
names(one.data) <- revigo.names;
one.data <- one.data [(one.data$plot_X != "null" & one.data$plot_Y != "null"), ];
one.data$plot_X <- as.numeric( as.character(one.data$plot_X) );
one.data$plot_Y <- as.numeric( as.character(one.data$plot_Y) );
one.data$plot_size <- as.numeric( as.character(one.data$plot_size) );
one.data$log10_p_value <- as.numeric( as.character(one.data$log10_p_value) );
one.data$frequency <- as.numeric( as.character(one.data$frequency) );
one.data$uniqueness <- as.numeric( as.character(one.data$uniqueness) );
one.data$dispensability <- as.numeric( as.character(one.data$dispensability) );
#head(one.data);


# --------------------------------------------------------------------------
# Names of the axes, sizes of the numbers and letters, names of the columns,
# etc. can be changed below

p1 <- ggplot( data = one.data );
p1 <- p1 + geom_point( aes( plot_X, plot_Y, colour = log10_p_value, size = plot_size), alpha = I(0.6) ) + scale_size_area();
p1 <- p1 + scale_colour_gradientn( colours = c("blue", "green", "yellow", "red"), limits = c( min(one.data$log10_p_value), 0) );
p1 <- p1 + geom_point( aes(plot_X, plot_Y, size = plot_size), shape = 21, fill = "transparent", colour = I (alpha ("black", 0.6) )) + scale_size_area();
p1 <- p1 + scale_size( range=c(5, 30)) + theme_bw(); # + scale_fill_gradientn(colours = heat_hcl(7), limits = c(-300, 0) );
ex <- one.data [ one.data$dispensability < 0.15, ]; 
p1 <- p1 + geom_text( data = ex, aes(plot_X, plot_Y, label = description), colour = I(alpha("black", 0.85)), size = 3 );
p1 <- p1 + labs (y = "semantic space x", x = "semantic space y");
p1 <- p1 + theme(legend.key = element_blank()) ;
one.x_range = max(one.data$plot_X) - min(one.data$plot_X);
one.y_range = max(one.data$plot_Y) - min(one.data$plot_Y);
p1 <- p1 + xlim(min(one.data$plot_X)-one.x_range/10,max(one.data$plot_X)+one.x_range/10);
p1 <- p1 + ylim(min(one.data$plot_Y)-one.y_range/10,max(one.data$plot_Y)+one.y_range/10);



# --------------------------------------------------------------------------
# Output the plot to screen

p1;


 cut -f 2,4 K9me3_5mC_I_GO.filter_2  > lala





# --------------------------------------------------------------------------
# Here is your data from REVIGO. Scroll down for plot configuration options.

revigo.names <- c("term_ID","description","frequency_%","plot_X","plot_Y","plot_size","log10_p_value","uniqueness","dispensability");
revigo.data <- rbind(c("GO:0010561","negative regulation of glycoprotein biosynthetic process", 0.058, 1.043,-7.272, 1.041,-1.9691,0.801,0.000),
c("GO:0031032","actomyosin structure organization", 0.975, 5.975,-3.434, 2.230,-0.7458,0.765,0.002),
c("GO:0001944","vasculature development", 3.480,-1.087, 4.054, 2.781,-0.6705,0.878,0.013),
c("GO:0048145","regulation of fibroblast proliferation", 0.490, 2.235, 5.946, 1.934,-1.1056,0.857,0.031),
c("GO:0070243","regulation of thymocyte apoptotic process", 0.069, 4.929, 3.662, 1.114,-0.7458,0.829,0.032),
c("GO:0048017","inositol lipid-mediated signaling", 1.212,-5.681,-3.303, 2.324,-0.7458,0.770,0.041),
c("GO:0030334","regulation of cell migration", 3.912,-1.046, 6.767, 2.832,-0.6790,0.837,0.052),
c("GO:0043161","proteasome-mediated ubiquitin-dependent protein catabolic process", 2.222, 0.014, 0.190, 2.587,-0.7458,0.878,0.065),
c("GO:0036092","phosphatidylinositol-3-phosphate biosynthetic process", 0.289,-5.054, 4.553, 1.708,-0.7458,0.828,0.098),
c("GO:0031334","positive regulation of protein complex assembly", 1.344, 5.050,-1.385, 2.369,-0.7458,0.742,0.148),
c("GO:0010507","negative regulation of autophagy", 0.387,-0.987,-7.396, 1.833,-0.6705,0.814,0.160),
c("GO:0019221","cytokine-mediated signaling pathway", 3.260,-6.833,-1.364, 2.753,-0.6705,0.712,0.185),
c("GO:0033234","negative regulation of protein sumoylation", 0.052, 0.290,-5.555, 1.000,-0.7458,0.799,0.232),
c("GO:0035024","negative regulation of Rho protein signal transduction", 0.098,-3.599,-4.967, 1.255,-0.6705,0.751,0.283),
c("GO:1901216","positive regulation of neuron death", 0.398, 5.205, 2.511, 1.845,-0.7458,0.809,0.346),
c("GO:0048015","phosphatidylinositol-mediated signaling", 1.194,-6.006,-3.906, 2.318,-0.6705,0.770,0.363),
c("GO:0030308","negative regulation of cell growth", 0.923, 2.563,-4.957, 2.207,-0.6705,0.742,0.388),
c("GO:0036109","alpha-linolenic acid metabolic process", 0.075,-5.546, 4.097, 1.146,-0.6705,0.838,0.397),
c("GO:0036159","inner dynein arm assembly", 0.075, 6.112,-2.149, 1.146,-0.6705,0.765,0.422),
c("GO:0016055","Wnt signaling pathway", 2.724,-6.506,-0.869, 2.675,-0.6705,0.762,0.441));

one.data <- data.frame(revigo.data);
names(one.data) <- revigo.names;
one.data <- one.data [(one.data$plot_X != "null" & one.data$plot_Y != "null"), ];
one.data$plot_X <- as.numeric( as.character(one.data$plot_X) );
one.data$plot_Y <- as.numeric( as.character(one.data$plot_Y) );
one.data$plot_size <- as.numeric( as.character(one.data$plot_size) );
one.data$log10_p_value <- as.numeric( as.character(one.data$log10_p_value) );
one.data$frequency <- as.numeric( as.character(one.data$frequency) );
one.data$uniqueness <- as.numeric( as.character(one.data$uniqueness) );
one.data$dispensability <- as.numeric( as.character(one.data$dispensability) );
#head(one.data);


# --------------------------------------------------------------------------
# Names of the axes, sizes of the numbers and letters, names of the columns,
# etc. can be changed below

p1 <- ggplot( data = one.data );
p1 <- p1 + geom_point( aes( plot_X, plot_Y, colour = log10_p_value, size = plot_size), alpha = I(0.6) ) + scale_size_area();
p1 <- p1 + scale_colour_gradientn( colours = c("blue", "green", "yellow", "red"), limits = c( min(one.data$log10_p_value), 0) );
p1 <- p1 + geom_point( aes(plot_X, plot_Y, size = plot_size), shape = 21, fill = "transparent", colour = I (alpha ("black", 0.6) )) + scale_size_area();
p1 <- p1 + scale_size( range=c(5, 30)) + theme_bw(); # + scale_fill_gradientn(colours = heat_hcl(7), limits = c(-300, 0) );
ex <- one.data [ one.data$dispensability < 0.3, ]; 
p1 <- p1 + geom_text_repel( data = ex, aes(plot_X, plot_Y, label = description), colour = I(alpha("black", 0.85)), size = 4 );
p1 <- p1 + labs (y = "semantic space x", x = "semantic space y");
p1 <- p1 + theme(legend.key = element_blank()) ;
one.x_range = max(one.data$plot_X) - min(one.data$plot_X);
one.y_range = max(one.data$plot_Y) - min(one.data$plot_Y);
p1 <- p1 + xlim(min(one.data$plot_X)-one.x_range/10,max(one.data$plot_X)+one.x_range/10);
p1 <- p1 + ylim(min(one.data$plot_Y)-one.y_range/10,max(one.data$plot_Y)+one.y_range/10);



# --------------------------------------------------------------------------
# Output the plot to screen

p1;

```